{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fbef926e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from BCBio.GFF import GFFExaminer"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18ceb6d5",
   "metadata": {},
   "source": [
    "https://gffpandas.readthedocs.io/en/latest/tutorial.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "88084c43",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gffpandas.gffpandas as gffpd\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0e4c4aab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\anaconda3\\lib\\site-packages\\gffpandas\\gffpandas.py:32: DtypeWarning: Columns (1,2,5,6,7,8) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  self.df = pd.read_table(self._gff_file, comment='#',\n"
     ]
    }
   ],
   "source": [
    "annotation = gffpd.read_gff3('YPS1009_S288C_v4_polished.gff')\n",
    "annot_s288c = gffpd.read_gff3('saccharomyces_cerevisiae_R64-3.gff')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "05ec8c7a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CDS                                7178\n",
       "gene                               6707\n",
       "mRNA                               6707\n",
       "noncoding_exon                      493\n",
       "tRNA_gene                           321\n",
       "tRNA                                321\n",
       "transposable_element_gene            99\n",
       "transposable_element                 99\n",
       "snoRNA                               77\n",
       "snoRNA_gene                          77\n",
       "plus_1_translational_frameshift      47\n",
       "five_prime_UTR_intron                24\n",
       "ncRNA_gene                           23\n",
       "ncRNA                                23\n",
       "snRNA_gene                            6\n",
       "snRNA                                 6\n",
       "uORF                                  6\n",
       "telomeric_repeat                      3\n",
       "intein_encoding_region                1\n",
       "Name: type, dtype: int64"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation.df['type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1f380f29",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "mRNA                                  11119\n",
       "CDS                                    7066\n",
       "gene                                   6607\n",
       "ARS                                     543\n",
       "noncoding_exon                          489\n",
       "long_terminal_repeat                    384\n",
       "intron                                  378\n",
       "tRNA                                    299\n",
       "tRNA_gene                               299\n",
       "ARS_consensus_sequence                  196\n",
       "transposable_element_gene                91\n",
       "transposable_element                     91\n",
       "region                                   78\n",
       "snoRNA_gene                              77\n",
       "snoRNA                                   77\n",
       "telomere                                 64\n",
       "LTR_retrotransposon                      50\n",
       "plus_1_translational_frameshift          47\n",
       "X_element                                32\n",
       "centromere                               32\n",
       "telomeric_repeat                         31\n",
       "X_element_combinatorial_repeat           28\n",
       "rRNA                                     24\n",
       "rRNA_gene                                24\n",
       "five_prime_UTR_intron                    24\n",
       "ncRNA                                    23\n",
       "ncRNA_gene                               23\n",
       "Y_prime_element                          19\n",
       "chromosome                               17\n",
       "centromere_DNA_Element_III               16\n",
       "centromere_DNA_Element_II                16\n",
       "centromere_DNA_Element_I                 16\n",
       "pseudogene                               12\n",
       "blocked_reading_frame                    12\n",
       "uORF                                     11\n",
       "internal_transcribed_spacer_region        8\n",
       "external_transcribed_spacer_region        8\n",
       "origin_of_replication                     8\n",
       "matrix_attachment_site                    8\n",
       "snRNA                                     6\n",
       "snRNA_gene                                6\n",
       "silent_mating_type_cassette_array         4\n",
       "X_region                                  3\n",
       "Y_region                                  3\n",
       "non_transcribed_region                    3\n",
       "Z1_region                                 3\n",
       "W_region                                  2\n",
       "Z2_region                                 2\n",
       "mating_type_region                        2\n",
       "intein_encoding_region                    2\n",
       "recombination_enhancer                    1\n",
       "telomerase_RNA                            1\n",
       "telomerase_RNA_gene                       1\n",
       "Name: type, dtype: int64"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annot_s288c.df['type'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1009c702",
   "metadata": {},
   "source": [
    "## Make df by features YPS1009"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "45985f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_df = annotation.filter_feature_of_type(['transposable_element_gene'])\n",
    "sno_df = annotation.filter_feature_of_type(['snoRNA_gene'])\n",
    "tRNA_df = annotation.filter_feature_of_type(['tRNA_gene'])\n",
    "gene_df = annotation.filter_feature_of_type(['gene'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "aa90ef94",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_attr = gene_df.attributes_to_columns()\n",
    "gene = gene_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
    "       'partial_mapping', 'sequence_ID', 'valid_ORFs']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "da9223db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    6175\n",
       "0     530\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr['valid_ORFs'].value_counts()\n",
    "# so valid_ORFs in gene does count itself and I should look carefully at the 439 genes with no valid ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a2feefd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_attr = TE_df.attributes_to_columns()\n",
    "sno_attr = sno_df.attributes_to_columns()\n",
    "tRNA_attr = tRNA_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "65fb5bab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'Ontology_term',\n",
       "       'copy_num_ID', 'coverage', 'curie', 'dbxref', 'display',\n",
       "       'extra_copy_number', 'gene', 'sequence_ID'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "8d70a21e",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE = TE_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'sequence_ID', 'valid_ORFs']]\n",
    "\n",
    "sno = sno_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'sequence_ID']]\n",
    "\n",
    "tRNA = tRNA_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'sequence_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "4b9d5296",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This code just convert the chromosome numbering to integer and remove mitochondrial genes.\n",
    "def lat_to_arabic(DF):\n",
    "    Latin_to_arabic = {'I': 1, 'II':2, 'III':3, 'IV':4, 'V':5, 'VI':6, 'VII':7, 'VIII':8, 'IX':9, 'X':10,\n",
    "                      'XI':11, 'XII':12, 'XIII':13, 'XIV':14, 'XV':15, 'XVI':16, 'Mito': 'mitoch', 'MANY': 7}\n",
    "    seq_id = DF['seq_id']\n",
    "    ch = []\n",
    "    contig = []\n",
    "    for i in seq_id:\n",
    "        chrom, cont = i.split('_')\n",
    "        contig.append(cont)\n",
    "        chrom = chrom.replace('chr', '')\n",
    "        if '-' in chrom:\n",
    "            chrom, e = chrom.split('-')\n",
    "        ch.append(Latin_to_arabic[chrom])\n",
    "    DF['Chromosome'] = ch\n",
    "    DF['Contig'] = contig\n",
    "    return DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "aae0f024",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n"
     ]
    }
   ],
   "source": [
    "gene = lat_to_arabic(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "47a1360f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n"
     ]
    }
   ],
   "source": [
    "TE = lat_to_arabic(TE)\n",
    "sno = lat_to_arabic(sno)\n",
    "tRNA = lat_to_arabic(tRNA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8aa57968",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>...</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>YBR298C</td>\n",
       "      <td>MAL31,MAL3T,MALT,maltose%20permease</td>\n",
       "      <td>Maltose%20permease%3B%20high-affinity%20maltos...</td>\n",
       "      <td>GO:0000017,GO:0005352,GO:0071944,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Maltose%20permease</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL31</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>YBR298C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function%3...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>YBR299W</td>\n",
       "      <td>MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32</td>\n",
       "      <td>Maltase%20%28alpha-D-glucosidase%29%3B%20induc...</td>\n",
       "      <td>GO:0000025,GO:0004575,GO:0005886,GO:0005987,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Maltase%20%28alpha-D-glucosidase%29</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL32</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious%20open%20reading%20frame%3B%20unlikely...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious%20open%20reading%20frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>YGR294W</td>\n",
       "      <td>PAU12,seripauperin%20PAU12</td>\n",
       "      <td>Protein%20of%20unknown%20function%3B%20member%...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>PAU12</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seq_id  start    end       Name  \\\n",
       "0   chrII-2_tig00004927   8297  10119    YBR298C   \n",
       "3   chrII-2_tig00004927  10675  10894  YBR298C-A   \n",
       "6   chrII-2_tig00004927  10985  12717    YBR299W   \n",
       "9   chrII-2_tig00004927  14242  14700    YGR293C   \n",
       "12  chrII-2_tig00004927  14660  15022    YGR294W   \n",
       "\n",
       "                                         Alias  \\\n",
       "0          MAL31,MAL3T,MALT,maltose%20permease   \n",
       "3                                         None   \n",
       "6   MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32   \n",
       "9                                         None   \n",
       "12                  PAU12,seripauperin%20PAU12   \n",
       "\n",
       "                                                 Note  \\\n",
       "0   Maltose%20permease%3B%20high-affinity%20maltos...   \n",
       "3   Putative%20protein%20of%20unknown%20function%3...   \n",
       "6   Maltase%20%28alpha-D-glucosidase%29%3B%20induc...   \n",
       "9   Dubious%20open%20reading%20frame%3B%20unlikely...   \n",
       "12  Protein%20of%20unknown%20function%3B%20member%...   \n",
       "\n",
       "                                        Ontology_term score strand phase  ...  \\\n",
       "0         GO:0000017,GO:0005352,GO:0071944,SO:0000704     .      -     .  ...   \n",
       "3         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .  ...   \n",
       "6   GO:0000025,GO:0004575,GO:0005886,GO:0005987,GO...     .      +     .  ...   \n",
       "9         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .  ...   \n",
       "12        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .  ...   \n",
       "\n",
       "                                         display extra_copy_number   gene  \\\n",
       "0                             Maltose%20permease                 1  MAL31   \n",
       "3   Putative%20protein%20of%20unknown%20function                 1   None   \n",
       "6            Maltase%20%28alpha-D-glucosidase%29                 1  MAL32   \n",
       "9               Dubious%20open%20reading%20frame                 2   None   \n",
       "12             Protein%20of%20unknown%20function                 1  PAU12   \n",
       "\n",
       "   low_identity orf_classification partial_mapping sequence_ID valid_ORFs  \\\n",
       "0          None           Verified            None       0.946          0   \n",
       "3          None    Uncharacterized            None       0.981          0   \n",
       "6          None           Verified            None       0.981          0   \n",
       "9          None            Dubious            None       0.989          0   \n",
       "12         None           Verified            None       0.983          1   \n",
       "\n",
       "   Chromosome       Contig  \n",
       "0           2  tig00004927  \n",
       "3           2  tig00004927  \n",
       "6           2  tig00004927  \n",
       "9           2  tig00004927  \n",
       "12          2  tig00004927  \n",
       "\n",
       "[5 rows x 23 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "88d25b7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean20(DF):\n",
    "    note = DF['Note']\n",
    "    dis = DF['display']\n",
    "    clean = []\n",
    "    display = []\n",
    "    for n in note:\n",
    "        clean.append(n.replace('%20', ' '))\n",
    "    for d in dis:\n",
    "        display.append(str(d).replace('%20', ' '))\n",
    "    DF['Note'] = clean\n",
    "    DF['display'] = display\n",
    "    return DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "50940e10",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\741321415.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Note'] = clean\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\741321415.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['display'] = display\n"
     ]
    }
   ],
   "source": [
    "gene = clean20(gene)\n",
    "TE = clean20(TE)\n",
    "sno = clean20(sno)\n",
    "tRNA = clean20(tRNA)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "259832d6",
   "metadata": {},
   "source": [
    "## Df by attribute for S288C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "685b570a",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_df_s288c = annot_s288c.filter_feature_of_type(['transposable_element_gene'])\n",
    "sno_df_s288c = annot_s288c.filter_feature_of_type(['snoRNA_gene'])\n",
    "tRNA_df_s288c = annot_s288c.filter_feature_of_type(['tRNA_gene'])\n",
    "gene_df_s288c = annot_s288c.filter_feature_of_type(['gene'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "98713fc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_attr_s288c = gene_df_s288c.attributes_to_columns()\n",
    "TE_attr_s288c = TE_df_s288c.attributes_to_columns()\n",
    "sno_attr_s288c = sno_df_s288c.attributes_to_columns()\n",
    "tRNA_attr_s288c = tRNA_df_s288c.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "4dc8b278",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'Ontology_term', 'curie',\n",
       "       'dbxref', 'display'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE_attr_s288c.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a65bdbc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_s288c = gene_attr_s288c[['seq_id', 'start', 'end', 'Name', 'gene',  'Alias',  'Note', 'Ontology_term', 'strand', 'phase', 'curie',\n",
    "        'orf_classification', 'display']]\n",
    "\n",
    "TE_s288c = TE_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "sno_s288c = sno_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "tRNA_s288c = tRNA_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "d230909b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def lat_to_arabic_S288(df):\n",
    "    # This code just convert the chromosome numbering to integer and remove mitochondrial genes.\n",
    "    Latin_to_arabic = {'I': 1, 'II':2, 'III':3, 'IV':4, 'V':5, 'VI':6, 'VII':7, 'VIII':8, 'IX':9, 'X':10,\n",
    "                      'XI':11, 'XII':12, 'XIII':13, 'XIV':14, 'XV':15, 'XVI':16, 'Mito': 'mitoch'}\n",
    "    chLat = df['seq_id']\n",
    "    ch = []\n",
    "    for chrom in chLat:\n",
    "        chrom = chrom.replace('chr', '')\n",
    "        ch.append(Latin_to_arabic[chrom])\n",
    "    df['Chromosome'] = ch\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "4ef36a0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n"
     ]
    }
   ],
   "source": [
    "gene_s288c = lat_to_arabic_S288(gene_s288c)\n",
    "TE_s288c = lat_to_arabic_S288(TE_s288c)\n",
    "sno_s288c = lat_to_arabic_S288(sno_s288c)\n",
    "tRNA_s288c = lat_to_arabic_S288(tRNA_s288c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "ea5b328c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\741321415.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Note'] = clean\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_12224\\741321415.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['display'] = display\n"
     ]
    }
   ],
   "source": [
    "gene_s288c = clean20(gene_s288c)\n",
    "TE_s288c = clean20(TE_s288c)\n",
    "sno_s288c = clean20(sno_s288c)\n",
    "tRNA_s288c = clean20(tRNA_s288c)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d2920e19",
   "metadata": {},
   "source": [
    "# TE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "cd662573",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE.to_excel('TE_liftoff4_yps1009.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "afedef40",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4     17\n",
       "7     10\n",
       "16     9\n",
       "12     8\n",
       "13     8\n",
       "15     8\n",
       "2      6\n",
       "10     6\n",
       "5      4\n",
       "8      4\n",
       "14     4\n",
       "1      2\n",
       "3      2\n",
       "6      2\n",
       "9      1\n",
       "Name: Chromosome, dtype: int64"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE_s288c['Chromosome'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "63004716",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7     54\n",
       "12    14\n",
       "4     12\n",
       "5      8\n",
       "2      2\n",
       "6      2\n",
       "15     2\n",
       "3      1\n",
       "1      1\n",
       "8      1\n",
       "16     1\n",
       "10     1\n",
       "Name: Chromosome, dtype: int64"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE['Chromosome'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "54f1cf55",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    61\n",
       "0    38\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "19aaa023",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_count = TE['Chromosome'].value_counts().to_frame()\n",
    "TE_count = TE_count.rename(columns = {'Chromosome': 'yps1009 TE'})\n",
    "TE_count_s = TE_s288c['Chromosome'].value_counts().to_frame()\n",
    "TE_count_s = TE_count_s.rename(columns = {'Chromosome': 'S288C TE'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "6db5428c",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_count = TE_count_s.join(TE_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "eb2b5881",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C TE</th>\n",
       "      <th>yps1009 TE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>17</td>\n",
       "      <td>12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>10</td>\n",
       "      <td>54.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>8</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>8</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>8</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>6</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>4</td>\n",
       "      <td>8.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>4</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>4</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C TE  yps1009 TE\n",
       "4         17        12.0\n",
       "7         10        54.0\n",
       "16         9         1.0\n",
       "12         8        14.0\n",
       "13         8         NaN\n",
       "15         8         2.0\n",
       "2          6         2.0\n",
       "10         6         1.0\n",
       "5          4         8.0\n",
       "8          4         1.0\n",
       "14         4         NaN\n",
       "1          2         1.0\n",
       "3          2         1.0\n",
       "6          2         2.0\n",
       "9          1         NaN"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "e051b7e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE7_yps = list(TE[TE['Chromosome'] == 7]['Name'])\n",
    "TE7_s288 = list(TE_s288c[TE_s288c['Chromosome'] == 7]['Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c7d0810a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'YGR027W-A', 'YGR027W-B', 'YGR038C-A', 'YGR038C-B', 'YGR161C-C'}"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set(TE7_s288) & set(TE7_yps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "548569f0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    61\n",
       "0    38\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59ce1fe0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "1ec45b2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>curie</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5939</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177837</td>\n",
       "      <td>179152</td>\n",
       "      <td>YDR034C-C</td>\n",
       "      <td>gag%20protein</td>\n",
       "      <td>Retrotransposon TYA Gag gene co-transcribed wi...</td>\n",
       "      <td>GO:0000943,GO:0003723,GO:0032197,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDR034C-C_4</td>\n",
       "      <td>0.999</td>\n",
       "      <td>SGD:S000007344</td>\n",
       "      <td>Retrotransposon TYA Gag gene co-transcribed wi...</td>\n",
       "      <td>4</td>\n",
       "      <td>0.973</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                seq_id   start     end       Name          Alias  \\\n",
       "5939  chrI_tig00000033  177837  179152  YDR034C-C  gag%20protein   \n",
       "\n",
       "                                                   Note  \\\n",
       "5939  Retrotransposon TYA Gag gene co-transcribed wi...   \n",
       "\n",
       "                                    Ontology_term score strand phase  \\\n",
       "5939  GO:0000943,GO:0003723,GO:0032197,SO:0000704     .      +     .   \n",
       "\n",
       "      copy_num_ID coverage           curie  \\\n",
       "5939  YDR034C-C_4    0.999  SGD:S000007344   \n",
       "\n",
       "                                                display extra_copy_number  \\\n",
       "5939  Retrotransposon TYA Gag gene co-transcribed wi...                 4   \n",
       "\n",
       "     sequence_ID valid_ORFs  Chromosome       Contig  \n",
       "5939       0.973          0           1  tig00000033  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TE[TE['Chromosome'] == 1]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec032b45",
   "metadata": {},
   "source": [
    "# snoRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "7726d381",
   "metadata": {},
   "outputs": [],
   "source": [
    "sno_count = sno['Chromosome'].value_counts().to_frame()\n",
    "sno_count = sno_count.rename(columns = {'Chromosome': 'yps1009 snoRNA'})\n",
    "sno_count_s = sno_s288c['Chromosome'].value_counts().to_frame()\n",
    "sno_count_s = sno_count_s.rename(columns = {'Chromosome': 'S288C snoRNA'})\n",
    "sno_count = sno_count_s.join(sno_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "0d0ff52b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C snoRNA</th>\n",
       "      <th>yps1009 snoRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C snoRNA  yps1009 snoRNA\n",
       "13            13              13\n",
       "15            11              11\n",
       "12             7               7\n",
       "7              6               6\n",
       "16             6               6\n",
       "5              5               5\n",
       "10             5               5\n",
       "11             5               5\n",
       "14             5               5\n",
       "3              4               4\n",
       "4              4               4\n",
       "2              2               2\n",
       "8              2               2\n",
       "1              1               1\n",
       "9              1               1"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sno_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530e98aa",
   "metadata": {},
   "source": [
    "# tRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "4962e83e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    299\n",
       "1     22\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "bbef7664",
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_count = tRNA['Chromosome'].value_counts().to_frame()\n",
    "tRNA_count = tRNA_count.rename(columns = {'Chromosome': 'yps1009 tRNARNA'})\n",
    "tRNA_count_s = tRNA_s288c['Chromosome'].value_counts().to_frame()\n",
    "tRNA_count_s = tRNA_count_s.rename(columns = {'Chromosome': 'S288C tRNARNA'})\n",
    "tRNA_count = tRNA_count_s.join(tRNA_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "bda5f137",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C tRNARNA</th>\n",
       "      <th>yps1009 tRNARNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>36</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>28</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mitoch</th>\n",
       "      <td>24</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>24</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>21</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>20</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>11</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        S288C tRNARNA  yps1009 tRNARNA\n",
       "7                  36               38\n",
       "4                  28               30\n",
       "mitoch             24               26\n",
       "10                 24               27\n",
       "12                 21               22\n",
       "13                 21               22\n",
       "5                  20               21\n",
       "15                 20               20\n",
       "16                 17               18\n",
       "11                 16               18\n",
       "14                 14               14\n",
       "2                  13               13\n",
       "8                  11               13\n",
       "9                  10               11\n",
       "6                  10               11\n",
       "3                  10               13\n",
       "1                   4                4"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8effbe52",
   "metadata": {},
   "source": [
    "# GENE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "c117842e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Verified           5249\n",
       "Uncharacterized     737\n",
       "Dubious             719\n",
       "Name: orf_classification, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene['orf_classification'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "6b91315f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    524\n",
       "0    195\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dubious = gene[gene['orf_classification'] == 'Dubious']\n",
    "dubious['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "fd97eac3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    564\n",
       "0    173\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uncharacterized = gene[gene['orf_classification'] == 'Uncharacterized']\n",
    "uncharacterized['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8b062b2",
   "metadata": {},
   "source": [
    "# Verified ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "faddeacc",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified = gene[gene['orf_classification'] == 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "7b49021f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    5087\n",
       "0     162\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "bd6300ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_s288c = gene_s288c[gene_s288c['orf_classification'] == 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "d019b4f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5216, 14)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_s288c.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "6a766625",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid = verified[verified['valid_ORFs'] == '1']\n",
    "verified_unvalid = verified[verified['valid_ORFs'] == '0']\n",
    "\n",
    "verified_valid_count = verified_valid['Chromosome'].value_counts().to_frame()\n",
    "verified_valid_count = verified_valid_count.rename(columns={'Chromosome': 'Number valid genes'})\n",
    "verified_unvalid_count = verified_unvalid['Chromosome'].value_counts().to_frame()\n",
    "verified_unvalid_count = verified_unvalid_count.rename(columns={'Chromosome': 'Number unvalid genes'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "32c1a04f",
   "metadata": {},
   "outputs": [],
   "source": [
    "unval_dic = {}\n",
    "for ind, row in verified_unvalid.iterrows():\n",
    "    if row['Chromosome'] in unval_dic:\n",
    "        unval_dic[row['Chromosome']] = unval_dic[row['Chromosome']] + [row['Name']]\n",
    "    else:\n",
    "        unval_dic[row['Chromosome']] = [row['Name']]    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "63ab8ca2",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum = verified_valid_count.join(verified_unvalid_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "52c99402",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = []\n",
    "for ind in verified_sum.index:\n",
    "    tmp.append(unval_dic[ind])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "ec34117a",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum['unvalid genes'] = tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "8942bc2d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>...</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>YBR298C</td>\n",
       "      <td>MAL31,MAL3T,MALT,maltose%20permease</td>\n",
       "      <td>Maltose permease%3B high-affinity maltose tran...</td>\n",
       "      <td>GO:0000017,GO:0005352,GO:0071944,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Maltose permease</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL31</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>YBR299W</td>\n",
       "      <td>MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32</td>\n",
       "      <td>Maltase %28alpha-D-glucosidase%29%3B inducible...</td>\n",
       "      <td>GO:0000025,GO:0004575,GO:0005886,GO:0005987,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Maltase %28alpha-D-glucosidase%29</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL32</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                seq_id  start    end     Name  \\\n",
       "0  chrII-2_tig00004927   8297  10119  YBR298C   \n",
       "6  chrII-2_tig00004927  10985  12717  YBR299W   \n",
       "\n",
       "                                        Alias  \\\n",
       "0         MAL31,MAL3T,MALT,maltose%20permease   \n",
       "6  MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32   \n",
       "\n",
       "                                                Note  \\\n",
       "0  Maltose permease%3B high-affinity maltose tran...   \n",
       "6  Maltase %28alpha-D-glucosidase%29%3B inducible...   \n",
       "\n",
       "                                       Ontology_term score strand phase  ...  \\\n",
       "0        GO:0000017,GO:0005352,GO:0071944,SO:0000704     .      -     .  ...   \n",
       "6  GO:0000025,GO:0004575,GO:0005886,GO:0005987,GO...     .      +     .  ...   \n",
       "\n",
       "                             display extra_copy_number   gene low_identity  \\\n",
       "0                   Maltose permease                 1  MAL31         None   \n",
       "6  Maltase %28alpha-D-glucosidase%29                 1  MAL32         None   \n",
       "\n",
       "  orf_classification partial_mapping sequence_ID valid_ORFs Chromosome  \\\n",
       "0           Verified            None       0.946          0          2   \n",
       "6           Verified            None       0.981          0          2   \n",
       "\n",
       "        Contig  \n",
       "0  tig00004927  \n",
       "6  tig00004927  \n",
       "\n",
       "[2 rows x 23 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "e7ad4015",
   "metadata": {},
   "outputs": [],
   "source": [
    "missed = []\n",
    "moved = []\n",
    "count_miss = []\n",
    "count_move = []\n",
    "for Ch in verified_sum.index:\n",
    "    s288 = verified_s288c[verified_s288c['Chromosome'] == Ch]\n",
    "    yps = verified[verified['Chromosome'] == Ch]\n",
    "    tmp = list(set(s288['Name']) - set(yps['Name']))\n",
    "    miss = []\n",
    "    mov = []\n",
    "    others = verified[verified['Chromosome'] != Ch]\n",
    "    for g in tmp:\n",
    "        search = others[others['Name'] == g]\n",
    "        if search.shape[0] >= 1:\n",
    "            mov.append((g, list(search['Chromosome']), list(search['valid_ORFs'])))\n",
    "        else:\n",
    "            miss.append(g)  \n",
    "    moved.append(mov)\n",
    "    missed.append(miss)\n",
    "    count_miss.append(len(miss))\n",
    "    count_move.append(len(mov))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "0e731ec9",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum['Number of missing genes'] = count_miss\n",
    "verified_sum['Missing genes'] = missed\n",
    "verified_sum['Number of moved genes'] = count_move\n",
    "verified_sum['Moved genes, new chromosome(s), orf validity'] = moved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "385874ef",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Number valid genes</th>\n",
       "      <th>Number unvalid genes</th>\n",
       "      <th>unvalid genes</th>\n",
       "      <th>Number of missing genes</th>\n",
       "      <th>Missing genes</th>\n",
       "      <th>Number of moved genes</th>\n",
       "      <th>Moved genes, new chromosome(s), orf validity</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>673</td>\n",
       "      <td>9</td>\n",
       "      <td>[YDL247W, YNL331C, YDL227C, YDL080C, YDL039C, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>[(YDR385W, [15], [1]), (YDL248W, [13], [0]), (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>467</td>\n",
       "      <td>14</td>\n",
       "      <td>[YOR388C, YOR249C, YOR244W, YOR245C, YCR106W, ...</td>\n",
       "      <td>3</td>\n",
       "      <td>[YOR246C, YOR255W, YOR247W]</td>\n",
       "      <td>3</td>\n",
       "      <td>[(YOR394W, [16], [0]), (YOR390W, [16], [1]), (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>457</td>\n",
       "      <td>5</td>\n",
       "      <td>[YGL243W, YGL131C, YAR027W, YGR140W, YBR298C]</td>\n",
       "      <td>4</td>\n",
       "      <td>[YGL133W, YGL137W, YGL134W, YGL136C]</td>\n",
       "      <td>6</td>\n",
       "      <td>[(YGL263W, [12, 12], [0, 1]), (YGR296W, [5], [...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>432</td>\n",
       "      <td>13</td>\n",
       "      <td>[YGL263W, YLL063C, YLL057C, YLL052C, YLR119W, ...</td>\n",
       "      <td>5</td>\n",
       "      <td>[YLR160C, YLR157C, YLR158C, YLR154W-C, YLR155C]</td>\n",
       "      <td>3</td>\n",
       "      <td>[(YLR467W, [5], [1]), (YLR466W, [5], [0]), (YL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>405</td>\n",
       "      <td>8</td>\n",
       "      <td>[YOR394W, YPL282C, YMR323W, YPL274W, YPL255W, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>[(YPL281C, [15], [1]), (YPL279C, [15], [1]), (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>394</td>\n",
       "      <td>8</td>\n",
       "      <td>[YDL248W, YJR161C, YML118W, YMR019W, YMR104C, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>[(YMR105C, [11], [1]), (YMR325W, [15, 16], [1,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>360</td>\n",
       "      <td>7</td>\n",
       "      <td>[YBR298C, YBR299W, YGR295C, YBR031W, YBR073W, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>3</td>\n",
       "      <td>[(YBL111C, [5], [0]), (YBL108C-A, [9, 12, 12],...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>326</td>\n",
       "      <td>18</td>\n",
       "      <td>[YNL322C, YNL318C, YNL317W, YNL316C, YNL315C, ...</td>\n",
       "      <td>7</td>\n",
       "      <td>[YNL329C, YNL328C, YNL323W, YNL325C, YNL330C, ...</td>\n",
       "      <td>8</td>\n",
       "      <td>[(YNL334C, [10], [1]), (YNL333W, [10], [1]), (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>316</td>\n",
       "      <td>12</td>\n",
       "      <td>[YJL222W, YJL219W, YJL222W, YJL217W, YJL077C, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>[(YJR159W, [4], [1]), (YJR161C, [13], [0]), (Y...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>291</td>\n",
       "      <td>12</td>\n",
       "      <td>[YKR102W, YKR103W, YKR104W, YKR105C, YKL161C, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>[(YKL224C, [9], [1]), (YKR106W, [3], [1])]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>249</td>\n",
       "      <td>5</td>\n",
       "      <td>[YLR466W, YBL111C, YEL069C, YER106W, YER170W]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>232</td>\n",
       "      <td>5</td>\n",
       "      <td>[YHL043W, YHR032W, YHR175W, YHR185C, YAR071W]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>4</td>\n",
       "      <td>[(YHL048W, [9], [0]), (YHR216W, [1], [1]), (YH...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>178</td>\n",
       "      <td>10</td>\n",
       "      <td>[YIL173W, YJL222W, YIL172C, YIL169C, YIL164C, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>130</td>\n",
       "      <td>6</td>\n",
       "      <td>[YCL073C, YKR105C, YCL067C, YCR039C, YCR073C, ...</td>\n",
       "      <td>1</td>\n",
       "      <td>[YCR105W]</td>\n",
       "      <td>4</td>\n",
       "      <td>[(YCR107W, [6], [0]), (YCR106W, [15], [0]), (Y...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>97</td>\n",
       "      <td>5</td>\n",
       "      <td>[YFR055W, YCR107W, YFR012W, YFR026C, YFR055W]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>5</td>\n",
       "      <td>[(YFL058W, [4, 10], [1, 1]), (YFL060C, [10], [...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>76</td>\n",
       "      <td>6</td>\n",
       "      <td>[YAL063C, YAL059W, YAL032C, YAR020C, YAR031W, ...</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "      <td>2</td>\n",
       "      <td>[(YAL068C, [7, 10], [1, 1]), (YAR027W, [7], [0])]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mitoch</th>\n",
       "      <td>4</td>\n",
       "      <td>19</td>\n",
       "      <td>[Q0105, Q0110, Q0115, Q0120, Q0140, Q0250, Q02...</td>\n",
       "      <td>1</td>\n",
       "      <td>[Q0160]</td>\n",
       "      <td>0</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Number valid genes  Number unvalid genes  \\\n",
       "4                      673                     9   \n",
       "15                     467                    14   \n",
       "7                      457                     5   \n",
       "12                     432                    13   \n",
       "16                     405                     8   \n",
       "13                     394                     8   \n",
       "2                      360                     7   \n",
       "14                     326                    18   \n",
       "10                     316                    12   \n",
       "11                     291                    12   \n",
       "5                      249                     5   \n",
       "8                      232                     5   \n",
       "9                      178                    10   \n",
       "3                      130                     6   \n",
       "6                       97                     5   \n",
       "1                       76                     6   \n",
       "mitoch                   4                    19   \n",
       "\n",
       "                                            unvalid genes  \\\n",
       "4       [YDL247W, YNL331C, YDL227C, YDL080C, YDL039C, ...   \n",
       "15      [YOR388C, YOR249C, YOR244W, YOR245C, YCR106W, ...   \n",
       "7           [YGL243W, YGL131C, YAR027W, YGR140W, YBR298C]   \n",
       "12      [YGL263W, YLL063C, YLL057C, YLL052C, YLR119W, ...   \n",
       "16      [YOR394W, YPL282C, YMR323W, YPL274W, YPL255W, ...   \n",
       "13      [YDL248W, YJR161C, YML118W, YMR019W, YMR104C, ...   \n",
       "2       [YBR298C, YBR299W, YGR295C, YBR031W, YBR073W, ...   \n",
       "14      [YNL322C, YNL318C, YNL317W, YNL316C, YNL315C, ...   \n",
       "10      [YJL222W, YJL219W, YJL222W, YJL217W, YJL077C, ...   \n",
       "11      [YKR102W, YKR103W, YKR104W, YKR105C, YKL161C, ...   \n",
       "5           [YLR466W, YBL111C, YEL069C, YER106W, YER170W]   \n",
       "8           [YHL043W, YHR032W, YHR175W, YHR185C, YAR071W]   \n",
       "9       [YIL173W, YJL222W, YIL172C, YIL169C, YIL164C, ...   \n",
       "3       [YCL073C, YKR105C, YCL067C, YCR039C, YCR073C, ...   \n",
       "6           [YFR055W, YCR107W, YFR012W, YFR026C, YFR055W]   \n",
       "1       [YAL063C, YAL059W, YAL032C, YAR020C, YAR031W, ...   \n",
       "mitoch  [Q0105, Q0110, Q0115, Q0120, Q0140, Q0250, Q02...   \n",
       "\n",
       "        Number of missing genes  \\\n",
       "4                             0   \n",
       "15                            3   \n",
       "7                             4   \n",
       "12                            5   \n",
       "16                            0   \n",
       "13                            0   \n",
       "2                             0   \n",
       "14                            7   \n",
       "10                            0   \n",
       "11                            0   \n",
       "5                             0   \n",
       "8                             0   \n",
       "9                             0   \n",
       "3                             1   \n",
       "6                             0   \n",
       "1                             0   \n",
       "mitoch                        1   \n",
       "\n",
       "                                            Missing genes  \\\n",
       "4                                                      []   \n",
       "15                            [YOR246C, YOR255W, YOR247W]   \n",
       "7                    [YGL133W, YGL137W, YGL134W, YGL136C]   \n",
       "12        [YLR160C, YLR157C, YLR158C, YLR154W-C, YLR155C]   \n",
       "16                                                     []   \n",
       "13                                                     []   \n",
       "2                                                      []   \n",
       "14      [YNL329C, YNL328C, YNL323W, YNL325C, YNL330C, ...   \n",
       "10                                                     []   \n",
       "11                                                     []   \n",
       "5                                                      []   \n",
       "8                                                      []   \n",
       "9                                                      []   \n",
       "3                                               [YCR105W]   \n",
       "6                                                      []   \n",
       "1                                                      []   \n",
       "mitoch                                            [Q0160]   \n",
       "\n",
       "        Number of moved genes  \\\n",
       "4                           5   \n",
       "15                          3   \n",
       "7                           6   \n",
       "12                          3   \n",
       "16                          5   \n",
       "13                          5   \n",
       "2                           3   \n",
       "14                          8   \n",
       "10                          5   \n",
       "11                          2   \n",
       "5                           0   \n",
       "8                           4   \n",
       "9                           0   \n",
       "3                           4   \n",
       "6                           5   \n",
       "1                           2   \n",
       "mitoch                      0   \n",
       "\n",
       "             Moved genes, new chromosome(s), orf validity  \n",
       "4       [(YDR385W, [15], [1]), (YDL248W, [13], [0]), (...  \n",
       "15      [(YOR394W, [16], [0]), (YOR390W, [16], [1]), (...  \n",
       "7       [(YGL263W, [12, 12], [0, 1]), (YGR296W, [5], [...  \n",
       "12      [(YLR467W, [5], [1]), (YLR466W, [5], [0]), (YL...  \n",
       "16      [(YPL281C, [15], [1]), (YPL279C, [15], [1]), (...  \n",
       "13      [(YMR105C, [11], [1]), (YMR325W, [15, 16], [1,...  \n",
       "2       [(YBL111C, [5], [0]), (YBL108C-A, [9, 12, 12],...  \n",
       "14      [(YNL334C, [10], [1]), (YNL333W, [10], [1]), (...  \n",
       "10      [(YJR159W, [4], [1]), (YJR161C, [13], [0]), (Y...  \n",
       "11             [(YKL224C, [9], [1]), (YKR106W, [3], [1])]  \n",
       "5                                                      []  \n",
       "8       [(YHL048W, [9], [0]), (YHR216W, [1], [1]), (YH...  \n",
       "9                                                      []  \n",
       "3       [(YCR107W, [6], [0]), (YCR106W, [15], [0]), (Y...  \n",
       "6       [(YFL058W, [4, 10], [1, 1]), (YFL060C, [10], [...  \n",
       "1       [(YAL068C, [7, 10], [1, 1]), (YAR027W, [7], [0])]  \n",
       "mitoch                                                 []  "
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_sum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "5a783f74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Number valid genes                                                                           5087\n",
       "Number unvalid genes                                                                          162\n",
       "unvalid genes                                   [YDL247W, YNL331C, YDL227C, YDL080C, YDL039C, ...\n",
       "Number of missing genes                                                                        21\n",
       "Missing genes                                   [YOR246C, YOR255W, YOR247W, YGL133W, YGL137W, ...\n",
       "Number of moved genes                                                                          60\n",
       "Moved genes, new chromosome(s), orf validity    [(YDR385W, [15], [1]), (YDL248W, [13], [0]), (...\n",
       "dtype: object"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_sum.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "9ad232b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum.to_excel('verified_genes_summary.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97e390b0",
   "metadata": {},
   "source": [
    "# Missing genes dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "41ed1b4d",
   "metadata": {},
   "outputs": [],
   "source": [
    "flat_list = [item for sublist in missed for item in sublist]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "26acf4b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_genes = verified_s288c[verified_s288c['Name'].isin(flat_list)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "144dc49e",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_genes.to_excel('missing_verified_genes.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b43324b",
   "metadata": {},
   "source": [
    "# Blast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "3fd80818",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import Biopython tools for running local BLASTX\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "92b36ee0",
   "metadata": {},
   "outputs": [],
   "source": [
    "ids = []\n",
    "seq = []\n",
    "for seq_record in SeqIO.parse(\"fasta_file/orf_coding_all_R64-3-1_20210421.fasta\", \"fasta\"):\n",
    "    ids.append(str(seq_record.id))\n",
    "    seq.append(str(seq_record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "29340814",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n",
      "STDOUT: \n",
      "STDERR: \n"
     ]
    }
   ],
   "source": [
    "for g in flat_list:\n",
    "    name = g + '.txt'\n",
    "    with open(name , \"w\") as file:\n",
    "        file.write('> ' + g + '\\n')\n",
    "        file.write(seq[ids.index(g)])\n",
    "    outfile = 'Blast_results/' + g + '.tab'\n",
    "    cmd_blastn = NcbiblastnCommandline(query= name, out= outfile, evalue=0.001, db= 'YPS1009_DB/YPS1009_DB')\n",
    "    stdout, stderr = cmd_blastn()\n",
    "    print(\"STDOUT: %s\" % stdout)\n",
    "    print(\"STDERR: %s\" % stderr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa42fcbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "cmd_blastn = NcbiblastnCommandline(query= 'fasta_file/S288C_YDR038C_ENA5_genomic.fsa', out='Blast_results/YDR038C.tab', evalue=0.001, db= 'YPS1009_DB/YPS1009_DB')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82a58ff0",
   "metadata": {},
   "outputs": [],
   "source": [
    "cmd_blastn = NcbiblastnCommandline(query= 'fasta_file/S288C_YDR039C_ENA2_genomic.fsa', out='Blast_results/YDR039C.tab', evalue=0.001, db= 'YPS1009_DB/YPS1009_DB')\n",
    "stdout, stderr = cmd_blastn()\n",
    "# Check STDOUT, STDERR\n",
    "print(\"STDOUT: %s\" % stdout)\n",
    "print(\"STDERR: %s\" % stderr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c6a1c7b",
   "metadata": {},
   "source": [
    "# Duplicated genes\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "c130b50c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    6563\n",
       "1      94\n",
       "2      22\n",
       "3      12\n",
       "4       7\n",
       "5       5\n",
       "6       2\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Many duplicated genes in this version\n",
    "gene['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "bda8b613",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>...</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>17994</td>\n",
       "      <td>18290</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.943</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>767</td>\n",
       "      <td>1021</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>1323</td>\n",
       "      <td>2063</td>\n",
       "      <td>YCL076W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>6426</td>\n",
       "      <td>8271</td>\n",
       "      <td>YCL073C</td>\n",
       "      <td>GEX1,glutathione%20exchanger</td>\n",
       "      <td>Proton:glutathione antiporter%3B localized to ...</td>\n",
       "      <td>GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Proton:glutathione antiporter</td>\n",
       "      <td>2</td>\n",
       "      <td>GEX1</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.990</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>760</td>\n",
       "      <td>1014</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>1316</td>\n",
       "      <td>2055</td>\n",
       "      <td>YCL076W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>310848</td>\n",
       "      <td>311102</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2097</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>789332</td>\n",
       "      <td>789467</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.890</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00000010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2106</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1219</td>\n",
       "      <td>1356</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4835</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>2653</td>\n",
       "      <td>2790</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4838</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>2759</td>\n",
       "      <td>2986</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.947</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4841</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>3329</td>\n",
       "      <td>3525</td>\n",
       "      <td>YCR102W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5644</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>676</td>\n",
       "      <td>975</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6054</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>238815</td>\n",
       "      <td>239136</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.867</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6057</th>\n",
       "      <td>chrMANY_tig00000038</td>\n",
       "      <td>737</td>\n",
       "      <td>1036</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6063</th>\n",
       "      <td>chrMANY_tig00000038</td>\n",
       "      <td>2728</td>\n",
       "      <td>2958</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6917</th>\n",
       "      <td>chrVI-2_tig00004937</td>\n",
       "      <td>12581</td>\n",
       "      <td>12879</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.943</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>tig00004937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6920</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>680</td>\n",
       "      <td>934</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7906</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>522253</td>\n",
       "      <td>522633</td>\n",
       "      <td>YAL064W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Fungal-specific protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005783,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Fungal-specific protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.934</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9422</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>805604</td>\n",
       "      <td>806749</td>\n",
       "      <td>YGR295C</td>\n",
       "      <td>COS6</td>\n",
       "      <td>Endosomal protein involved in turnover of plas...</td>\n",
       "      <td>GO:0000324,GO:0000324,GO:0003674,GO:0005768,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Endosomal protein involved in turnover of plas...</td>\n",
       "      <td>2</td>\n",
       "      <td>COS6</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.961</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9425</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>807356</td>\n",
       "      <td>807492</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9428</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>807660</td>\n",
       "      <td>807956</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.93</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9888</th>\n",
       "      <td>chrVI_tig00004938</td>\n",
       "      <td>298356</td>\n",
       "      <td>298655</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>tig00004938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11170</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>642</td>\n",
       "      <td>835</td>\n",
       "      <td>YCR108C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11173</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>2619</td>\n",
       "      <td>2756</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11182</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>4754</td>\n",
       "      <td>5859</td>\n",
       "      <td>YCR102C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative quinone oxidoreductase%3B overexpress...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative quinone oxidoreductase</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11185</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>7160</td>\n",
       "      <td>11768</td>\n",
       "      <td>YJL222W</td>\n",
       "      <td>VTH2,signal%20sequence-binding%20protein</td>\n",
       "      <td>Putative membrane glycoprotein%3B has strong s...</td>\n",
       "      <td>GO:0000324,GO:0005048,GO:0005048,GO:0006623,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative membrane glycoprotein</td>\n",
       "      <td>2</td>\n",
       "      <td>VTH2</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.972</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11188</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>12417</td>\n",
       "      <td>14186</td>\n",
       "      <td>YIL172C</td>\n",
       "      <td>IMA3,oligo-1%2C6-glucosidase%20IMA3</td>\n",
       "      <td>Alpha-glucosidase%3B weak%2C but broad substra...</td>\n",
       "      <td>GO:0004574,GO:0004574,GO:0004575,GO:0005737,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Alpha-glucosidase</td>\n",
       "      <td>2</td>\n",
       "      <td>IMA3</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11221</th>\n",
       "      <td>chrXI-2_tig00004931</td>\n",
       "      <td>28713</td>\n",
       "      <td>28967</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11224</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>699</td>\n",
       "      <td>999</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.926</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12027</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>6551</td>\n",
       "      <td>6679</td>\n",
       "      <td>YBL108C-A</td>\n",
       "      <td>PAU9,seripauperin%20PAU9</td>\n",
       "      <td>Protein of unknown function%3B member of the s...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>PAU9</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12030</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>7121</td>\n",
       "      <td>7351</td>\n",
       "      <td>YGL260W</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B transc...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.948</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12060</th>\n",
       "      <td>chrXIII-2_tig00000029</td>\n",
       "      <td>1228</td>\n",
       "      <td>1363</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.948</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13703</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>467347</td>\n",
       "      <td>467484</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13709</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>467923</td>\n",
       "      <td>468106</td>\n",
       "      <td>YOR394C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%3B identified by g...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.822</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14775</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>554706</td>\n",
       "      <td>554903</td>\n",
       "      <td>YCR102W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.954</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00000020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14790</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>557160</td>\n",
       "      <td>557459</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00000020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16142</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>728468</td>\n",
       "      <td>728695</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.929</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>tig00004929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16145</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>728662</td>\n",
       "      <td>728801</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.921</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>tig00004929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16158</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>726</td>\n",
       "      <td>980</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17809</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>743</td>\n",
       "      <td>997</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.941</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19446</th>\n",
       "      <td>chrXVI_tig00000013</td>\n",
       "      <td>719024</td>\n",
       "      <td>719337</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.895</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19500</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>24888</td>\n",
       "      <td>25351</td>\n",
       "      <td>YIL171W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20876</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>2582</td>\n",
       "      <td>2719</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20879</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>2688</td>\n",
       "      <td>2915</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.939</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20885</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>4721</td>\n",
       "      <td>5826</td>\n",
       "      <td>YCR102C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative quinone oxidoreductase%3B overexpress...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative quinone oxidoreductase</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20897</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>13824</td>\n",
       "      <td>14287</td>\n",
       "      <td>YIL171W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.952</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      seq_id   start     end       Name  \\\n",
       "9        chrII-2_tig00004927   14242   14700    YGR293C   \n",
       "21       chrII-2_tig00004927   17994   18290  YER188C-A   \n",
       "24      chrIII-2_tig00004935     767    1021    YNR077C   \n",
       "27      chrIII-2_tig00004935    1323    2063    YCL076W   \n",
       "30      chrIII-2_tig00004935    6426    8271    YCL073C   \n",
       "33        chrIII_tig00004936     760    1014    YNR077C   \n",
       "36        chrIII_tig00004936    1316    2055    YCL076W   \n",
       "651       chrIII_tig00004936  310848  311102    YNL337W   \n",
       "2097       chrII_tig00000010  789332  789467  YHL048C-A   \n",
       "2106       chrIV_tig00000001    1219    1356  YHL048C-A   \n",
       "4835       chrIX_tig00000023    2653    2790  YJL222W-B   \n",
       "4838       chrIX_tig00000023    2759    2986  YJL222W-A   \n",
       "4841       chrIX_tig00000023    3329    3525  YCR102W-A   \n",
       "5644        chrI_tig00000033     676     975    YDR543C   \n",
       "6054        chrI_tig00000033  238815  239136    YAL069W   \n",
       "6057     chrMANY_tig00000038     737    1036    YDR543C   \n",
       "6063     chrMANY_tig00000038    2728    2958  YJL222W-A   \n",
       "6917     chrVI-2_tig00004937   12581   12879    YDR543C   \n",
       "6920     chrVIII_tig00000017     680     934    YNR077C   \n",
       "7906     chrVIII_tig00000017  522253  522633  YAL064W-B   \n",
       "9422      chrVII_tig00004928  805604  806749    YGR295C   \n",
       "9425      chrVII_tig00004928  807356  807492  YHL048C-A   \n",
       "9428      chrVII_tig00004928  807660  807956  YER188C-A   \n",
       "9888       chrVI_tig00004938  298356  298655    YDR543C   \n",
       "11170     chrX-2_tig00001099     642     835    YCR108C   \n",
       "11173     chrX-2_tig00001099    2619    2756  YJL222W-B   \n",
       "11182     chrX-2_tig00001099    4754    5859    YCR102C   \n",
       "11185     chrX-2_tig00001099    7160   11768    YJL222W   \n",
       "11188     chrX-2_tig00001099   12417   14186    YIL172C   \n",
       "11221    chrXI-2_tig00004931   28713   28967    YNR077C   \n",
       "11224   chrXII-2_tig00004934     699     999    YDR543C   \n",
       "12027   chrXII-3_tig00004933    6551    6679  YBL108C-A   \n",
       "12030   chrXII-3_tig00004933    7121    7351    YGL260W   \n",
       "12060  chrXIII-2_tig00000029    1228    1363  YHL048C-A   \n",
       "13703    chrXIII_tig00000018  467347  467484  YHL048C-A   \n",
       "13709    chrXIII_tig00000018  467923  468106  YOR394C-A   \n",
       "14775     chrXII_tig00000020  554706  554903  YCR102W-A   \n",
       "14790     chrXII_tig00000020  557160  557459  YER188C-A   \n",
       "16142     chrXIV_tig00004929  728468  728695  YJL222W-A   \n",
       "16145     chrXIV_tig00004929  728662  728801  YJL222W-B   \n",
       "16158      chrXI_tig00004932     726     980    YNL337W   \n",
       "17809   chrXVI-2_tig00000040     743     997    YNL337W   \n",
       "19446     chrXVI_tig00000013  719024  719337    YAL069W   \n",
       "19500      chrXV_tig00000007   24888   25351  YIL171W-A   \n",
       "20876       chrX_tig00001098    2582    2719  YJL222W-B   \n",
       "20879       chrX_tig00001098    2688    2915  YJL222W-A   \n",
       "20885       chrX_tig00001098    4721    5826    YCR102C   \n",
       "20897       chrX_tig00001098   13824   14287  YIL171W-A   \n",
       "\n",
       "                                          Alias  \\\n",
       "9                                          None   \n",
       "21                                         None   \n",
       "24                                         None   \n",
       "27                                         None   \n",
       "30                 GEX1,glutathione%20exchanger   \n",
       "33                                         None   \n",
       "36                                         None   \n",
       "651                                        None   \n",
       "2097                                       None   \n",
       "2106                                       None   \n",
       "4835                                       None   \n",
       "4838                                       None   \n",
       "4841                                       None   \n",
       "5644                                       None   \n",
       "6054                                       None   \n",
       "6057                                       None   \n",
       "6063                                       None   \n",
       "6917                                       None   \n",
       "6920                                       None   \n",
       "7906                                       None   \n",
       "9422                                       COS6   \n",
       "9425                                       None   \n",
       "9428                                       None   \n",
       "9888                                       None   \n",
       "11170                                      None   \n",
       "11173                                      None   \n",
       "11182                                      None   \n",
       "11185  VTH2,signal%20sequence-binding%20protein   \n",
       "11188       IMA3,oligo-1%2C6-glucosidase%20IMA3   \n",
       "11221                                      None   \n",
       "11224                                      None   \n",
       "12027                  PAU9,seripauperin%20PAU9   \n",
       "12030                                      None   \n",
       "12060                                      None   \n",
       "13703                                      None   \n",
       "13709                                      None   \n",
       "14775                                      None   \n",
       "14790                                      None   \n",
       "16142                                      None   \n",
       "16145                                      None   \n",
       "16158                                      None   \n",
       "17809                                      None   \n",
       "19446                                      None   \n",
       "19500                                      None   \n",
       "20876                                      None   \n",
       "20879                                      None   \n",
       "20885                                      None   \n",
       "20897                                      None   \n",
       "\n",
       "                                                    Note  \\\n",
       "9      Dubious open reading frame%3B unlikely to enco...   \n",
       "21                  Putative protein of unknown function   \n",
       "24     Protein of unknown function%2C abundance chang...   \n",
       "27     Dubious open reading frame%3B unlikely to enco...   \n",
       "30     Proton:glutathione antiporter%3B localized to ...   \n",
       "33     Protein of unknown function%2C abundance chang...   \n",
       "36     Dubious open reading frame%3B unlikely to enco...   \n",
       "651    Dubious open reading frame%3B unlikely to enco...   \n",
       "2097   Putative protein of unknown function%3B identi...   \n",
       "2106   Putative protein of unknown function%3B identi...   \n",
       "4835   Dubious open reading frame%3B unlikely to enco...   \n",
       "4838   Dubious open reading frame%3B unlikely to enco...   \n",
       "4841   Dubious open reading frame%3B unlikely to enco...   \n",
       "5644   Dubious open reading frame%3B unlikely to enco...   \n",
       "6054   Dubious open reading frame%3B unlikely to enco...   \n",
       "6057   Dubious open reading frame%3B unlikely to enco...   \n",
       "6063   Dubious open reading frame%3B unlikely to enco...   \n",
       "6917   Dubious open reading frame%3B unlikely to enco...   \n",
       "6920   Protein of unknown function%2C abundance chang...   \n",
       "7906         Fungal-specific protein of unknown function   \n",
       "9422   Endosomal protein involved in turnover of plas...   \n",
       "9425   Putative protein of unknown function%3B identi...   \n",
       "9428                Putative protein of unknown function   \n",
       "9888   Dubious open reading frame%3B unlikely to enco...   \n",
       "11170  Putative protein of unknown function%3B identi...   \n",
       "11173  Dubious open reading frame%3B unlikely to enco...   \n",
       "11182  Putative quinone oxidoreductase%3B overexpress...   \n",
       "11185  Putative membrane glycoprotein%3B has strong s...   \n",
       "11188  Alpha-glucosidase%3B weak%2C but broad substra...   \n",
       "11221  Protein of unknown function%2C abundance chang...   \n",
       "11224  Dubious open reading frame%3B unlikely to enco...   \n",
       "12027  Protein of unknown function%3B member of the s...   \n",
       "12030  Putative protein of unknown function%3B transc...   \n",
       "12060  Putative protein of unknown function%3B identi...   \n",
       "13703  Putative protein of unknown function%3B identi...   \n",
       "13709  Protein of unknown function%3B identified by g...   \n",
       "14775  Dubious open reading frame%3B unlikely to enco...   \n",
       "14790               Putative protein of unknown function   \n",
       "16142  Dubious open reading frame%3B unlikely to enco...   \n",
       "16145  Dubious open reading frame%3B unlikely to enco...   \n",
       "16158  Dubious open reading frame%3B unlikely to enco...   \n",
       "17809  Dubious open reading frame%3B unlikely to enco...   \n",
       "19446  Dubious open reading frame%3B unlikely to enco...   \n",
       "19500  Dubious open reading frame%3B unlikely to enco...   \n",
       "20876  Dubious open reading frame%3B unlikely to enco...   \n",
       "20879  Dubious open reading frame%3B unlikely to enco...   \n",
       "20885  Putative quinone oxidoreductase%3B overexpress...   \n",
       "20897  Dubious open reading frame%3B unlikely to enco...   \n",
       "\n",
       "                                           Ontology_term score strand phase  \\\n",
       "9            GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "21           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "24           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "27           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "30     GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...     .      -     .   \n",
       "33           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "36           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "651          GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "2097         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "2106         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "4835         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "4838         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "4841         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "5644         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6054         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "6057         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6063         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6917         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "6920         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "7906         GO:0003674,GO:0005783,GO:0008150,SO:0000704     .      -     .   \n",
       "9422   GO:0000324,GO:0000324,GO:0003674,GO:0005768,GO...     .      -     .   \n",
       "9425         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "9428         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "9888         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "11170        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "11173        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "11182  GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...     .      +     .   \n",
       "11185  GO:0000324,GO:0005048,GO:0005048,GO:0006623,GO...     .      +     .   \n",
       "11188  GO:0004574,GO:0004574,GO:0004575,GO:0005737,GO...     .      -     .   \n",
       "11221        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "11224        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "12027        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "12030        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "12060        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "13703        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "13709        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "14775        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "14790        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16142        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16145        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16158        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "17809        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "19446        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "19500        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20876        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20879        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20885  GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...     .      +     .   \n",
       "20897        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "\n",
       "       ...                                            display  \\\n",
       "9      ...                         Dubious open reading frame   \n",
       "21     ...               Putative protein of unknown function   \n",
       "24     ...  Protein of unknown function%2C abundance chang...   \n",
       "27     ...                         Dubious open reading frame   \n",
       "30     ...                      Proton:glutathione antiporter   \n",
       "33     ...  Protein of unknown function%2C abundance chang...   \n",
       "36     ...                         Dubious open reading frame   \n",
       "651    ...                         Dubious open reading frame   \n",
       "2097   ...               Putative protein of unknown function   \n",
       "2106   ...               Putative protein of unknown function   \n",
       "4835   ...                         Dubious open reading frame   \n",
       "4838   ...                         Dubious open reading frame   \n",
       "4841   ...                         Dubious open reading frame   \n",
       "5644   ...                         Dubious open reading frame   \n",
       "6054   ...                         Dubious open reading frame   \n",
       "6057   ...                         Dubious open reading frame   \n",
       "6063   ...                         Dubious open reading frame   \n",
       "6917   ...                         Dubious open reading frame   \n",
       "6920   ...  Protein of unknown function%2C abundance chang...   \n",
       "7906   ...        Fungal-specific protein of unknown function   \n",
       "9422   ...  Endosomal protein involved in turnover of plas...   \n",
       "9425   ...               Putative protein of unknown function   \n",
       "9428   ...               Putative protein of unknown function   \n",
       "9888   ...                         Dubious open reading frame   \n",
       "11170  ...               Putative protein of unknown function   \n",
       "11173  ...                         Dubious open reading frame   \n",
       "11182  ...                    Putative quinone oxidoreductase   \n",
       "11185  ...                     Putative membrane glycoprotein   \n",
       "11188  ...                                  Alpha-glucosidase   \n",
       "11221  ...  Protein of unknown function%2C abundance chang...   \n",
       "11224  ...                         Dubious open reading frame   \n",
       "12027  ...                        Protein of unknown function   \n",
       "12030  ...               Putative protein of unknown function   \n",
       "12060  ...               Putative protein of unknown function   \n",
       "13703  ...               Putative protein of unknown function   \n",
       "13709  ...                        Protein of unknown function   \n",
       "14775  ...                         Dubious open reading frame   \n",
       "14790  ...               Putative protein of unknown function   \n",
       "16142  ...                         Dubious open reading frame   \n",
       "16145  ...                         Dubious open reading frame   \n",
       "16158  ...                         Dubious open reading frame   \n",
       "17809  ...                         Dubious open reading frame   \n",
       "19446  ...                         Dubious open reading frame   \n",
       "19500  ...                         Dubious open reading frame   \n",
       "20876  ...                         Dubious open reading frame   \n",
       "20879  ...                         Dubious open reading frame   \n",
       "20885  ...                    Putative quinone oxidoreductase   \n",
       "20897  ...                         Dubious open reading frame   \n",
       "\n",
       "      extra_copy_number  gene low_identity orf_classification partial_mapping  \\\n",
       "9                     2  None         None            Dubious            None   \n",
       "21                    3  None         None    Uncharacterized            None   \n",
       "24                    3  None         None    Uncharacterized            None   \n",
       "27                    2  None         None            Dubious            None   \n",
       "30                    2  GEX1         None           Verified            None   \n",
       "33                    2  None         None    Uncharacterized            None   \n",
       "36                    3  None         None            Dubious            None   \n",
       "651                   4  None         None            Dubious            None   \n",
       "2097                  6  None         None    Uncharacterized            None   \n",
       "2106                  4  None         None    Uncharacterized            None   \n",
       "4835                  4  None         None            Dubious            None   \n",
       "4838                  2  None         None            Dubious            None   \n",
       "4841                  3  None         None            Dubious            None   \n",
       "5644                  4  None         None            Dubious            None   \n",
       "6054                  2  None         None            Dubious            None   \n",
       "6057                  2  None         None            Dubious            None   \n",
       "6063                  5  None         None            Dubious            None   \n",
       "6917                  3  None         None            Dubious            None   \n",
       "6920                  5  None         None    Uncharacterized            None   \n",
       "7906                  2  None         None    Uncharacterized            None   \n",
       "9422                  2  COS6         None           Verified            None   \n",
       "9425                  5  None         None    Uncharacterized            None   \n",
       "9428                  4  None         None    Uncharacterized            None   \n",
       "9888                  5  None         None            Dubious            None   \n",
       "11170                 2  None         None    Uncharacterized            None   \n",
       "11173                 5  None         None            Dubious            None   \n",
       "11182                 2  None         None    Uncharacterized            None   \n",
       "11185                 2  VTH2         None           Verified            None   \n",
       "11188                 2  IMA3         None           Verified            None   \n",
       "11221                 4  None         None    Uncharacterized            None   \n",
       "11224                 6  None         None            Dubious            None   \n",
       "12027                 2  PAU9         None           Verified            None   \n",
       "12030                 2  None         None    Uncharacterized            None   \n",
       "12060                 2  None         None    Uncharacterized            None   \n",
       "13703                 3  None         None    Uncharacterized            None   \n",
       "13709                 2  None         None    Uncharacterized            None   \n",
       "14775                 2  None         None            Dubious            None   \n",
       "14790                 2  None         None    Uncharacterized            None   \n",
       "16142                 3  None         None            Dubious            None   \n",
       "16145                 3  None         None            Dubious            None   \n",
       "16158                 2  None         None            Dubious            None   \n",
       "17809                 3  None         None            Dubious            None   \n",
       "19446                 3  None         None            Dubious            None   \n",
       "19500                 2  None         None            Dubious            None   \n",
       "20876                 2  None         None            Dubious            None   \n",
       "20879                 4  None         None            Dubious            None   \n",
       "20885                 3  None         None    Uncharacterized            None   \n",
       "20897                 3  None         None            Dubious            None   \n",
       "\n",
       "      sequence_ID valid_ORFs Chromosome       Contig  \n",
       "9           0.989          0          2  tig00004927  \n",
       "21          0.943          1          2  tig00004927  \n",
       "24          0.918          1          3  tig00004935  \n",
       "27          0.981          0          3  tig00004935  \n",
       "30          0.990          0          3  tig00004935  \n",
       "33          0.918          1          3  tig00004936  \n",
       "36          0.979          0          3  tig00004936  \n",
       "651         0.949          1          3  tig00004936  \n",
       "2097        0.890          0          2  tig00000010  \n",
       "2106        0.920          0          4  tig00000001  \n",
       "4835        0.942          1          9  tig00000023  \n",
       "4838        0.947          1          9  tig00000023  \n",
       "4841        0.949          0          9  tig00000023  \n",
       "5644        0.946          0          1  tig00000033  \n",
       "6054        0.867          0          1  tig00000033  \n",
       "6057        0.936          0          7  tig00000038  \n",
       "6063        0.917          0          7  tig00000038  \n",
       "6917        0.943          0          6  tig00004937  \n",
       "6920        0.917          0          8  tig00000017  \n",
       "7906        0.934          1          8  tig00000017  \n",
       "9422        0.961          1          7  tig00004928  \n",
       "9425        0.927          0          7  tig00004928  \n",
       "9428         0.93          1          7  tig00004928  \n",
       "9888        0.936          0          6  tig00004938  \n",
       "11170       0.917          0         10  tig00001099  \n",
       "11173       0.949          1         10  tig00001099  \n",
       "11182       0.981          0         10  tig00001099  \n",
       "11185       0.972          0         10  tig00001099  \n",
       "11188       0.989          1         10  tig00001099  \n",
       "11221       0.918          1         11  tig00004931  \n",
       "11224       0.926          0         12  tig00004934  \n",
       "12027         1.0          1         12  tig00004933  \n",
       "12030       0.948          1         12  tig00004933  \n",
       "12060       0.948          0         13  tig00000029  \n",
       "13703       0.920          0         13  tig00000018  \n",
       "13709       0.822          0         13  tig00000018  \n",
       "14775       0.954          0         12  tig00000020  \n",
       "14790       0.956          1         12  tig00000020  \n",
       "16142       0.929          1         14  tig00004929  \n",
       "16145       0.921          0         14  tig00004929  \n",
       "16158       0.956          1         11  tig00004932  \n",
       "17809       0.941          1         16  tig00000040  \n",
       "19446       0.895          0         16  tig00000013  \n",
       "19500       0.956          0         15  tig00000007  \n",
       "20876       0.956          1         10  tig00001098  \n",
       "20879       0.939          0         10  tig00001098  \n",
       "20885       0.979          0         10  tig00001098  \n",
       "20897       0.952          0         10  tig00001098  \n",
       "\n",
       "[48 rows x 23 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[gene['extra_copy_number'] > '1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "73d236f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl = gene[gene.duplicated(subset=['Name'], keep=False) == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "af2f32ce",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl = dupl.sort_values(by = 'Name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "b4f28869",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl.to_excel('duplicated_verified_genes.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d155e624",
   "metadata": {},
   "source": [
    "Duplicated genes:\n",
    "\n",
    "ATP8/Q0080: Subunit 8 of the F0 sector of mitochondrial F1F0 ATP synthase; encoded on the mitochondrial genome; ATP8 and ATP6 mRNAs are not translated in the absence of the F1 sector of ATPase\n",
    "\n",
    "OLI1/Q0130: F0-ATP synthase subunit c (ATPase-associated proteolipid); encoded on the mitochondrial genome; mutation confers oligomycin resistance; expression is specifically dependent on the nuclear genes AEP1 and AEP2 1 2 3 4\n",
    "Name Description\n",
    "OLIgomycin resistance 1\n",
    "\n",
    "Both on mitochondria so not that relevant to my project.\n",
    "\n",
    "PAU9/YBL108C-A: Protein of unknown function; member of the seripauperin multigene family encoded mainly in subtelomeric regions; SWAT-GFP and mCherry fusion proteins localize to the endoplasmic reticulum and vacuole respectively\n",
    "\n",
    "YHR213W-A: Putative protein of unknown function; identified by gene-trapping, microarray-based expression analysis, and genome-wide homology searching "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
