{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fbef926e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from BCBio.GFF import GFFExaminer"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18ceb6d5",
   "metadata": {},
   "source": [
    "https://gffpandas.readthedocs.io/en/latest/tutorial.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "id": "88084c43",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gffpandas.gffpandas as gffpd\n",
    "import pandas as pd\n",
    "# Import Biopython tools for running local BLASTX\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0e4c4aab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\anaconda3\\lib\\site-packages\\gffpandas\\gffpandas.py:32: DtypeWarning: Columns (1,2,5,6,7,8) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  self.df = pd.read_table(self._gff_file, comment='#',\n"
     ]
    }
   ],
   "source": [
    "annotation = gffpd.read_gff3('YPS1009_S288C_v5_polished.gff')\n",
    "annot_s288c = gffpd.read_gff3('saccharomyces_cerevisiae_R64-3.gff')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "05ec8c7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "features = annotation.df['type'].value_counts().to_frame()\n",
    "features = features.rename(columns = {'type': 'yps1009'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1f380f29",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "features = features.join(annot_s288c.df['type'].value_counts().to_frame())\n",
    "features = features.rename(columns={'type': 'S288C'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "356223c4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>yps1009</th>\n",
       "      <th>S288C</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDS</th>\n",
       "      <td>7164</td>\n",
       "      <td>7066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <td>6691</td>\n",
       "      <td>6607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mRNA</th>\n",
       "      <td>6691</td>\n",
       "      <td>11119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARS</th>\n",
       "      <td>554</td>\n",
       "      <td>543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>noncoding_exon</th>\n",
       "      <td>512</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>long_terminal_repeat</th>\n",
       "      <td>361</td>\n",
       "      <td>384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA</th>\n",
       "      <td>323</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA_gene</th>\n",
       "      <td>323</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARS_consensus_sequence</th>\n",
       "      <td>199</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomere</th>\n",
       "      <td>98</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transposable_element</th>\n",
       "      <td>95</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transposable_element_gene</th>\n",
       "      <td>95</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>region</th>\n",
       "      <td>82</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snoRNA</th>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snoRNA_gene</th>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LTR_retrotransposon</th>\n",
       "      <td>53</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>plus_1_translational_frameshift</th>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomeric_repeat</th>\n",
       "      <td>42</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_element</th>\n",
       "      <td>35</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_element_combinatorial_repeat</th>\n",
       "      <td>33</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere</th>\n",
       "      <td>30</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pseudogene</th>\n",
       "      <td>25</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>five_prime_UTR_intron</th>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA_gene</th>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA</th>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA_gene</th>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA</th>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chromosome</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y_prime_element</th>\n",
       "      <td>17</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere_DNA_Element_II</th>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere_DNA_Element_III</th>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere_DNA_Element_I</th>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>blocked_reading_frame</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>origin_of_replication</th>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>external_transcribed_spacer_region</th>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>uORF</th>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snRNA_gene</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snRNA</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>silent_mating_type_cassette_array</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>internal_transcribed_spacer_region</th>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Z1_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>non_transcribed_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intein_encoding_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mating_type_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Z2_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomerase_RNA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomerase_RNA_gene</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>recombination_enhancer</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    yps1009  S288C\n",
       "CDS                                    7164   7066\n",
       "gene                                   6691   6607\n",
       "mRNA                                   6691  11119\n",
       "ARS                                     554    543\n",
       "noncoding_exon                          512    489\n",
       "long_terminal_repeat                    361    384\n",
       "tRNA                                    323    299\n",
       "tRNA_gene                               323    299\n",
       "ARS_consensus_sequence                  199    196\n",
       "telomere                                 98     64\n",
       "transposable_element                     95     91\n",
       "transposable_element_gene                95     91\n",
       "region                                   82     78\n",
       "snoRNA                                   77     77\n",
       "snoRNA_gene                              77     77\n",
       "LTR_retrotransposon                      53     50\n",
       "plus_1_translational_frameshift          47     47\n",
       "telomeric_repeat                         42     31\n",
       "X_element                                35     32\n",
       "X_element_combinatorial_repeat           33     28\n",
       "centromere                               30     32\n",
       "pseudogene                               25     12\n",
       "five_prime_UTR_intron                    24     24\n",
       "ncRNA_gene                               23     23\n",
       "ncRNA                                    23     23\n",
       "rRNA_gene                                20     24\n",
       "rRNA                                     20     24\n",
       "chromosome                               17     17\n",
       "Y_prime_element                          17     19\n",
       "centromere_DNA_Element_II                15     16\n",
       "centromere_DNA_Element_III               15     16\n",
       "centromere_DNA_Element_I                 15     16\n",
       "blocked_reading_frame                    12     12\n",
       "origin_of_replication                    10      8\n",
       "external_transcribed_spacer_region        6      8\n",
       "uORF                                      6     11\n",
       "snRNA_gene                                6      6\n",
       "snRNA                                     6      6\n",
       "silent_mating_type_cassette_array         4      4\n",
       "internal_transcribed_spacer_region        4      8\n",
       "X_region                                  3      3\n",
       "Y_region                                  3      3\n",
       "Z1_region                                 3      3\n",
       "non_transcribed_region                    3      3\n",
       "intein_encoding_region                    2      2\n",
       "W_region                                  2      2\n",
       "mating_type_region                        2      2\n",
       "Z2_region                                 2      2\n",
       "telomerase_RNA                            1      1\n",
       "telomerase_RNA_gene                       1      1\n",
       "recombination_enhancer                    1      1"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "301cc2fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find genes with major overlap (I only round the last digit of start and end position to zero, \n",
    "# so not perfect but will cover the biggest overlaps)\n",
    "def overlap_feat(df):\n",
    "    id_overlap = []\n",
    "    dupl = {}\n",
    "    for ind, row in df.iterrows():\n",
    "        start = str(row['start'])\n",
    "        start = start.replace(start[len(start) - 1:], '0')\n",
    "        end = str(row['end'])\n",
    "        end = end.replace(end[len(end) - 1:], '0')\n",
    "\n",
    "        tmp = str(row['seq_id']) + start + end\n",
    "        id_overlap.append(tmp)\n",
    "        if tmp in dupl:\n",
    "            dupl[tmp] += 1\n",
    "        else:\n",
    "            dupl[tmp] = 1\n",
    "    df['overlap id'] = id_overlap\n",
    "    filtered_dict = {k:v for (k,v) in dupl.items() if v > 1}\n",
    "    print('Number of site annotated several time: ' + str(len(filtered_dict)))\n",
    "    return df, filtered_dict\n",
    "\n",
    "def drop_overlap(df, filtered_dict):\n",
    "    df['sequence_ID'] = df['sequence_ID'].astype(float)\n",
    "    ind = []\n",
    "    for k in filtered_dict.keys():\n",
    "        tmp = df[df['overlap id'] == k]\n",
    "        ind.append(tmp.nsmallest(1,\"sequence_ID\").index[0])\n",
    "    df.drop(index=ind, inplace=True)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "678a7792",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This code just convert the chromosome numbering to integer and remove mitochondrial genes.\n",
    "def lat_to_arabic(DF):\n",
    "    Latin_to_arabic = {'I': 1, 'II':2, 'III':3, 'IV':4, 'V':5, 'VI':6, 'VII':7, 'VIII':8, 'IX':9, 'X':10,\n",
    "                      'XI':11, 'XII':12, 'XIII':13, 'XIV':14, 'XV':15, 'XVI':16, 'Mito': 'mitoch', 'MANY': 7}\n",
    "    seq_id = DF['seq_id']\n",
    "    ch = []\n",
    "    contig = []\n",
    "    for i in seq_id:\n",
    "        chrom, cont = i.split('_')\n",
    "        contig.append(cont)\n",
    "        chrom = chrom.replace('chr', '')\n",
    "        if '-' in chrom:\n",
    "            chrom, e = chrom.split('-')\n",
    "        ch.append(Latin_to_arabic[chrom])\n",
    "    DF['Chromosome'] = ch\n",
    "    DF['Contig'] = contig\n",
    "    return DF\n",
    "\n",
    "def lat_to_arabic_S288(df):\n",
    "    # This code just convert the chromosome numbering to integer and remove mitochondrial genes.\n",
    "    Latin_to_arabic = {'I': 1, 'II':2, 'III':3, 'IV':4, 'V':5, 'VI':6, 'VII':7, 'VIII':8, 'IX':9, 'X':10,\n",
    "                      'XI':11, 'XII':12, 'XIII':13, 'XIV':14, 'XV':15, 'XVI':16, 'Mito': 'mitoch'}\n",
    "    chLat = df['seq_id']\n",
    "    ch = []\n",
    "    for chrom in chLat:\n",
    "        chrom = chrom.replace('chr', '')\n",
    "        ch.append(Latin_to_arabic[chrom])\n",
    "    df['Chromosome'] = ch\n",
    "    return df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "563e58d8",
   "metadata": {},
   "source": [
    "# ARS\n",
    "Actual number in s288c should be 352"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a4021c90",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_df = annotation.filter_feature_of_type(['ARS'])\n",
    "ARS_attr = ARS_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0674542f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_s288c_df = annot_s288c.filter_feature_of_type(['ARS'])\n",
    "ARS_s288c_attr = ARS_s288c_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "d12bda0f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_s288c_attr = ARS_s288c_attr[ARS_s288c_attr['dbxref'].notna()]\n",
    "# Hurray, right number"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7e9a2a8b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_attr = ARS_attr[ARS_attr['dbxref'].notna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "11eb34d0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 2\n"
     ]
    }
   ],
   "source": [
    "ARS_attr, ARS_dupl = overlap_feat(ARS_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d45d63f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_attr = drop_overlap(ARS_attr, ARS_dupl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "7aec3bcc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>attributes</th>\n",
       "      <th>Alias</th>\n",
       "      <th>...</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>curie</th>\n",
       "      <th>dbxref</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>overlap id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>17490</td>\n",
       "      <td>18088</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS1025;Name=ARS1025;Note=Autonomously%20Re...</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS1025_0</td>\n",
       "      <td>0.990</td>\n",
       "      <td>SGD:S000028461</td>\n",
       "      <td>SGD:S000028461</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.905</td>\n",
       "      <td>chrII-2_tig000049271749010000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>17496</td>\n",
       "      <td>18717</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS400;Name=ARS400;Alias=ARSIV-1;Note=Auton...</td>\n",
       "      <td>ARSIV-1</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS400_0</td>\n",
       "      <td>0.963</td>\n",
       "      <td>SGD:S000121226</td>\n",
       "      <td>SGD:S000121226</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.890</td>\n",
       "      <td>chrII-2_tig000049271749018010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>11705</td>\n",
       "      <td>11883</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS301;Name=ARS301;gene=ARS301;Alias=ARS301...</td>\n",
       "      <td>ARS301,HML-E%20ARS</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS301_0</td>\n",
       "      <td>0.711</td>\n",
       "      <td>SGD:S000028477</td>\n",
       "      <td>SGD:S000028477</td>\n",
       "      <td>Inactive%20replication%20origin%20linked%20to%...</td>\n",
       "      <td>0</td>\n",
       "      <td>ARS301</td>\n",
       "      <td>None</td>\n",
       "      <td>0.670</td>\n",
       "      <td>chrIII-2_tig000049351170011880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>15042</td>\n",
       "      <td>15315</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS302;Name=ARS302;gene=ARS302;Alias=ARS302...</td>\n",
       "      <td>ARS302,HML-I%20ARS</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS302_0</td>\n",
       "      <td>0.996</td>\n",
       "      <td>SGD:S000028478</td>\n",
       "      <td>SGD:S000028478</td>\n",
       "      <td>Inactive%20replication%20origin%20linked%20to%...</td>\n",
       "      <td>0</td>\n",
       "      <td>ARS302</td>\n",
       "      <td>None</td>\n",
       "      <td>0.968</td>\n",
       "      <td>chrIII_tig000049361504010310</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>15337</td>\n",
       "      <td>15680</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS303;Name=ARS303;gene=ARS303;Alias=ARS303...</td>\n",
       "      <td>ARS303</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS303_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000028479</td>\n",
       "      <td>SGD:S000028479</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence%20on%20C...</td>\n",
       "      <td>0</td>\n",
       "      <td>ARS303</td>\n",
       "      <td>None</td>\n",
       "      <td>0.969</td>\n",
       "      <td>chrIII_tig000049361533015680</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23613</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>596429</td>\n",
       "      <td>596946</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS1019;Name=ARS1019;Note=Autonomously%20Re...</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS1019_0</td>\n",
       "      <td>0.998</td>\n",
       "      <td>SGD:S000028456</td>\n",
       "      <td>SGD:S000028456</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.990</td>\n",
       "      <td>chrX_tig00001098506420590940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23710</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>637949</td>\n",
       "      <td>638237</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS1020;Name=ARS1020;Alias=ARSX-654;Note=Au...</td>\n",
       "      <td>ARSX-654</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS1020_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000028457</td>\n",
       "      <td>SGD:S000028457</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.975</td>\n",
       "      <td>chrX_tig00001098637040638230</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23757</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>667581</td>\n",
       "      <td>667688</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS1021;Name=ARS1021;Alias=ARS121;Note=Auto...</td>\n",
       "      <td>ARS121</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS1021_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000007553</td>\n",
       "      <td>SGD:S000007553</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.953</td>\n",
       "      <td>chrX_tig00001098667580667600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23801</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>695435</td>\n",
       "      <td>695731</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS1022;Name=ARS1022;Alias=ARSX-712;Note=Au...</td>\n",
       "      <td>ARSX-712</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS1022_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000028458</td>\n",
       "      <td>SGD:S000028458</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.993</td>\n",
       "      <td>chrX_tig00001098690430695730</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23859</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ARS</td>\n",
       "      <td>720126</td>\n",
       "      <td>720729</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ARS1025_2;Name=ARS1025;Note=Autonomously%20...</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>ARS1025_2</td>\n",
       "      <td>0.990</td>\n",
       "      <td>SGD:S000028461</td>\n",
       "      <td>SGD:S000028461</td>\n",
       "      <td>Autonomously%20Replicating%20Sequence</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0.889</td>\n",
       "      <td>chrX_tig00001098720120720720</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>358 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     seq_id   source type   start     end score strand phase  \\\n",
       "18      chrII-2_tig00004927  Liftoff  ARS   17490   18088     .      .     .   \n",
       "21      chrII-2_tig00004927  Liftoff  ARS   17496   18717     .      -     .   \n",
       "48     chrIII-2_tig00004935  Liftoff  ARS   11705   11883     .      .     .   \n",
       "111      chrIII_tig00004936  Liftoff  ARS   15042   15315     .      .     .   \n",
       "114      chrIII_tig00004936  Liftoff  ARS   15337   15680     .      .     .   \n",
       "...                     ...      ...  ...     ...     ...   ...    ...   ...   \n",
       "23613      chrX_tig00001098  Liftoff  ARS  596429  596946     .      .     .   \n",
       "23710      chrX_tig00001098  Liftoff  ARS  637949  638237     .      .     .   \n",
       "23757      chrX_tig00001098  Liftoff  ARS  667581  667688     .      .     .   \n",
       "23801      chrX_tig00001098  Liftoff  ARS  695435  695731     .      .     .   \n",
       "23859      chrX_tig00001098  Liftoff  ARS  720126  720729     .      .     .   \n",
       "\n",
       "                                              attributes               Alias  \\\n",
       "18     ID=ARS1025;Name=ARS1025;Note=Autonomously%20Re...                None   \n",
       "21     ID=ARS400;Name=ARS400;Alias=ARSIV-1;Note=Auton...             ARSIV-1   \n",
       "48     ID=ARS301;Name=ARS301;gene=ARS301;Alias=ARS301...  ARS301,HML-E%20ARS   \n",
       "111    ID=ARS302;Name=ARS302;gene=ARS302;Alias=ARS302...  ARS302,HML-I%20ARS   \n",
       "114    ID=ARS303;Name=ARS303;gene=ARS303;Alias=ARS303...              ARS303   \n",
       "...                                                  ...                 ...   \n",
       "23613  ID=ARS1019;Name=ARS1019;Note=Autonomously%20Re...                None   \n",
       "23710  ID=ARS1020;Name=ARS1020;Alias=ARSX-654;Note=Au...            ARSX-654   \n",
       "23757  ID=ARS1021;Name=ARS1021;Alias=ARS121;Note=Auto...              ARS121   \n",
       "23801  ID=ARS1022;Name=ARS1022;Alias=ARSX-712;Note=Au...            ARSX-712   \n",
       "23859  ID=ARS1025_2;Name=ARS1025;Note=Autonomously%20...                None   \n",
       "\n",
       "       ... copy_num_ID coverage           curie          dbxref  \\\n",
       "18     ...   ARS1025_0    0.990  SGD:S000028461  SGD:S000028461   \n",
       "21     ...    ARS400_0    0.963  SGD:S000121226  SGD:S000121226   \n",
       "48     ...    ARS301_0    0.711  SGD:S000028477  SGD:S000028477   \n",
       "111    ...    ARS302_0    0.996  SGD:S000028478  SGD:S000028478   \n",
       "114    ...    ARS303_0      1.0  SGD:S000028479  SGD:S000028479   \n",
       "...    ...         ...      ...             ...             ...   \n",
       "23613  ...   ARS1019_0    0.998  SGD:S000028456  SGD:S000028456   \n",
       "23710  ...   ARS1020_0      1.0  SGD:S000028457  SGD:S000028457   \n",
       "23757  ...   ARS1021_0      1.0  SGD:S000007553  SGD:S000007553   \n",
       "23801  ...   ARS1022_0      1.0  SGD:S000028458  SGD:S000028458   \n",
       "23859  ...   ARS1025_2    0.990  SGD:S000028461  SGD:S000028461   \n",
       "\n",
       "                                                 display extra_copy_number  \\\n",
       "18                 Autonomously%20Replicating%20Sequence                 0   \n",
       "21                 Autonomously%20Replicating%20Sequence                 0   \n",
       "48     Inactive%20replication%20origin%20linked%20to%...                 0   \n",
       "111    Inactive%20replication%20origin%20linked%20to%...                 0   \n",
       "114    Autonomously%20Replicating%20Sequence%20on%20C...                 0   \n",
       "...                                                  ...               ...   \n",
       "23613              Autonomously%20Replicating%20Sequence                 0   \n",
       "23710              Autonomously%20Replicating%20Sequence                 0   \n",
       "23757              Autonomously%20Replicating%20Sequence                 0   \n",
       "23801              Autonomously%20Replicating%20Sequence                 0   \n",
       "23859              Autonomously%20Replicating%20Sequence                 2   \n",
       "\n",
       "         gene low_identity sequence_ID                      overlap id  \n",
       "18       None         None       0.905   chrII-2_tig000049271749010000  \n",
       "21       None         None       0.890   chrII-2_tig000049271749018010  \n",
       "48     ARS301         None       0.670  chrIII-2_tig000049351170011880  \n",
       "111    ARS302         None       0.968    chrIII_tig000049361504010310  \n",
       "114    ARS303         None       0.969    chrIII_tig000049361533015680  \n",
       "...       ...          ...         ...                             ...  \n",
       "23613    None         None       0.990    chrX_tig00001098506420590940  \n",
       "23710    None         None       0.975    chrX_tig00001098637040638230  \n",
       "23757    None         None       0.953    chrX_tig00001098667580667600  \n",
       "23801    None         None       0.993    chrX_tig00001098690430695730  \n",
       "23859    None         None       0.889    chrX_tig00001098720120720720  \n",
       "\n",
       "[358 rows x 24 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ARS_attr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "8c49d49d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_attr = lat_to_arabic(ARS_attr)\n",
    "ARS_s288c_attr = lat_to_arabic_S288(ARS_s288c_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ee7de14b",
   "metadata": {},
   "outputs": [],
   "source": [
    "ARS_count = ARS_attr['Chromosome'].value_counts().to_frame()\n",
    "ARS_count = ARS_count.rename(columns = {'Chromosome': 'yps1009 ARS'})\n",
    "ARS_count_s = ARS_s288c_attr['Chromosome'].value_counts().to_frame()\n",
    "ARS_count_s = ARS_count_s.rename(columns = {'Chromosome': 'S288C ARS'})\n",
    "ARS_count = ARS_count_s.join(ARS_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "97602747",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C ARS</th>\n",
       "      <th>yps1009 ARS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>41</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>29</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>28</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>27</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>26</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>25</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>23</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>21</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>19</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>19</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>16</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>16</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>16</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>14</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C ARS  yps1009 ARS\n",
       "4          41           44\n",
       "12         29           29\n",
       "7          28           28\n",
       "16         27           28\n",
       "13         26           25\n",
       "10         25           24\n",
       "2          23           24\n",
       "5          21           24\n",
       "14         21           20\n",
       "3          19           17\n",
       "15         19           21\n",
       "8          16           15\n",
       "9          16           16\n",
       "11         16           18\n",
       "6          14           14\n",
       "1          11           11"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ARS_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7de1edf7",
   "metadata": {},
   "source": [
    "# Long terminal repeats \n",
    "384 in s288c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "60df0270",
   "metadata": {},
   "outputs": [],
   "source": [
    "ltr_df = annotation.filter_feature_of_type(['long_terminal_repeat'])\n",
    "ltr_attr = ltr_df.attributes_to_columns()\n",
    "\n",
    "ltr_s288c_df = annot_s288c.filter_feature_of_type(['long_terminal_repeat'])\n",
    "ltr_s288c_attr = ltr_s288c_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3391da89",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 28\n"
     ]
    }
   ],
   "source": [
    "ltr_attr, ltr_dupl = overlap_feat(ltr_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "50bfa405",
   "metadata": {},
   "outputs": [],
   "source": [
    "ltr_attr = drop_overlap(ltr_attr, ltr_dupl)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "dc21c93d",
   "metadata": {},
   "outputs": [],
   "source": [
    "ltr_attr = lat_to_arabic(ltr_attr)\n",
    "ltr_s288c_attr = lat_to_arabic_S288(ltr_s288c_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "a2862cf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "ltr_count = ltr_attr['Chromosome'].value_counts().to_frame()\n",
    "ltr_count = ltr_count.rename(columns = {'Chromosome': 'yps1009 ltr'})\n",
    "ltr_count_s = ltr_s288c_attr['Chromosome'].value_counts().to_frame()\n",
    "ltr_count_s = ltr_count_s.rename(columns = {'Chromosome': 'S288C ltr'})\n",
    "ltr_count = ltr_count_s.join(ltr_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "3c50dc06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C ltr</th>\n",
       "      <th>yps1009 ltr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>44</td>\n",
       "      <td>85</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>36</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>34</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>32</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>32</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>32</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>24</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>24</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>22</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>16</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>11</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>9</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C ltr  yps1009 ltr\n",
       "7          44           85\n",
       "4          36           12\n",
       "5          34           31\n",
       "12         32           24\n",
       "15         32           25\n",
       "16         32           16\n",
       "8          24           24\n",
       "10         24           16\n",
       "13         24           14\n",
       "2          22           13\n",
       "3          20           18\n",
       "14         16            9\n",
       "11         14           12\n",
       "6          11           16\n",
       "9          10            6\n",
       "1           9           12"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ltr_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "0d77388b",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = ARS_count.join(ltr_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b3f2f13",
   "metadata": {},
   "source": [
    "# ncRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "f69e3f4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ncRNA_df = annotation.filter_feature_of_type(['ncRNA'])\n",
    "ncRNA_attr = ncRNA_df.attributes_to_columns()\n",
    "\n",
    "ncRNA_s288c_df = annot_s288c.filter_feature_of_type(['ncRNA'])\n",
    "ncRNA_s288c_attr = ncRNA_s288c_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "b96836a6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 1\n"
     ]
    }
   ],
   "source": [
    "ncRNA_attr, ncRNA_dupl = overlap_feat(ncRNA_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3949bd12",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>attributes</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Parent</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>overlap id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7919</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>201330</td>\n",
       "      <td>202036</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=YNCH0007W_ncRNA;Name=YNCH0007W_ncRNA;Parent...</td>\n",
       "      <td>YNCH0007W_ncRNA</td>\n",
       "      <td>YNCH0007W_ncRNA</td>\n",
       "      <td>YNCH0007W</td>\n",
       "      <td>0</td>\n",
       "      <td>chrVIII_tig00000017201330202030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7922</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>201330</td>\n",
       "      <td>202036</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=YNCH0008W_ncRNA;Name=YNCH0008W_ncRNA;Parent...</td>\n",
       "      <td>YNCH0008W_ncRNA</td>\n",
       "      <td>YNCH0008W_ncRNA</td>\n",
       "      <td>YNCH0008W</td>\n",
       "      <td>0</td>\n",
       "      <td>chrVIII_tig00000017201330202030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   seq_id   source   type   start     end score strand phase  \\\n",
       "7919  chrVIII_tig00000017  Liftoff  ncRNA  201330  202036     .      +     .   \n",
       "7922  chrVIII_tig00000017  Liftoff  ncRNA  201330  202036     .      +     .   \n",
       "\n",
       "                                             attributes               ID  \\\n",
       "7919  ID=YNCH0007W_ncRNA;Name=YNCH0007W_ncRNA;Parent...  YNCH0007W_ncRNA   \n",
       "7922  ID=YNCH0008W_ncRNA;Name=YNCH0008W_ncRNA;Parent...  YNCH0008W_ncRNA   \n",
       "\n",
       "                 Name     Parent extra_copy_number  \\\n",
       "7919  YNCH0007W_ncRNA  YNCH0007W                 0   \n",
       "7922  YNCH0008W_ncRNA  YNCH0008W                 0   \n",
       "\n",
       "                           overlap id  \n",
       "7919  chrVIII_tig00000017201330202030  \n",
       "7922  chrVIII_tig00000017201330202030  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ncRNA_attr[ncRNA_attr['overlap id'].isin(ncRNA_dupl)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "e65c2ecd",
   "metadata": {},
   "outputs": [],
   "source": [
    "ncRNA_attr = lat_to_arabic(ncRNA_attr)\n",
    "ncRNA_s288c_attr = lat_to_arabic_S288(ncRNA_s288c_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "a4d9bcd0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# no sequence identity so let's just drop one\n",
    "ncRNA_attr.drop(index=7922, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "ad4f602f",
   "metadata": {},
   "outputs": [],
   "source": [
    "ncRNA_count = ncRNA_attr['Chromosome'].value_counts().to_frame()\n",
    "ncRNA_count = ncRNA_count.rename(columns = {'Chromosome': 'yps1009 ncRNA'})\n",
    "ncRNA_count_s = ncRNA_s288c_attr['Chromosome'].value_counts().to_frame()\n",
    "ncRNA_count_s = ncRNA_count_s.rename(columns = {'Chromosome': 'S288C ncRNA'})\n",
    "ncRNA_count = ncRNA_count_s.join(ncRNA_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "baa8efab",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = FEATS.join(ncRNA_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8af024b1",
   "metadata": {},
   "source": [
    "# rRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "8e58e57c",
   "metadata": {},
   "outputs": [],
   "source": [
    "rRNA_df = annotation.filter_feature_of_type(['rRNA'])\n",
    "rRNA_attr = rRNA_df.attributes_to_columns()\n",
    "\n",
    "rRNA_s288c_df = annot_s288c.filter_feature_of_type(['rRNA'])\n",
    "rRNA_s288c_attr = rRNA_s288c_df.attributes_to_columns()\n",
    "\n",
    "rRNA_attr = lat_to_arabic(rRNA_attr)\n",
    "rRNA_s288c_attr = lat_to_arabic_S288(rRNA_s288c_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "2567428a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 7\n"
     ]
    }
   ],
   "source": [
    "rRNA_attr, rRNA_dupl = overlap_feat(rRNA_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7fabce65",
   "metadata": {},
   "outputs": [],
   "source": [
    "rRNA_attr = rRNA_attr.drop([13055, 14878, 14884, 14901, 14904, 14907, 14910])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "73a3f934",
   "metadata": {},
   "outputs": [],
   "source": [
    "rRNA_count = rRNA_attr['Chromosome'].value_counts().to_frame()\n",
    "rRNA_count = rRNA_count.rename(columns = {'Chromosome': 'yps1009 rRNA'})\n",
    "rRNA_count_s = rRNA_s288c_attr['Chromosome'].value_counts().to_frame()\n",
    "rRNA_count_s = rRNA_count_s.rename(columns = {'Chromosome': 'S288C rRNA'})\n",
    "rRNA_count = rRNA_count_s.join(rRNA_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "id": "55b661ef",
   "metadata": {},
   "outputs": [],
   "source": [
    " FEATS = FEATS.join(rRNA_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6334b3ee",
   "metadata": {},
   "source": [
    "# snRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "415c96da",
   "metadata": {},
   "outputs": [],
   "source": [
    "snRNA_df = annotation.filter_feature_of_type(['snRNA'])\n",
    "snRNA_attr = snRNA_df.attributes_to_columns()\n",
    "\n",
    "snRNA_s288c_df = annot_s288c.filter_feature_of_type(['snRNA'])\n",
    "snRNA_s288c_attr = snRNA_s288c_df.attributes_to_columns()\n",
    "\n",
    "snRNA_attr = lat_to_arabic(snRNA_attr)\n",
    "snRNA_s288c_attr = lat_to_arabic_S288(snRNA_s288c_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "32dd3e04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 0\n"
     ]
    }
   ],
   "source": [
    "snRNA_attr, snRNA_dupl = overlap_feat(snRNA_attr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "5d54629e",
   "metadata": {},
   "outputs": [],
   "source": [
    "snRNA_count = snRNA_attr['Chromosome'].value_counts().to_frame()\n",
    "snRNA_count = snRNA_count.rename(columns = {'Chromosome': 'yps1009 snRNA'})\n",
    "snRNA_count_s = snRNA_s288c_attr['Chromosome'].value_counts().to_frame()\n",
    "snRNA_count_s = snRNA_count_s.rename(columns = {'Chromosome': 'S288C snRNA'})\n",
    "snRNA_count = snRNA_count_s.join(snRNA_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "a1f4b4f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = FEATS.join(snRNA_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "43efab27",
   "metadata": {},
   "source": [
    "# Coding sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "ac34016c",
   "metadata": {},
   "outputs": [],
   "source": [
    "CDS_df = annotation.filter_feature_of_type(['CDS'])\n",
    "CDS_attr = CDS_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "09f2a21c",
   "metadata": {},
   "outputs": [],
   "source": [
    "CDS_attr['Name short'] = CDS_attr['Name'].str.replace('_CDS', '')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "547c27c6",
   "metadata": {},
   "source": [
    "# Centromere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a9a1cdf1",
   "metadata": {},
   "outputs": [],
   "source": [
    "CEN_df = annotation.filter_feature_of_type(['centromere'])\n",
    "CEN_attr = CEN_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "d392906d",
   "metadata": {},
   "outputs": [],
   "source": [
    "CEN_attr = CEN_attr[CEN_attr['Alias'].notna()]\n",
    "CEN_attr = CEN_attr.sort_values(by=['Alias'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8814528",
   "metadata": {},
   "source": [
    "# Telomere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "f3bd1a07",
   "metadata": {},
   "outputs": [],
   "source": [
    "telomere_df = annotation.filter_feature_of_type(['telomere'])\n",
    "telomere_attr = telomere_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "dbdfbbcb",
   "metadata": {},
   "outputs": [],
   "source": [
    "telomere_attr = telomere_attr[telomere_attr['Parent'].notna()]\n",
    "telomere_attr = telomere_attr.sort_values(by=['ID', 'start'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "6537dfb6",
   "metadata": {},
   "outputs": [],
   "source": [
    "telomere_attr['lenght'] = telomere_attr['end'] - telomere_attr['start']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "14c4df10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>attributes</th>\n",
       "      <th>ID</th>\n",
       "      <th>...</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>curie</th>\n",
       "      <th>dbxref</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>lenght</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>12145</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>566679</td>\n",
       "      <td>567486</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL01L_telomere;Name=TEL01L_telomere;Parent...</td>\n",
       "      <td>TEL01L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>807</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6770</th>\n",
       "      <td>chrVI-2_tig00004937</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>12572</td>\n",
       "      <td>13373</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL01L_telomere_1;Name=TEL01L_telomere;Pare...</td>\n",
       "      <td>TEL01L_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10765</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>407</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL02L_telomere;Name=TEL02L_telomere;Parent...</td>\n",
       "      <td>TEL02L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22440</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>235</td>\n",
       "      <td>1011</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL02R_telomere;Name=TEL02R_telomere;Parent...</td>\n",
       "      <td>TEL02R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14849</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>467643</td>\n",
       "      <td>468731</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03L_telomere;Name=TEL03L_telomere;Parent...</td>\n",
       "      <td>TEL03L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1088</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5169</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>215</td>\n",
       "      <td>1074</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03R_telomere;Name=TEL03R_telomere;Parent...</td>\n",
       "      <td>TEL03R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6043</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>176</td>\n",
       "      <td>984</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03R_telomere_1;Name=TEL03R_telomere;Pare...</td>\n",
       "      <td>TEL03R_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6775</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>219</td>\n",
       "      <td>1045</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03R_telomere_2;Name=TEL03R_telomere;Pare...</td>\n",
       "      <td>TEL03R_telomere_2</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>826</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19201</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>162</td>\n",
       "      <td>1006</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03R_telomere_3;Name=TEL03R_telomere;Pare...</td>\n",
       "      <td>TEL03R_telomere_3</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>844</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17476</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>730365</td>\n",
       "      <td>731164</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03R_telomere_4;Name=TEL03R_telomere;Pare...</td>\n",
       "      <td>TEL03R_telomere_4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>768</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>310839</td>\n",
       "      <td>311669</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL03R_telomere_5;Name=TEL03R_telomere;Pare...</td>\n",
       "      <td>TEL03R_telomere_5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22435</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>125</td>\n",
       "      <td>1011</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL04L_telomere;Name=TEL04L_telomere;Parent...</td>\n",
       "      <td>TEL04L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10780</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>464</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL04R_telomere;Name=TEL04R_telomere;Parent...</td>\n",
       "      <td>TEL04R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6628</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10775</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>423</td>\n",
       "      <td>7340</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL05L_telomere;Name=TEL05L_telomere;Parent...</td>\n",
       "      <td>TEL05L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10755</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>335</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL05R_telomere;Name=TEL05R_telomere;Parent...</td>\n",
       "      <td>TEL05R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10846</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>991</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL06L_telomere;Name=TEL06L_telomere;Parent...</td>\n",
       "      <td>TEL06L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6027</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>429036</td>\n",
       "      <td>429814</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere;Name=TEL07L_telomere;Parent...</td>\n",
       "      <td>TEL07L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12221</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>228</td>\n",
       "      <td>1008</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere_1;Name=TEL07L_telomere;Pare...</td>\n",
       "      <td>TEL07L_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13066</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>243</td>\n",
       "      <td>1027</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere_2;Name=TEL07L_telomere;Pare...</td>\n",
       "      <td>TEL07L_telomere_2</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12216</th>\n",
       "      <td>chrXI-2_tig00004931</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>28704</td>\n",
       "      <td>29461</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere_3;Name=TEL07L_telomere;Pare...</td>\n",
       "      <td>TEL07L_telomere_3</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>273</td>\n",
       "      <td>1030</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere_4;Name=TEL07L_telomere;Pare...</td>\n",
       "      <td>TEL07L_telomere_4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>257</td>\n",
       "      <td>1023</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere_5;Name=TEL07L_telomere;Pare...</td>\n",
       "      <td>TEL07L_telomere_5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18671</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>668361</td>\n",
       "      <td>669130</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07L_telomere_6;Name=TEL07L_telomere;Pare...</td>\n",
       "      <td>TEL07L_telomere_6</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10759</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>335</td>\n",
       "      <td>7340</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL07R_telomere;Name=TEL07R_telomere;Parent...</td>\n",
       "      <td>TEL07R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10744</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>242</td>\n",
       "      <td>7340</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL08L_telomere;Name=TEL08L_telomere;Parent...</td>\n",
       "      <td>TEL08L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>7098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10717</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>94</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL08R_telomere;Name=TEL08R_telomere;Parent...</td>\n",
       "      <td>TEL08R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10725</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>200</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL09L_telomere;Name=TEL09L_telomere;Parent...</td>\n",
       "      <td>TEL09L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6032</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>429036</td>\n",
       "      <td>429846</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL09R_telomere;Name=TEL09R_telomere;Parent...</td>\n",
       "      <td>TEL09R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23868</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>720627</td>\n",
       "      <td>721423</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL09R_telomere_1;Name=TEL09R_telomere;Pare...</td>\n",
       "      <td>TEL09R_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10733</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>219</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL10L_telomere;Name=TEL10L_telomere;Parent...</td>\n",
       "      <td>TEL10L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>17985</td>\n",
       "      <td>18806</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL10R_telomere;Name=TEL10R_telomere;Parent...</td>\n",
       "      <td>TEL10R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10177</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>807651</td>\n",
       "      <td>808483</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL10R_telomere_1;Name=TEL10R_telomere;Pare...</td>\n",
       "      <td>TEL10R_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2296</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>789625</td>\n",
       "      <td>790471</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL10R_telomere_2;Name=TEL10R_telomere;Pare...</td>\n",
       "      <td>TEL10R_telomere_2</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13112</th>\n",
       "      <td>chrXIII-2_tig00000029</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>224</td>\n",
       "      <td>1069</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL10R_telomere_3;Name=TEL10R_telomere;Pare...</td>\n",
       "      <td>TEL10R_telomere_3</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>845</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2305</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>211</td>\n",
       "      <td>1060</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL10R_telomere_4;Name=TEL10R_telomere;Pare...</td>\n",
       "      <td>TEL10R_telomere_4</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20939</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>154</td>\n",
       "      <td>963</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL11L_telomere;Name=TEL11L_telomere;Parent...</td>\n",
       "      <td>TEL11L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5154</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>1481069</td>\n",
       "      <td>1481896</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL11L_telomere_1;Name=TEL11L_telomere;Pare...</td>\n",
       "      <td>TEL11L_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5164</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>97</td>\n",
       "      <td>1068</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL11R_telomere;Name=TEL11R_telomere;Parent...</td>\n",
       "      <td>TEL11R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10729</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>204</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL12L_telomere;Name=TEL12L_telomere;Parent...</td>\n",
       "      <td>TEL12L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10768</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>407</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL12R_telomere;Name=TEL12R_telomere;Parent...</td>\n",
       "      <td>TEL12R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10737</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>230</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL13L_telomere;Name=TEL13L_telomere;Parent...</td>\n",
       "      <td>TEL13L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16045</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>557151</td>\n",
       "      <td>558049</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL13R_telomere;Name=TEL13R_telomere;Parent...</td>\n",
       "      <td>TEL13R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10752</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>320</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL14L_telomere;Name=TEL14L_telomere;Parent...</td>\n",
       "      <td>TEL14L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5159</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>5</td>\n",
       "      <td>1074</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL14R_telomere;Name=TEL14R_telomere;Parent...</td>\n",
       "      <td>TEL14R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10705</th>\n",
       "      <td>chrVI_tig00004938</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>298347</td>\n",
       "      <td>299355</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL14R_telomere_1;Name=TEL14R_telomere;Pare...</td>\n",
       "      <td>TEL14R_telomere_1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20933</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>113</td>\n",
       "      <td>963</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL15L_telomere;Name=TEL15L_telomere;Parent...</td>\n",
       "      <td>TEL15L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10748</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>260</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL15R_telomere;Name=TEL15R_telomere;Parent...</td>\n",
       "      <td>TEL15R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6832</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10771</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>407</td>\n",
       "      <td>7092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL16L_telomere;Name=TEL16L_telomere;Parent...</td>\n",
       "      <td>TEL16L_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6685</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10762</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>telomere</td>\n",
       "      <td>377</td>\n",
       "      <td>6980</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=TEL16R_telomere;Name=TEL16R_telomere;Parent...</td>\n",
       "      <td>TEL16R_telomere</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>6603</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49 rows Ã— 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      seq_id   source      type    start      end score  \\\n",
       "12145       chrV_tig00000015  Liftoff  telomere   566679   567486     .   \n",
       "6770     chrVI-2_tig00004937  Liftoff  telomere    12572    13373     .   \n",
       "10765       chrV_tig00000015  Liftoff  telomere      407     7092     .   \n",
       "22440       chrX_tig00001098  Liftoff  telomere      235     1011     .   \n",
       "14849    chrXIII_tig00000018  Liftoff  telomere   467643   468731     .   \n",
       "5169       chrIX_tig00000023  Liftoff  telomere      215     1074     .   \n",
       "6043        chrI_tig00000033  Liftoff  telomere      176      984     .   \n",
       "6775    chrVII-2_tig00000038  Liftoff  telomere      219     1045     .   \n",
       "19201   chrXVI-2_tig00000040  Liftoff  telomere      162     1006     .   \n",
       "17476     chrXIV_tig00004929  Liftoff  telomere   730365   731164     .   \n",
       "768       chrIII_tig00004936  Liftoff  telomere   310839   311669     .   \n",
       "22435       chrX_tig00001098  Liftoff  telomere      125     1011     .   \n",
       "10780       chrV_tig00000015  Liftoff  telomere      464     7092     .   \n",
       "10775       chrV_tig00000015  Liftoff  telomere      423     7340     .   \n",
       "10755       chrV_tig00000015  Liftoff  telomere      335     7092     .   \n",
       "10846       chrV_tig00000015  Liftoff  telomere      991     7092     .   \n",
       "6027       chrIX_tig00000023  Liftoff  telomere   429036   429814     .   \n",
       "12221   chrXII-2_tig00004934  Liftoff  telomere      228     1008     .   \n",
       "13066   chrXII-3_tig00004933  Liftoff  telomere      243     1027     .   \n",
       "12216    chrXI-2_tig00004931  Liftoff  telomere    28704    29461     .   \n",
       "35      chrIII-2_tig00004935  Liftoff  telomere      273     1030     .   \n",
       "55        chrIII_tig00004936  Liftoff  telomere      257     1023     .   \n",
       "18671      chrXI_tig00004932  Liftoff  telomere   668361   669130     .   \n",
       "10759       chrV_tig00000015  Liftoff  telomere      335     7340     .   \n",
       "10744       chrV_tig00000015  Liftoff  telomere      242     7340     .   \n",
       "10717       chrV_tig00000015  Liftoff  telomere       94     7092     .   \n",
       "10725       chrV_tig00000015  Liftoff  telomere      200     7092     .   \n",
       "6032       chrIX_tig00000023  Liftoff  telomere   429036   429846     .   \n",
       "23868       chrX_tig00001098  Liftoff  telomere   720627   721423     .   \n",
       "10733       chrV_tig00000015  Liftoff  telomere      219     7092     .   \n",
       "30       chrII-2_tig00004927  Liftoff  telomere    17985    18806     .   \n",
       "10177     chrVII_tig00004928  Liftoff  telomere   807651   808483     .   \n",
       "2296       chrII_tig00000010  Liftoff  telomere   789625   790471     .   \n",
       "13112  chrXIII-2_tig00000029  Liftoff  telomere      224     1069     .   \n",
       "2305       chrIV_tig00000001  Liftoff  telomere      211     1060     .   \n",
       "20939      chrXV_tig00000007  Liftoff  telomere      154      963     .   \n",
       "5154       chrIV_tig00000001  Liftoff  telomere  1481069  1481896     .   \n",
       "5164       chrIX_tig00000023  Liftoff  telomere       97     1068     .   \n",
       "10729       chrV_tig00000015  Liftoff  telomere      204     7092     .   \n",
       "10768       chrV_tig00000015  Liftoff  telomere      407     7092     .   \n",
       "10737       chrV_tig00000015  Liftoff  telomere      230     7092     .   \n",
       "16045     chrXII_tig00000020  Liftoff  telomere   557151   558049     .   \n",
       "10752       chrV_tig00000015  Liftoff  telomere      320     7092     .   \n",
       "5159       chrIX_tig00000023  Liftoff  telomere        5     1074     .   \n",
       "10705      chrVI_tig00004938  Liftoff  telomere   298347   299355     .   \n",
       "20933      chrXV_tig00000007  Liftoff  telomere      113      963     .   \n",
       "10748       chrV_tig00000015  Liftoff  telomere      260     7092     .   \n",
       "10771       chrV_tig00000015  Liftoff  telomere      407     7092     .   \n",
       "10762       chrV_tig00000015  Liftoff  telomere      377     6980     .   \n",
       "\n",
       "      strand phase                                         attributes  \\\n",
       "12145      +     .  ID=TEL01L_telomere;Name=TEL01L_telomere;Parent...   \n",
       "6770       +     .  ID=TEL01L_telomere_1;Name=TEL01L_telomere;Pare...   \n",
       "10765      -     .  ID=TEL02L_telomere;Name=TEL02L_telomere;Parent...   \n",
       "22440      -     .  ID=TEL02R_telomere;Name=TEL02R_telomere;Parent...   \n",
       "14849      +     .  ID=TEL03L_telomere;Name=TEL03L_telomere;Parent...   \n",
       "5169       -     .  ID=TEL03R_telomere;Name=TEL03R_telomere;Parent...   \n",
       "6043       -     .  ID=TEL03R_telomere_1;Name=TEL03R_telomere;Pare...   \n",
       "6775       -     .  ID=TEL03R_telomere_2;Name=TEL03R_telomere;Pare...   \n",
       "19201      -     .  ID=TEL03R_telomere_3;Name=TEL03R_telomere;Pare...   \n",
       "17476      +     .  ID=TEL03R_telomere_4;Name=TEL03R_telomere;Pare...   \n",
       "768        +     .  ID=TEL03R_telomere_5;Name=TEL03R_telomere;Pare...   \n",
       "22435      -     .  ID=TEL04L_telomere;Name=TEL04L_telomere;Parent...   \n",
       "10780      -     .  ID=TEL04R_telomere;Name=TEL04R_telomere;Parent...   \n",
       "10775      -     .  ID=TEL05L_telomere;Name=TEL05L_telomere;Parent...   \n",
       "10755      -     .  ID=TEL05R_telomere;Name=TEL05R_telomere;Parent...   \n",
       "10846      -     .  ID=TEL06L_telomere;Name=TEL06L_telomere;Parent...   \n",
       "6027       +     .  ID=TEL07L_telomere;Name=TEL07L_telomere;Parent...   \n",
       "12221      -     .  ID=TEL07L_telomere_1;Name=TEL07L_telomere;Pare...   \n",
       "13066      -     .  ID=TEL07L_telomere_2;Name=TEL07L_telomere;Pare...   \n",
       "12216      +     .  ID=TEL07L_telomere_3;Name=TEL07L_telomere;Pare...   \n",
       "35         -     .  ID=TEL07L_telomere_4;Name=TEL07L_telomere;Pare...   \n",
       "55         -     .  ID=TEL07L_telomere_5;Name=TEL07L_telomere;Pare...   \n",
       "18671      +     .  ID=TEL07L_telomere_6;Name=TEL07L_telomere;Pare...   \n",
       "10759      -     .  ID=TEL07R_telomere;Name=TEL07R_telomere;Parent...   \n",
       "10744      -     .  ID=TEL08L_telomere;Name=TEL08L_telomere;Parent...   \n",
       "10717      -     .  ID=TEL08R_telomere;Name=TEL08R_telomere;Parent...   \n",
       "10725      -     .  ID=TEL09L_telomere;Name=TEL09L_telomere;Parent...   \n",
       "6032       +     .  ID=TEL09R_telomere;Name=TEL09R_telomere;Parent...   \n",
       "23868      +     .  ID=TEL09R_telomere_1;Name=TEL09R_telomere;Pare...   \n",
       "10733      -     .  ID=TEL10L_telomere;Name=TEL10L_telomere;Parent...   \n",
       "30         +     .  ID=TEL10R_telomere;Name=TEL10R_telomere;Parent...   \n",
       "10177      +     .  ID=TEL10R_telomere_1;Name=TEL10R_telomere;Pare...   \n",
       "2296       +     .  ID=TEL10R_telomere_2;Name=TEL10R_telomere;Pare...   \n",
       "13112      -     .  ID=TEL10R_telomere_3;Name=TEL10R_telomere;Pare...   \n",
       "2305       -     .  ID=TEL10R_telomere_4;Name=TEL10R_telomere;Pare...   \n",
       "20939      -     .  ID=TEL11L_telomere;Name=TEL11L_telomere;Parent...   \n",
       "5154       +     .  ID=TEL11L_telomere_1;Name=TEL11L_telomere;Pare...   \n",
       "5164       -     .  ID=TEL11R_telomere;Name=TEL11R_telomere;Parent...   \n",
       "10729      -     .  ID=TEL12L_telomere;Name=TEL12L_telomere;Parent...   \n",
       "10768      -     .  ID=TEL12R_telomere;Name=TEL12R_telomere;Parent...   \n",
       "10737      -     .  ID=TEL13L_telomere;Name=TEL13L_telomere;Parent...   \n",
       "16045      +     .  ID=TEL13R_telomere;Name=TEL13R_telomere;Parent...   \n",
       "10752      -     .  ID=TEL14L_telomere;Name=TEL14L_telomere;Parent...   \n",
       "5159       -     .  ID=TEL14R_telomere;Name=TEL14R_telomere;Parent...   \n",
       "10705      +     .  ID=TEL14R_telomere_1;Name=TEL14R_telomere;Pare...   \n",
       "20933      -     .  ID=TEL15L_telomere;Name=TEL15L_telomere;Parent...   \n",
       "10748      -     .  ID=TEL15R_telomere;Name=TEL15R_telomere;Parent...   \n",
       "10771      -     .  ID=TEL16L_telomere;Name=TEL16L_telomere;Parent...   \n",
       "10762      -     .  ID=TEL16R_telomere;Name=TEL16R_telomere;Parent...   \n",
       "\n",
       "                      ID  ... copy_num_ID coverage curie dbxref display  \\\n",
       "12145    TEL01L_telomere  ...        None     None  None   None    None   \n",
       "6770   TEL01L_telomere_1  ...        None     None  None   None    None   \n",
       "10765    TEL02L_telomere  ...        None     None  None   None    None   \n",
       "22440    TEL02R_telomere  ...        None     None  None   None    None   \n",
       "14849    TEL03L_telomere  ...        None     None  None   None    None   \n",
       "5169     TEL03R_telomere  ...        None     None  None   None    None   \n",
       "6043   TEL03R_telomere_1  ...        None     None  None   None    None   \n",
       "6775   TEL03R_telomere_2  ...        None     None  None   None    None   \n",
       "19201  TEL03R_telomere_3  ...        None     None  None   None    None   \n",
       "17476  TEL03R_telomere_4  ...        None     None  None   None    None   \n",
       "768    TEL03R_telomere_5  ...        None     None  None   None    None   \n",
       "22435    TEL04L_telomere  ...        None     None  None   None    None   \n",
       "10780    TEL04R_telomere  ...        None     None  None   None    None   \n",
       "10775    TEL05L_telomere  ...        None     None  None   None    None   \n",
       "10755    TEL05R_telomere  ...        None     None  None   None    None   \n",
       "10846    TEL06L_telomere  ...        None     None  None   None    None   \n",
       "6027     TEL07L_telomere  ...        None     None  None   None    None   \n",
       "12221  TEL07L_telomere_1  ...        None     None  None   None    None   \n",
       "13066  TEL07L_telomere_2  ...        None     None  None   None    None   \n",
       "12216  TEL07L_telomere_3  ...        None     None  None   None    None   \n",
       "35     TEL07L_telomere_4  ...        None     None  None   None    None   \n",
       "55     TEL07L_telomere_5  ...        None     None  None   None    None   \n",
       "18671  TEL07L_telomere_6  ...        None     None  None   None    None   \n",
       "10759    TEL07R_telomere  ...        None     None  None   None    None   \n",
       "10744    TEL08L_telomere  ...        None     None  None   None    None   \n",
       "10717    TEL08R_telomere  ...        None     None  None   None    None   \n",
       "10725    TEL09L_telomere  ...        None     None  None   None    None   \n",
       "6032     TEL09R_telomere  ...        None     None  None   None    None   \n",
       "23868  TEL09R_telomere_1  ...        None     None  None   None    None   \n",
       "10733    TEL10L_telomere  ...        None     None  None   None    None   \n",
       "30       TEL10R_telomere  ...        None     None  None   None    None   \n",
       "10177  TEL10R_telomere_1  ...        None     None  None   None    None   \n",
       "2296   TEL10R_telomere_2  ...        None     None  None   None    None   \n",
       "13112  TEL10R_telomere_3  ...        None     None  None   None    None   \n",
       "2305   TEL10R_telomere_4  ...        None     None  None   None    None   \n",
       "20939    TEL11L_telomere  ...        None     None  None   None    None   \n",
       "5154   TEL11L_telomere_1  ...        None     None  None   None    None   \n",
       "5164     TEL11R_telomere  ...        None     None  None   None    None   \n",
       "10729    TEL12L_telomere  ...        None     None  None   None    None   \n",
       "10768    TEL12R_telomere  ...        None     None  None   None    None   \n",
       "10737    TEL13L_telomere  ...        None     None  None   None    None   \n",
       "16045    TEL13R_telomere  ...        None     None  None   None    None   \n",
       "10752    TEL14L_telomere  ...        None     None  None   None    None   \n",
       "5159     TEL14R_telomere  ...        None     None  None   None    None   \n",
       "10705  TEL14R_telomere_1  ...        None     None  None   None    None   \n",
       "20933    TEL15L_telomere  ...        None     None  None   None    None   \n",
       "10748    TEL15R_telomere  ...        None     None  None   None    None   \n",
       "10771    TEL16L_telomere  ...        None     None  None   None    None   \n",
       "10762    TEL16R_telomere  ...        None     None  None   None    None   \n",
       "\n",
       "      extra_copy_number low_identity partial_mapping sequence_ID lenght  \n",
       "12145                 0         None            None        None    807  \n",
       "6770                  1         None            None        None    801  \n",
       "10765                 0         None            None        None   6685  \n",
       "22440                 0         None            None        None    776  \n",
       "14849                 0         None            None        None   1088  \n",
       "5169                  0         None            None        None    859  \n",
       "6043                  1         None            None        None    808  \n",
       "6775                  2         None            None        None    826  \n",
       "19201                 3         None            None        None    844  \n",
       "17476                 4         None            None        None    799  \n",
       "768                   5         None            None        None    830  \n",
       "22435                 0         None            None        None    886  \n",
       "10780                 0         None            None        None   6628  \n",
       "10775                 0         None            None        None   6917  \n",
       "10755                 0         None            None        None   6757  \n",
       "10846                 0         None            None        None   6101  \n",
       "6027                  0         None            None        None    778  \n",
       "12221                 1         None            None        None    780  \n",
       "13066                 2         None            None        None    784  \n",
       "12216                 3         None            None        None    757  \n",
       "35                    4         None            None        None    757  \n",
       "55                    5         None            None        None    766  \n",
       "18671                 6         None            None        None    769  \n",
       "10759                 0         None            None        None   7005  \n",
       "10744                 0         None            None        None   7098  \n",
       "10717                 0         None            None        None   6998  \n",
       "10725                 0         None            None        None   6892  \n",
       "6032                  0         None            None        None    810  \n",
       "23868                 1         None            None        None    796  \n",
       "10733                 0         None            None        None   6873  \n",
       "30                    0         None            None        None    821  \n",
       "10177                 1         None            None        None    832  \n",
       "2296                  2         None            None        None    846  \n",
       "13112                 3         None            None        None    845  \n",
       "2305                  4         None            None        None    849  \n",
       "20939                 0         None            None        None    809  \n",
       "5154                  1         None            None        None    827  \n",
       "5164                  0         None            None        None    971  \n",
       "10729                 0         None            None        None   6888  \n",
       "10768                 0         None            None        None   6685  \n",
       "10737                 0         None            None        None   6862  \n",
       "16045                 0         None            None        None    898  \n",
       "10752                 0         None            None        None   6772  \n",
       "5159                  0         None            None        None   1069  \n",
       "10705                 1         None            None        None   1008  \n",
       "20933                 0         None            None        None    850  \n",
       "10748                 0         None            None        None   6832  \n",
       "10771                 0         None            None        None   6685  \n",
       "10762                 0         None            None        None   6603  \n",
       "\n",
       "[49 rows x 23 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# messy, forget it for now\n",
    "telomere_attr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1009c702",
   "metadata": {},
   "source": [
    "## Make df by features YPS1009"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "45985f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_df = annotation.filter_feature_of_type(['transposable_element_gene'])\n",
    "sno_df = annotation.filter_feature_of_type(['snoRNA_gene'])\n",
    "tRNA_df = annotation.filter_feature_of_type(['tRNA_gene'])\n",
    "gene_df = annotation.filter_feature_of_type(['gene'])\n",
    "LTR_retroP_df = annotation.filter_feature_of_type(['LTR_retrotransposon'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "aa90ef94",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_attr = gene_df.attributes_to_columns()\n",
    "gene = gene_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
    "       'partial_mapping', 'sequence_ID', 'valid_ORFs']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "65fb5bab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'Ontology_term',\n",
       "       'copy_num_ID', 'coverage', 'curie', 'dbxref', 'display',\n",
       "       'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
       "       'partial_mapping', 'sequence_ID', 'valid_ORFs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "da9223db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    6165\n",
       "0     524\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr['valid_ORFs'].value_counts()\n",
    "# so valid_ORFs in gene does count itself and I should look carefully at the 439 genes with no valid ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "a9acec8b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6565"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# number of annotated genes without potential duplicates\n",
    "# Doesn't works, we want duplicates\n",
    "len(set(gene['Name']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "9605e8fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6691, 21)"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "a2feefd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_attr = TE_df.attributes_to_columns()\n",
    "sno_attr = sno_df.attributes_to_columns()\n",
    "tRNA_attr = tRNA_df.attributes_to_columns()\n",
    "LTR_retroP_attr = LTR_retroP_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "5f4982fd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'copy_num_ID', 'coverage',\n",
       "       'curie', 'dbxref', 'display', 'extra_copy_number', 'low_identity',\n",
       "       'sequence_ID'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LTR_retroP_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "id": "8d70a21e",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE = TE_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'sequence_ID', 'valid_ORFs']]\n",
    "\n",
    "sno = sno_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'sequence_ID']]\n",
    "\n",
    "tRNA = tRNA_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'sequence_ID']]\n",
    "\n",
    "LTR_retroP = LTR_retroP_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'sequence_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "47a1360f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n"
     ]
    }
   ],
   "source": [
    "gene = lat_to_arabic(gene)\n",
    "TE = lat_to_arabic(TE)\n",
    "sno = lat_to_arabic(sno)\n",
    "tRNA = lat_to_arabic(tRNA)\n",
    "LTR_retroP = lat_to_arabic(LTR_retroP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "88d25b7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean20(DF):\n",
    "    note = DF['Note']\n",
    "    dis = DF['display']\n",
    "    clean = []\n",
    "    display = []\n",
    "    for n in note:\n",
    "        clean.append(n.replace('%20', ' '))\n",
    "    for d in dis:\n",
    "        display.append(str(d).replace('%20', ' '))\n",
    "    DF['Note'] = clean\n",
    "    DF['display'] = display\n",
    "    return DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "50940e10",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\741321415.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Note'] = clean\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\741321415.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['display'] = display\n"
     ]
    }
   ],
   "source": [
    "gene = clean20(gene)\n",
    "TE = clean20(TE)\n",
    "sno = clean20(sno)\n",
    "tRNA = clean20(tRNA)\n",
    "LTR_retroP = clean20(LTR_retroP)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "259832d6",
   "metadata": {},
   "source": [
    "## Df by attribute for S288C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "685b570a",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_df_s288c = annot_s288c.filter_feature_of_type(['transposable_element_gene'])\n",
    "sno_df_s288c = annot_s288c.filter_feature_of_type(['snoRNA_gene'])\n",
    "tRNA_df_s288c = annot_s288c.filter_feature_of_type(['tRNA_gene'])\n",
    "gene_df_s288c = annot_s288c.filter_feature_of_type(['gene'])\n",
    "LTR_retroP_df_s288c = annot_s288c.filter_feature_of_type(['LTR_retrotransposon'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "98713fc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_attr_s288c = gene_df_s288c.attributes_to_columns()\n",
    "TE_attr_s288c = TE_df_s288c.attributes_to_columns()\n",
    "sno_attr_s288c = sno_df_s288c.attributes_to_columns()\n",
    "tRNA_attr_s288c = tRNA_df_s288c.attributes_to_columns()\n",
    "LTR_retroP_attr_s288c = LTR_retroP_df_s288c.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "4dc8b278",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'curie', 'dbxref',\n",
       "       'display'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LTR_retroP_attr_s288c.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "a65bdbc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_s288c = gene_attr_s288c[['seq_id', 'start', 'end', 'Name', 'gene',  'Alias',  'Note', 'Ontology_term', 'strand', 'phase', 'curie',\n",
    "        'orf_classification', 'display']]\n",
    "\n",
    "TE_s288c = TE_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "sno_s288c = sno_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "tRNA_s288c = tRNA_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "LTR_retroP_s288c = LTR_retroP_attr_s288c[['seq_id', 'start', 'end',  'ID', 'Name',  'Alias', 'score', 'strand', 'phase',\n",
    "       'attributes', 'Note', 'curie', 'display']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "4ef36a0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2888190848.py:28: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n"
     ]
    }
   ],
   "source": [
    "gene_s288c = lat_to_arabic_S288(gene_s288c)\n",
    "TE_s288c = lat_to_arabic_S288(TE_s288c)\n",
    "sno_s288c = lat_to_arabic_S288(sno_s288c)\n",
    "tRNA_s288c = lat_to_arabic_S288(tRNA_s288c)\n",
    "LTR_retroP_s288c = lat_to_arabic_S288(LTR_retroP_s288c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "ea5b328c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\741321415.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Note'] = clean\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\741321415.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['display'] = display\n"
     ]
    }
   ],
   "source": [
    "gene_s288c = clean20(gene_s288c)\n",
    "TE_s288c = clean20(TE_s288c)\n",
    "sno_s288c = clean20(sno_s288c)\n",
    "tRNA_s288c = clean20(tRNA_s288c)\n",
    "LTR_retroP_s288c = clean20(LTR_retroP_s288c)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6c35c4c3",
   "metadata": {},
   "source": [
    "# TE \n",
    "\n",
    "### IMPORTANT:\n",
    "\n",
    "There are several TE mapped to the exact same location. Probably because map to 2 genes that have a very similar sequence!\n",
    "That's bad, I need to remove them. DO MANUALLY. --> Only 17 unique TE found!!!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "cd662573",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE.to_excel('TE_liftoff_v5_yps1009.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e4f72d62",
   "metadata": {},
   "source": [
    "## LTR Retrotransposons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "fc19f941",
   "metadata": {},
   "outputs": [],
   "source": [
    "LTR_RP_count = LTR_retroP['Chromosome'].value_counts().to_frame()\n",
    "LTR_RP_count = LTR_RP_count.rename(columns = {'Chromosome' : 'yps1009 LTR retrotransposon'})\n",
    "LTR_RP_count_s288 = LTR_retroP_s288c['Chromosome'].value_counts().to_frame()\n",
    "LTR_RP_count_s288 = LTR_RP_count_s288.rename(columns= {'Chromosome': 'S288C LTR retroposon'})\n",
    "LTR_RP_count = LTR_RP_count_s288.join(LTR_RP_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d21895d",
   "metadata": {},
   "source": [
    "# IMPORTANT \n",
    "Overlapping features, to filter manually "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "3bc2910b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>curie</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6374</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177542</td>\n",
       "      <td>183707</td>\n",
       "      <td>YDRWTy2-3</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy2-3_0</td>\n",
       "      <td>0.862</td>\n",
       "      <td>SGD:S000006885</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.835</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6375</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177544</td>\n",
       "      <td>183707</td>\n",
       "      <td>YGRWTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRWTy2-2_0</td>\n",
       "      <td>0.861</td>\n",
       "      <td>SGD:S000006983</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.834</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6376</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177545</td>\n",
       "      <td>183707</td>\n",
       "      <td>YDRWTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy2-2_0</td>\n",
       "      <td>0.860</td>\n",
       "      <td>SGD:S000006884</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>423</td>\n",
       "      <td>4934</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_1</td>\n",
       "      <td>0.993</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.948</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>790</td>\n",
       "      <td>4554</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_0</td>\n",
       "      <td>0.997</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.954</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>136337</td>\n",
       "      <td>142463</td>\n",
       "      <td>YBLWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YBLWTy2-1_0</td>\n",
       "      <td>0.856</td>\n",
       "      <td>SGD:S000006809</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.823</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4750</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297667</td>\n",
       "      <td>1303627</td>\n",
       "      <td>YLRWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRWTy2-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007100</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.812</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4751</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297674</td>\n",
       "      <td>1309617</td>\n",
       "      <td>YNLWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YNLWTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007142</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.426</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4752</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1303627</td>\n",
       "      <td>YDRCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy2-1_0</td>\n",
       "      <td>0.832</td>\n",
       "      <td>SGD:S000006862</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.812</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4753</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1304036</td>\n",
       "      <td>YFLWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YFLWTy2-1_0</td>\n",
       "      <td>0.866</td>\n",
       "      <td>SGD:S000006930</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.826</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4754</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1303630</td>\n",
       "      <td>YNLCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YNLCTy2-1_0</td>\n",
       "      <td>0.935</td>\n",
       "      <td>SGD:S000007135</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.913</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4755</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1303627</td>\n",
       "      <td>YORWTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YORWTy2-2_0</td>\n",
       "      <td>0.831</td>\n",
       "      <td>SGD:S000007183</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.810</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4806</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1303049</td>\n",
       "      <td>1309617</td>\n",
       "      <td>YJRWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YJRWTy1-1_0</td>\n",
       "      <td>0.864</td>\n",
       "      <td>SGD:S000007048</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.757</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4808</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1303305</td>\n",
       "      <td>1309617</td>\n",
       "      <td>YPLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YPLWTy1-1_0</td>\n",
       "      <td>0.877</td>\n",
       "      <td>SGD:S000007199</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.808</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12092</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>543496</td>\n",
       "      <td>548928</td>\n",
       "      <td>YILWTy3-1</td>\n",
       "      <td>Ty3</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YILWTy3-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007020</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>5</td>\n",
       "      <td>tig00000015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12097</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>543497</td>\n",
       "      <td>548928</td>\n",
       "      <td>YGRWTy3-1</td>\n",
       "      <td>Ty3</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRWTy3-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000006984</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.818</td>\n",
       "      <td>5</td>\n",
       "      <td>tig00000015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6975</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106301</td>\n",
       "      <td>112481</td>\n",
       "      <td>YERCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YERCTy1-1_0</td>\n",
       "      <td>0.868</td>\n",
       "      <td>SGD:S000006910</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.807</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6980</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106477</td>\n",
       "      <td>112460</td>\n",
       "      <td>YMRCTy1-4</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YMRCTy1-4_0</td>\n",
       "      <td>0.837</td>\n",
       "      <td>SGD:S000007120</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.790</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6981</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106488</td>\n",
       "      <td>112460</td>\n",
       "      <td>YDRCTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy1-3_0</td>\n",
       "      <td>0.839</td>\n",
       "      <td>SGD:S000006861</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.733</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6982</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106565</td>\n",
       "      <td>112460</td>\n",
       "      <td>YDRCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy1-1_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000006859</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.757</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6984</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106565</td>\n",
       "      <td>112485</td>\n",
       "      <td>YPRCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YPRCTy1-2_0</td>\n",
       "      <td>0.831</td>\n",
       "      <td>SGD:S000007208</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.761</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6986</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106566</td>\n",
       "      <td>112460</td>\n",
       "      <td>YBRWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YBRWTy1-2_0</td>\n",
       "      <td>0.829</td>\n",
       "      <td>SGD:S000006821</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.792</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6987</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106567</td>\n",
       "      <td>112460</td>\n",
       "      <td>YPRWTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YPRWTy1-3_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000007220</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.776</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6989</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YDRWTy1-4</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy1-4_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000006882</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.776</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6990</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YGRCTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRCTy1-3_0</td>\n",
       "      <td>0.829</td>\n",
       "      <td>SGD:S000006963</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.793</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6991</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YHRCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YHRCTy1-1_0</td>\n",
       "      <td>0.813</td>\n",
       "      <td>SGD:S000007006</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.769</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6992</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112459</td>\n",
       "      <td>YMLWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YMLWTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007108</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.771</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6995</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YORWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YORWTy1-2_0</td>\n",
       "      <td>0.829</td>\n",
       "      <td>SGD:S000007182</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.787</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6996</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YARCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YARCTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006792</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.769</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6998</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112481</td>\n",
       "      <td>YDRWTy1-5</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy1-5_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006883</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.766</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7001</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112460</td>\n",
       "      <td>YGRCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRCTy1-2_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000006962</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.759</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7002</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YGRWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRWTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006982</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.793</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7004</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YJRWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YJRWTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007049</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.787</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7006</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YLRCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRCTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007081</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.772</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7008</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112460</td>\n",
       "      <td>YLRWTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRWTy1-3_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000007099</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.763</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7010</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112460</td>\n",
       "      <td>YMLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YMLWTy1-1_0</td>\n",
       "      <td>0.828</td>\n",
       "      <td>SGD:S000007107</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.778</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7011</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112481</td>\n",
       "      <td>YNLCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YNLCTy1-1_0</td>\n",
       "      <td>0.832</td>\n",
       "      <td>SGD:S000007134</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.814</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7013</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YOLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YOLWTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007161</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.777</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7015</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YPRCTy1-4</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YPRCTy1-4_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007209</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.779</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7397</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>212802</td>\n",
       "      <td>219077</td>\n",
       "      <td>YHLWTy4-1</td>\n",
       "      <td>Ty4</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YHLWTy4-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000006991</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.821</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7399</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>212802</td>\n",
       "      <td>219075</td>\n",
       "      <td>YJLWTy4-1</td>\n",
       "      <td>Ty4</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YJLWTy4-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007039</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.819</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7401</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>212802</td>\n",
       "      <td>219075</td>\n",
       "      <td>YPLCTy4-1</td>\n",
       "      <td>Ty4</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YPLCTy4-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007188</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.821</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8302</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>376739</td>\n",
       "      <td>382705</td>\n",
       "      <td>YORCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YORCTy2-1_0</td>\n",
       "      <td>0.834</td>\n",
       "      <td>SGD:S000007168</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.815</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12205</th>\n",
       "      <td>chrXI-2_tig00004931</td>\n",
       "      <td>24798</td>\n",
       "      <td>29311</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_3</td>\n",
       "      <td>0.993</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.945</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18660</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>664456</td>\n",
       "      <td>668970</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_2</td>\n",
       "      <td>0.994</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.945</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12925</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414716</td>\n",
       "      <td>420627</td>\n",
       "      <td>YBLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YBLWTy1-1_0</td>\n",
       "      <td>0.832</td>\n",
       "      <td>SGD:S000006808</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.826</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12926</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414716</td>\n",
       "      <td>420627</td>\n",
       "      <td>YMRCTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YMRCTy1-3_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007119</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.825</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12927</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414736</td>\n",
       "      <td>420627</td>\n",
       "      <td>YDRCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy1-2_0</td>\n",
       "      <td>0.826</td>\n",
       "      <td>SGD:S000006860</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.755</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12928</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414910</td>\n",
       "      <td>420627</td>\n",
       "      <td>YERCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YERCTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006911</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.791</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21161</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>112119</td>\n",
       "      <td>118082</td>\n",
       "      <td>YGRCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRCTy2-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000006964</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.813</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21162</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>112119</td>\n",
       "      <td>118084</td>\n",
       "      <td>YLRCTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRCTy2-2_0</td>\n",
       "      <td>0.834</td>\n",
       "      <td>SGD:S000007082</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.751</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21359</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>228916</td>\n",
       "      <td>235127</td>\n",
       "      <td>YLRWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRWTy1-2_0</td>\n",
       "      <td>0.871</td>\n",
       "      <td>SGD:S000007098</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.793</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19303</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>54637</td>\n",
       "      <td>60826</td>\n",
       "      <td>YCLWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy2-1_0</td>\n",
       "      <td>0.865</td>\n",
       "      <td>SGD:S000006830</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.843</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000040</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     seq_id    start      end       Name Alias  \\\n",
       "6374       chrI_tig00000033   177542   183707  YDRWTy2-3   Ty2   \n",
       "6375       chrI_tig00000033   177544   183707  YGRWTy2-2   Ty2   \n",
       "6376       chrI_tig00000033   177545   183707  YDRWTy2-2   Ty2   \n",
       "37     chrIII-2_tig00004935      423     4934  YCLWTy5-1   Ty5   \n",
       "57       chrIII_tig00004936      790     4554  YCLWTy5-1   Ty5   \n",
       "407      chrIII_tig00004936   136337   142463  YBLWTy2-1   Ty2   \n",
       "4750      chrIV_tig00000001  1297667  1303627  YLRWTy2-1   Ty2   \n",
       "4751      chrIV_tig00000001  1297674  1309617  YNLWTy1-2   Ty1   \n",
       "4752      chrIV_tig00000001  1297677  1303627  YDRCTy2-1   Ty2   \n",
       "4753      chrIV_tig00000001  1297677  1304036  YFLWTy2-1   Ty2   \n",
       "4754      chrIV_tig00000001  1297677  1303630  YNLCTy2-1   Ty2   \n",
       "4755      chrIV_tig00000001  1297677  1303627  YORWTy2-2   Ty2   \n",
       "4806      chrIV_tig00000001  1303049  1309617  YJRWTy1-1   Ty1   \n",
       "4808      chrIV_tig00000001  1303305  1309617  YPLWTy1-1   Ty1   \n",
       "12092      chrV_tig00000015   543496   548928  YILWTy3-1   Ty3   \n",
       "12097      chrV_tig00000015   543497   548928  YGRWTy3-1   Ty3   \n",
       "6975   chrVII-2_tig00000038   106301   112481  YERCTy1-1   Ty1   \n",
       "6980   chrVII-2_tig00000038   106477   112460  YMRCTy1-4   Ty1   \n",
       "6981   chrVII-2_tig00000038   106488   112460  YDRCTy1-3   Ty1   \n",
       "6982   chrVII-2_tig00000038   106565   112460  YDRCTy1-1   Ty1   \n",
       "6984   chrVII-2_tig00000038   106565   112485  YPRCTy1-2   Ty1   \n",
       "6986   chrVII-2_tig00000038   106566   112460  YBRWTy1-2   Ty1   \n",
       "6987   chrVII-2_tig00000038   106567   112460  YPRWTy1-3   Ty1   \n",
       "6989   chrVII-2_tig00000038   106569   112460  YDRWTy1-4   Ty1   \n",
       "6990   chrVII-2_tig00000038   106569   112460  YGRCTy1-3   Ty1   \n",
       "6991   chrVII-2_tig00000038   106569   112460  YHRCTy1-1   Ty1   \n",
       "6992   chrVII-2_tig00000038   106569   112459  YMLWTy1-2   Ty1   \n",
       "6995   chrVII-2_tig00000038   106569   112460  YORWTy1-2   Ty1   \n",
       "6996   chrVII-2_tig00000038   106570   112480  YARCTy1-1   Ty1   \n",
       "6998   chrVII-2_tig00000038   106570   112481  YDRWTy1-5   Ty1   \n",
       "7001   chrVII-2_tig00000038   106570   112460  YGRCTy1-2   Ty1   \n",
       "7002   chrVII-2_tig00000038   106570   112480  YGRWTy1-1   Ty1   \n",
       "7004   chrVII-2_tig00000038   106570   112480  YJRWTy1-2   Ty1   \n",
       "7006   chrVII-2_tig00000038   106570   112480  YLRCTy1-1   Ty1   \n",
       "7008   chrVII-2_tig00000038   106570   112460  YLRWTy1-3   Ty1   \n",
       "7010   chrVII-2_tig00000038   106570   112460  YMLWTy1-1   Ty1   \n",
       "7011   chrVII-2_tig00000038   106570   112481  YNLCTy1-1   Ty1   \n",
       "7013   chrVII-2_tig00000038   106570   112480  YOLWTy1-1   Ty1   \n",
       "7015   chrVII-2_tig00000038   106570   112480  YPRCTy1-4   Ty1   \n",
       "7397   chrVII-2_tig00000038   212802   219077  YHLWTy4-1   Ty4   \n",
       "7399   chrVII-2_tig00000038   212802   219075  YJLWTy4-1   Ty4   \n",
       "7401   chrVII-2_tig00000038   212802   219075  YPLCTy4-1   Ty4   \n",
       "8302    chrVIII_tig00000017   376739   382705  YORCTy2-1   Ty2   \n",
       "12205   chrXI-2_tig00004931    24798    29311  YCLWTy5-1   Ty5   \n",
       "18660     chrXI_tig00004932   664456   668970  YCLWTy5-1   Ty5   \n",
       "12925  chrXII-2_tig00004934   414716   420627  YBLWTy1-1   Ty1   \n",
       "12926  chrXII-2_tig00004934   414716   420627  YMRCTy1-3   Ty1   \n",
       "12927  chrXII-2_tig00004934   414736   420627  YDRCTy1-2   Ty1   \n",
       "12928  chrXII-2_tig00004934   414910   420627  YERCTy1-2   Ty1   \n",
       "21161     chrXV_tig00000007   112119   118082  YGRCTy2-1   Ty2   \n",
       "21162     chrXV_tig00000007   112119   118084  YLRCTy2-2   Ty2   \n",
       "21359     chrXV_tig00000007   228916   235127  YLRWTy1-2   Ty1   \n",
       "19303  chrXVI-2_tig00000040    54637    60826  YCLWTy2-1   Ty2   \n",
       "\n",
       "                                                    Note score strand phase  \\\n",
       "6374   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6375   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6376   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "37     Ty5 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "57     Ty5 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "407    Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4750   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4751   Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4752   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4753   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4754   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4755   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4806   Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4808   Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12092  Ty3 element%2C LTR retrotransposon of the Gyps...     .      +     .   \n",
       "12097  Ty3 element%2C LTR retrotransposon of the Gyps...     .      +     .   \n",
       "6975   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6980   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6981   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6982   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6984   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6986   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6987   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6989   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6990   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6991   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6992   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6995   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6996   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6998   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7001   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7002   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7004   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7006   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7008   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7010   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7011   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7013   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7015   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7397   Ty4 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "7399   Ty4 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "7401   Ty4 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "8302   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "12205  Ty5 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "18660  Ty5 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12925  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12926  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12927  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12928  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "21161  Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "21162  Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "21359  Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "19303  Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "\n",
       "       copy_num_ID coverage           curie  \\\n",
       "6374   YDRWTy2-3_0    0.862  SGD:S000006885   \n",
       "6375   YGRWTy2-2_0    0.861  SGD:S000006983   \n",
       "6376   YDRWTy2-2_0    0.860  SGD:S000006884   \n",
       "37     YCLWTy5-1_1    0.993  SGD:S000006831   \n",
       "57     YCLWTy5-1_0    0.997  SGD:S000006831   \n",
       "407    YBLWTy2-1_0    0.856  SGD:S000006809   \n",
       "4750   YLRWTy2-1_0    0.833  SGD:S000007100   \n",
       "4751   YNLWTy1-2_0    0.830  SGD:S000007142   \n",
       "4752   YDRCTy2-1_0    0.832  SGD:S000006862   \n",
       "4753   YFLWTy2-1_0    0.866  SGD:S000006930   \n",
       "4754   YNLCTy2-1_0    0.935  SGD:S000007135   \n",
       "4755   YORWTy2-2_0    0.831  SGD:S000007183   \n",
       "4806   YJRWTy1-1_0    0.864  SGD:S000007048   \n",
       "4808   YPLWTy1-1_0    0.877  SGD:S000007199   \n",
       "12092  YILWTy3-1_0    0.833  SGD:S000007020   \n",
       "12097  YGRWTy3-1_0    0.833  SGD:S000006984   \n",
       "6975   YERCTy1-1_0    0.868  SGD:S000006910   \n",
       "6980   YMRCTy1-4_0    0.837  SGD:S000007120   \n",
       "6981   YDRCTy1-3_0    0.839  SGD:S000006861   \n",
       "6982   YDRCTy1-1_0    0.827  SGD:S000006859   \n",
       "6984   YPRCTy1-2_0    0.831  SGD:S000007208   \n",
       "6986   YBRWTy1-2_0    0.829  SGD:S000006821   \n",
       "6987   YPRWTy1-3_0    0.827  SGD:S000007220   \n",
       "6989   YDRWTy1-4_0    0.827  SGD:S000006882   \n",
       "6990   YGRCTy1-3_0    0.829  SGD:S000006963   \n",
       "6991   YHRCTy1-1_0    0.813  SGD:S000007006   \n",
       "6992   YMLWTy1-2_0    0.830  SGD:S000007108   \n",
       "6995   YORWTy1-2_0    0.829  SGD:S000007182   \n",
       "6996   YARCTy1-1_0    0.830  SGD:S000006792   \n",
       "6998   YDRWTy1-5_0    0.830  SGD:S000006883   \n",
       "7001   YGRCTy1-2_0    0.827  SGD:S000006962   \n",
       "7002   YGRWTy1-1_0    0.830  SGD:S000006982   \n",
       "7004   YJRWTy1-2_0    0.830  SGD:S000007049   \n",
       "7006   YLRCTy1-1_0    0.830  SGD:S000007081   \n",
       "7008   YLRWTy1-3_0    0.827  SGD:S000007099   \n",
       "7010   YMLWTy1-1_0    0.828  SGD:S000007107   \n",
       "7011   YNLCTy1-1_0    0.832  SGD:S000007134   \n",
       "7013   YOLWTy1-1_0    0.830  SGD:S000007161   \n",
       "7015   YPRCTy1-4_0    0.830  SGD:S000007209   \n",
       "7397   YHLWTy4-1_0    0.833  SGD:S000006991   \n",
       "7399   YJLWTy4-1_0    0.833  SGD:S000007039   \n",
       "7401   YPLCTy4-1_0    0.833  SGD:S000007188   \n",
       "8302   YORCTy2-1_0    0.834  SGD:S000007168   \n",
       "12205  YCLWTy5-1_3    0.993  SGD:S000006831   \n",
       "18660  YCLWTy5-1_2    0.994  SGD:S000006831   \n",
       "12925  YBLWTy1-1_0    0.832  SGD:S000006808   \n",
       "12926  YMRCTy1-3_0    0.833  SGD:S000007119   \n",
       "12927  YDRCTy1-2_0    0.826  SGD:S000006860   \n",
       "12928  YERCTy1-2_0    0.830  SGD:S000006911   \n",
       "21161  YGRCTy2-1_0    0.833  SGD:S000006964   \n",
       "21162  YLRCTy2-2_0    0.834  SGD:S000007082   \n",
       "21359  YLRWTy1-2_0    0.871  SGD:S000007098   \n",
       "19303  YCLWTy2-1_0    0.865  SGD:S000006830   \n",
       "\n",
       "                                                 display extra_copy_number  \\\n",
       "6374   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6375   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6376   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "37     Ty5 element%2C LTR retrotransposon of the Copi...                 1   \n",
       "57     Ty5 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "407    Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4750   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4751   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4752   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4753   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4754   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4755   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4806   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4808   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12092  Ty3 element%2C LTR retrotransposon of the Gyps...                 0   \n",
       "12097  Ty3 element%2C LTR retrotransposon of the Gyps...                 0   \n",
       "6975   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6980   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6981   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6982   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6984   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6986   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6987   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6989   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6990   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6991   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6992   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6995   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6996   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6998   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7001   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7002   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7004   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7006   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7008   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7010   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7011   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7013   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7015   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7397   Ty4 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7399   Ty4 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7401   Ty4 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "8302   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12205  Ty5 element%2C LTR retrotransposon of the Copi...                 3   \n",
       "18660  Ty5 element%2C LTR retrotransposon of the Copi...                 2   \n",
       "12925  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12926  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12927  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12928  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "21161  Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "21162  Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "21359  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "19303  Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "\n",
       "      sequence_ID  Chromosome       Contig  \n",
       "6374        0.835           1  tig00000033  \n",
       "6375        0.834           1  tig00000033  \n",
       "6376        0.833           1  tig00000033  \n",
       "37          0.948           3  tig00004935  \n",
       "57          0.954           3  tig00004936  \n",
       "407         0.823           3  tig00004936  \n",
       "4750        0.812           4  tig00000001  \n",
       "4751        0.426           4  tig00000001  \n",
       "4752        0.812           4  tig00000001  \n",
       "4753        0.826           4  tig00000001  \n",
       "4754        0.913           4  tig00000001  \n",
       "4755        0.810           4  tig00000001  \n",
       "4806        0.757           4  tig00000001  \n",
       "4808        0.808           4  tig00000001  \n",
       "12092       0.827           5  tig00000015  \n",
       "12097       0.818           5  tig00000015  \n",
       "6975        0.807           7  tig00000038  \n",
       "6980        0.790           7  tig00000038  \n",
       "6981        0.733           7  tig00000038  \n",
       "6982        0.757           7  tig00000038  \n",
       "6984        0.761           7  tig00000038  \n",
       "6986        0.792           7  tig00000038  \n",
       "6987        0.776           7  tig00000038  \n",
       "6989        0.776           7  tig00000038  \n",
       "6990        0.793           7  tig00000038  \n",
       "6991        0.769           7  tig00000038  \n",
       "6992        0.771           7  tig00000038  \n",
       "6995        0.787           7  tig00000038  \n",
       "6996        0.769           7  tig00000038  \n",
       "6998        0.766           7  tig00000038  \n",
       "7001        0.759           7  tig00000038  \n",
       "7002        0.793           7  tig00000038  \n",
       "7004        0.787           7  tig00000038  \n",
       "7006        0.772           7  tig00000038  \n",
       "7008        0.763           7  tig00000038  \n",
       "7010        0.778           7  tig00000038  \n",
       "7011        0.814           7  tig00000038  \n",
       "7013        0.777           7  tig00000038  \n",
       "7015        0.779           7  tig00000038  \n",
       "7397        0.821           7  tig00000038  \n",
       "7399        0.819           7  tig00000038  \n",
       "7401        0.821           7  tig00000038  \n",
       "8302        0.815           8  tig00000017  \n",
       "12205       0.945          11  tig00004931  \n",
       "18660       0.945          11  tig00004932  \n",
       "12925       0.826          12  tig00004934  \n",
       "12926       0.825          12  tig00004934  \n",
       "12927       0.755          12  tig00004934  \n",
       "12928       0.791          12  tig00004934  \n",
       "21161       0.813          15  tig00000007  \n",
       "21162       0.751          15  tig00000007  \n",
       "21359       0.793          15  tig00000007  \n",
       "19303       0.843          16  tig00000040  "
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LTR_retroP.sort_values(by=['Chromosome', 'start'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "94b233f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "LTR_retroP.to_excel('Liftoff_LTR_retrop.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "97626402",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['overlap id'] = id_overlap\n"
     ]
    }
   ],
   "source": [
    "LTR_retroP_F, LTR_dic = overlap_feat(LTR_retroP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "3a0d7e9d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['sequence_ID'] = df['sequence_ID'].astype(float)\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df.drop(index=ind, inplace=True)\n"
     ]
    }
   ],
   "source": [
    "LTR_retroP_F = drop_overlap(LTR_retroP_F, LTR_dic)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5df993b9",
   "metadata": {},
   "source": [
    "How do I reconciliate these results? it somehow feels not right\n",
    "compare to the TE Ultimate results. See TE_annotation jupyter notebook "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "d0ef8150",
   "metadata": {},
   "outputs": [],
   "source": [
    "filteredRetroP = pd.read_excel('Liftoff_LTR_retrop_filtered.xlsx', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "id": "a000640c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(16, 13)"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filteredRetroP.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "84289231",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = filteredRetroP['Chromosome'].value_counts().to_frame()\n",
    "RetroP_count = RetroP_count.rename(columns = {'Chromosome': 'yps1009 RetroP'})\n",
    "RetroP_count_s = LTR_retroP_s288c['Chromosome'].value_counts().to_frame()\n",
    "RetroP_count_s = RetroP_count_s.rename(columns = {'Chromosome': 'S288C RetroP'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "9faf00ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = pd.concat((RetroP_count_s, RetroP_count), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "22744891",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C RetroP</th>\n",
       "      <th>yps1009 RetroP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C RetroP  yps1009 RetroP\n",
       "1            1.0             1.0\n",
       "2            3.0             NaN\n",
       "3            2.0             3.0\n",
       "4            8.0             2.0\n",
       "5            2.0             1.0\n",
       "6            1.0             NaN\n",
       "7            6.0             2.0\n",
       "8            2.0             1.0\n",
       "9            1.0             NaN\n",
       "10           3.0             NaN\n",
       "11           NaN             2.0\n",
       "12           5.0             1.0\n",
       "13           4.0             NaN\n",
       "14           3.0             NaN\n",
       "15           4.0             2.0\n",
       "16           5.0             1.0"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "42baec89",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S288C RetroP      50.0\n",
       "yps1009 RetroP    16.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8d8e2744",
   "metadata": {},
   "source": [
    "# Compared to the filtered ReasonaTE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "3a41edb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "Reasona = pd.read_csv('ReasonaTE_filteredblast.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "17369133",
   "metadata": {},
   "outputs": [],
   "source": [
    "Reasona_count = Reasona['chromosome'].value_counts().to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "bba67f27",
   "metadata": {},
   "outputs": [],
   "source": [
    "Reasona_count.rename(columns={'chromosome': 'ReasonaTE yps RetroP'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "896f75e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = RetroP_count.join(Reasona_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "4c2546f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C RetroP</th>\n",
       "      <th>yps1009 RetroP</th>\n",
       "      <th>ReasonaTE yps RetroP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C RetroP  yps1009 RetroP  ReasonaTE yps RetroP\n",
       "4            8.0             2.0                   4.0\n",
       "7            6.0             2.0                   2.0\n",
       "12           5.0             1.0                   1.0\n",
       "16           5.0             1.0                   1.0\n",
       "13           4.0             NaN                   NaN\n",
       "15           4.0             2.0                   2.0\n",
       "2            3.0             NaN                   1.0\n",
       "10           3.0             NaN                   1.0\n",
       "14           3.0             NaN                   NaN\n",
       "3            2.0             3.0                   1.0\n",
       "5            2.0             1.0                   2.0\n",
       "8            2.0             1.0                   1.0\n",
       "1            1.0             1.0                   2.0\n",
       "6            1.0             NaN                   2.0\n",
       "9            1.0             NaN                   NaN\n",
       "11           NaN             2.0                   NaN"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "id": "c77db124",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S288C RetroP            50.0\n",
       "yps1009 RetroP          16.0\n",
       "ReasonaTE yps RetroP    20.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "id": "d74731da",
   "metadata": {},
   "outputs": [],
   "source": [
    "combiTE = pd.read_csv('TE_liftoffv5_reasonaTE_combined.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "07bff9f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "combi_count = combiTE['chromosome'].value_counts().to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "5bea6700",
   "metadata": {},
   "outputs": [],
   "source": [
    "combi_count.rename(columns={'chromosome': 'Reasona+Liftoff'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "d428e5ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = RetroP_count.join(combi_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "91634294",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C RetroP</th>\n",
       "      <th>yps1009 RetroP</th>\n",
       "      <th>ReasonaTE yps RetroP</th>\n",
       "      <th>Reasona+Liftoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C RetroP  yps1009 RetroP  ReasonaTE yps RetroP  Reasona+Liftoff\n",
       "1            1.0             1.0                   2.0              2.0\n",
       "2            3.0             NaN                   1.0              1.0\n",
       "3            2.0             3.0                   1.0              3.0\n",
       "4            8.0             2.0                   4.0              4.0\n",
       "5            2.0             1.0                   2.0              2.0\n",
       "6            1.0             NaN                   2.0              2.0\n",
       "7            6.0             2.0                   2.0              2.0\n",
       "8            2.0             1.0                   1.0              1.0\n",
       "9            1.0             NaN                   NaN              NaN\n",
       "10           3.0             NaN                   1.0              1.0\n",
       "11           NaN             2.0                   NaN              2.0\n",
       "12           5.0             1.0                   1.0              1.0\n",
       "13           4.0             NaN                   NaN              NaN\n",
       "14           3.0             NaN                   NaN              NaN\n",
       "15           4.0             2.0                   2.0              2.0\n",
       "16           5.0             1.0                   1.0              1.0"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "a13390d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = FEATS.join(RetroP_count[['Reasona+Liftoff']])\n",
    "FEATS = FEATS.rename(columns={'Reasona+Liftoff': 'yps1009 LTR RetroP'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "84eb975c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S288C ARS             352.0\n",
       "yps1009 ARS           358.0\n",
       "S288C ltr             384.0\n",
       "yps1009 ltr           333.0\n",
       "S288C ncRNA            22.0\n",
       "yps1009 ncRNA          21.0\n",
       "S288C snRNA             6.0\n",
       "yps1009 snRNA           6.0\n",
       "yps1009 LTR RetroP     24.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FEATS.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec032b45",
   "metadata": {},
   "source": [
    "# snoRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "1eea9675",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 0\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['overlap id'] = id_overlap\n"
     ]
    }
   ],
   "source": [
    "sno, sno_dupl = overlap_feat(sno)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "7726d381",
   "metadata": {},
   "outputs": [],
   "source": [
    "sno_count = sno['Chromosome'].value_counts().to_frame()\n",
    "sno_count = sno_count.rename(columns = {'Chromosome': 'yps1009 snoRNA'})\n",
    "sno_count_s = sno_s288c['Chromosome'].value_counts().to_frame()\n",
    "sno_count_s = sno_count_s.rename(columns = {'Chromosome': 'S288C snoRNA'})\n",
    "sno_count = sno_count_s.join(sno_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "0d0ff52b",
   "metadata": {},
   "outputs": [],
   "source": [
    "sno_count.to_excel('snoRNA_count_v5.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "3dd78fdf",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = FEATS.join(sno_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530e98aa",
   "metadata": {},
   "source": [
    "# tRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "7816618b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 21\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['overlap id'] = id_overlap\n"
     ]
    }
   ],
   "source": [
    "tRNA, dupl_tRNA = overlap_feat(tRNA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "e2549171",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['sequence_ID'] = df['sequence_ID'].astype(float)\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df.drop(index=ind, inplace=True)\n"
     ]
    }
   ],
   "source": [
    "tRNA = drop_overlap(tRNA, dupl_tRNA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "1328562a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(302, 20)"
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "4962e83e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    278\n",
       "1     24\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "bbef7664",
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_count = tRNA['Chromosome'].value_counts().to_frame()\n",
    "tRNA_count = tRNA_count.rename(columns = {'Chromosome': 'yps1009 tRNA'})\n",
    "tRNA_count_s = tRNA_s288c['Chromosome'].value_counts().to_frame()\n",
    "tRNA_count_s = tRNA_count_s.rename(columns = {'Chromosome': 'S288C tRNA'})\n",
    "tRNA_count = tRNA_count_s.join(tRNA_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "b62c76eb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S288C tRNA      299\n",
       "yps1009 tRNA    302\n",
       "dtype: int64"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA_count.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "bda5f137",
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_count.to_excel('tRNA_count_v5.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "d957dbf5",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = FEATS.join(tRNA_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8effbe52",
   "metadata": {},
   "source": [
    "# GENE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "10cbb4d7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of site annotated several time: 56\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:18: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['overlap id'] = id_overlap\n"
     ]
    }
   ],
   "source": [
    "# First removed ORF that were assigned to 2 different genes,\n",
    "# keep the highest identity \n",
    "gene, gene_dic = overlap_feat(gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "055091fb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['sequence_ID'] = df['sequence_ID'].astype(float)\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\33627071.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df.drop(index=ind, inplace=True)\n"
     ]
    }
   ],
   "source": [
    "gene = drop_overlap(gene, gene_dic)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "c7dc489c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Verified           5217\n",
       "Uncharacterized     717\n",
       "Dubious             699\n",
       "Name: orf_classification, dtype: int64"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene['orf_classification'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "6b91315f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    511\n",
       "0    188\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dubious = gene[gene['orf_classification'] == 'Dubious']\n",
    "dubious['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "fd97eac3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    552\n",
       "0    165\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 108,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uncharacterized = gene[gene['orf_classification'] == 'Uncharacterized']\n",
    "uncharacterized['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "e851651c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_valid = gene[gene['valid_ORFs'] == '1']\n",
    "gene_unvalid = gene[gene['valid_ORFs'] == '0']\n",
    "\n",
    "gene_valid_count = gene_valid['Chromosome'].value_counts().to_frame()\n",
    "gene_valid_count = gene_valid_count.rename(columns={'Chromosome': 'Number valid genes'})\n",
    "gene_unvalid_count = gene_unvalid['Chromosome'].value_counts().to_frame()\n",
    "gene_unvalid_count = gene_unvalid_count.rename(columns={'Chromosome': 'Number unvalid genes'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "bda5a0c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "unval_dic = {}\n",
    "for ind, row in gene_unvalid.iterrows():\n",
    "    if row['Chromosome'] in unval_dic:\n",
    "        unval_dic[row['Chromosome']] = unval_dic[row['Chromosome']] + [row['Name']]\n",
    "    else:\n",
    "        unval_dic[row['Chromosome']] = [row['Name']]\n",
    "        \n",
    "gene_sum = gene_valid_count.join(gene_unvalid_count)\n",
    "tmp = []\n",
    "for ind in gene_sum.index:\n",
    "    tmp.append(unval_dic[ind])\n",
    "gene_sum['unvalid genes'] = tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "5ca2266c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# summing what are the missings genes on chromosome and if moved to other chromosomes\n",
    "missed = []\n",
    "moved = []\n",
    "count_miss = []\n",
    "count_move = []\n",
    "for Ch in gene_sum.index:\n",
    "    s288 = gene_s288c[gene_s288c['Chromosome'] == Ch]\n",
    "    yps = gene[gene['Chromosome'] == Ch]\n",
    "    tmp = list(set(s288['Name']) - set(yps['Name']))\n",
    "    miss = []\n",
    "    mov = []\n",
    "    others = gene[gene['Chromosome'] != Ch]\n",
    "    for g in tmp:\n",
    "        search = others[others['Name'] == g]\n",
    "        if search.shape[0] >= 1:\n",
    "            mov.append((g, list(search['Chromosome']), list(search['valid_ORFs'])))\n",
    "        else:\n",
    "            miss.append(g)  \n",
    "    moved.append(mov)\n",
    "    missed.append(miss)\n",
    "    count_miss.append(len(miss))\n",
    "    count_move.append(len(mov))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "701aa46a",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_sum['Number of missing genes'] = count_miss\n",
    "gene_sum['Missing genes'] = missed\n",
    "gene_sum['Number of moved genes'] = count_move\n",
    "gene_sum['Moved genes, new chromosome(s), orf validity'] = moved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "id": "8ffefac7",
   "metadata": {},
   "outputs": [],
   "source": [
    "flat_list = [item for sublist in missed for item in sublist]\n",
    "missing_genes = gene_s288c[gene_s288c['Name'].isin(flat_list)]\n",
    "missing_genes.to_excel('missing_genes_v5.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "d88437e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Blast\n",
    "ids = []\n",
    "seq = []\n",
    "for seq_record in SeqIO.parse(\"fasta_file/orf_coding_all_R64-3-1_20210421.fasta\", \"fasta\"):\n",
    "    ids.append(str(seq_record.id))\n",
    "    seq.append(str(seq_record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f8392e51",
   "metadata": {},
   "outputs": [],
   "source": [
    "for g in flat_list:\n",
    "    name = g + '.txt'\n",
    "    with open(name , \"w\") as file:\n",
    "        file.write('> ' + g + '\\n')\n",
    "        file.write(seq[ids.index(g)])\n",
    "    outfile = 'Blast_results/' + g + '.tab'\n",
    "    cmd_blastn = NcbiblastnCommandline(query= name, out= outfile, evalue=0.001, db= 'YPS1009_DB/YPS1009_DB')\n",
    "    stdout, stderr = cmd_blastn()\n",
    "    #print(\"STDOUT: %s\" % stdout)\n",
    "    #print(\"STDERR: %s\" % stderr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "8fbc4a84",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "89"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Used to be 42 missing genes, but after removing genes mapped to the same ORF, that's more\n",
    "# Meaning that the 47 extra missing are copies of genes mapped and can therefore be ignored.\n",
    "len(flat_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8b062b2",
   "metadata": {},
   "source": [
    "# Verified ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "faddeacc",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified = gene[gene['orf_classification'] == 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "7b49021f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    5060\n",
       "0     157\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 117,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "bd6300ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_s288c = gene_s288c[gene_s288c['orf_classification'] == 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "d019b4f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5216, 14)"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_s288c.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "6a766625",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid = verified[verified['valid_ORFs'] == '1']\n",
    "verified_unvalid = verified[verified['valid_ORFs'] == '0']\n",
    "\n",
    "verified_valid_count = verified_valid['Chromosome'].value_counts().to_frame()\n",
    "verified_valid_count = verified_valid_count.rename(columns={'Chromosome': 'Number valid genes'})\n",
    "verified_unvalid_count = verified_unvalid['Chromosome'].value_counts().to_frame()\n",
    "verified_unvalid_count = verified_unvalid_count.rename(columns={'Chromosome': 'Number unvalid genes'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "id": "32c1a04f",
   "metadata": {},
   "outputs": [],
   "source": [
    "unval_dic = {}\n",
    "for ind, row in verified_unvalid.iterrows():\n",
    "    if row['Chromosome'] in unval_dic:\n",
    "        unval_dic[row['Chromosome']] = unval_dic[row['Chromosome']] + [row['Name']]\n",
    "    else:\n",
    "        unval_dic[row['Chromosome']] = [row['Name']]    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "63ab8ca2",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum = verified_valid_count.join(verified_unvalid_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "52c99402",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = []\n",
    "for ind in verified_sum.index:\n",
    "    tmp.append(unval_dic[ind])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "ec34117a",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum['unvalid genes'] = tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "e7ad4015",
   "metadata": {},
   "outputs": [],
   "source": [
    "missed = []\n",
    "moved = []\n",
    "count_miss = []\n",
    "count_move = []\n",
    "for Ch in verified_sum.index:\n",
    "    s288 = verified_s288c[verified_s288c['Chromosome'] == Ch]\n",
    "    yps = verified[verified['Chromosome'] == Ch]\n",
    "    tmp = list(set(s288['Name']) - set(yps['Name']))\n",
    "    miss = []\n",
    "    mov = []\n",
    "    others = verified[verified['Chromosome'] != Ch]\n",
    "    for g in tmp:\n",
    "        search = others[others['Name'] == g]\n",
    "        if search.shape[0] >= 1:\n",
    "            mov.append((g, list(search['Chromosome']), list(search['valid_ORFs'])))\n",
    "        else:\n",
    "            miss.append(g)  \n",
    "    moved.append(mov)\n",
    "    missed.append(miss)\n",
    "    count_miss.append(len(miss))\n",
    "    count_move.append(len(mov))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "id": "0e731ec9",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum['Number of missing genes'] = count_miss\n",
    "verified_sum['Missing genes'] = missed\n",
    "verified_sum['Number of moved genes'] = count_move\n",
    "verified_sum['Moved genes, new chromosome(s), orf validity'] = moved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "9ad232b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum.to_excel('Liftoff_v5_verified_genes_summary.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a1a07a6",
   "metadata": {},
   "source": [
    "# Verified missing genes dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "146d74ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "flat_list = [item for sublist in missed for item in sublist]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "b9fa3baf",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_genes = verified_s288c[verified_s288c['Name'].isin(flat_list)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "d93156f6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(50, 14)"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_genes.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "2532382c",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_genes.to_excel('missing_verified_genes_v5.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c6a1c7b",
   "metadata": {},
   "source": [
    "# Duplicated genes\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "c130b50c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    6507\n",
       "1      91\n",
       "2      19\n",
       "3      10\n",
       "4       3\n",
       "5       2\n",
       "6       1\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Many duplicated genes in this version\n",
    "gene['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "bda8b613",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>...</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "      <th>overlap id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "      <td>chrII-2_tig000049271404014700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>1323</td>\n",
       "      <td>2063</td>\n",
       "      <td>YCL076W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "      <td>chrIII-2_tig0000493510202060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>6426</td>\n",
       "      <td>8271</td>\n",
       "      <td>YCL073C</td>\n",
       "      <td>GEX1,glutathione%20exchanger</td>\n",
       "      <td>Proton:glutathione antiporter%3B localized to ...</td>\n",
       "      <td>GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>GEX1</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.990</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "      <td>chrIII-2_tig0000493504208270</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>1316</td>\n",
       "      <td>2055</td>\n",
       "      <td>YCL076W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "      <td>chrIII_tig0000493613102000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>770</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>310848</td>\n",
       "      <td>311102</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "      <td>chrIII_tig00004936310040311100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2290</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>789332</td>\n",
       "      <td>789467</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.890</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00000010</td>\n",
       "      <td>chrII_tig00000010789330089460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2313</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1219</td>\n",
       "      <td>1356</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "      <td>chrIV_tig0000000112101350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5183</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>3329</td>\n",
       "      <td>3525</td>\n",
       "      <td>YCR102W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "      <td>chrIX_tig0000002333203020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6045</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>676</td>\n",
       "      <td>975</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "      <td>chrI_tig00000033070970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6506</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>238815</td>\n",
       "      <td>239136</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.867</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "      <td>chrI_tig00000033238810239130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6778</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>737</td>\n",
       "      <td>1036</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "      <td>chrVII-2_tig000000380301030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8550</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>522253</td>\n",
       "      <td>522633</td>\n",
       "      <td>YAL064W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Fungal-specific protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005783,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.934</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "      <td>chrVIII_tig00000017522250522600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10168</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>805604</td>\n",
       "      <td>806749</td>\n",
       "      <td>YGR295C</td>\n",
       "      <td>COS6</td>\n",
       "      <td>Endosomal protein involved in turnover of plas...</td>\n",
       "      <td>GO:0000324,GO:0000324,GO:0003674,GO:0005768,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>COS6</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.961</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "      <td>chrVII_tig00004928805600806740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10171</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>807356</td>\n",
       "      <td>807492</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "      <td>chrVII_tig00004928807350807490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10179</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>807660</td>\n",
       "      <td>807956</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.930</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "      <td>chrVII_tig00004928807660807950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10707</th>\n",
       "      <td>chrVI_tig00004938</td>\n",
       "      <td>298356</td>\n",
       "      <td>298655</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>tig00004938</td>\n",
       "      <td>chrVI_tig00004938298350298600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12156</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>642</td>\n",
       "      <td>835</td>\n",
       "      <td>YCR108C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "      <td>chrX-2_tig00001099640830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12167</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>4754</td>\n",
       "      <td>5859</td>\n",
       "      <td>YCR102C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative quinone oxidoreductase%3B overexpress...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "      <td>chrX-2_tig0000109907505850</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12170</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>7160</td>\n",
       "      <td>11768</td>\n",
       "      <td>YJL222W</td>\n",
       "      <td>VTH2,signal%20sequence-binding%20protein</td>\n",
       "      <td>Putative membrane glycoprotein%3B has strong s...</td>\n",
       "      <td>GO:0000324,GO:0005048,GO:0005048,GO:0006623,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>VTH2</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.972</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "      <td>chrX-2_tig00001099716011760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12174</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>12417</td>\n",
       "      <td>14186</td>\n",
       "      <td>YIL172C</td>\n",
       "      <td>IMA3,oligo-1%2C6-glucosidase%20IMA3</td>\n",
       "      <td>Alpha-glucosidase%3B weak%2C but broad substra...</td>\n",
       "      <td>GO:0004574,GO:0004574,GO:0004575,GO:0005737,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>IMA3</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "      <td>chrX-2_tig000010991241014180</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12202</th>\n",
       "      <td>chrXI-2_tig00004931</td>\n",
       "      <td>21483</td>\n",
       "      <td>23330</td>\n",
       "      <td>YCL073C</td>\n",
       "      <td>GEX1,glutathione%20exchanger</td>\n",
       "      <td>Proton:glutathione antiporter%3B localized to ...</td>\n",
       "      <td>GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>GEX1</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.991</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004931</td>\n",
       "      <td>chrXI-2_tig000049312148023330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12223</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>699</td>\n",
       "      <td>999</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.926</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "      <td>chrXII-2_tig00004934600000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13078</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>6551</td>\n",
       "      <td>6679</td>\n",
       "      <td>YBL108C-A</td>\n",
       "      <td>PAU9,seripauperin%20PAU9</td>\n",
       "      <td>Protein of unknown function%3B member of the s...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>PAU9</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004933</td>\n",
       "      <td>chrXII-3_tig0000493365506670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13081</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>7121</td>\n",
       "      <td>7351</td>\n",
       "      <td>YGL260W</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B transc...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.948</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004933</td>\n",
       "      <td>chrXII-3_tig0000493370207350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13118</th>\n",
       "      <td>chrXIII-2_tig00000029</td>\n",
       "      <td>1228</td>\n",
       "      <td>1363</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.948</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000029</td>\n",
       "      <td>chrXIII-2_tig0000002912201060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14840</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>467347</td>\n",
       "      <td>467484</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000018</td>\n",
       "      <td>chrXIII_tig00000018460340067080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14851</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>467923</td>\n",
       "      <td>468106</td>\n",
       "      <td>YOR394C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%3B identified by g...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.822</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000018</td>\n",
       "      <td>chrXIII_tig00000018467920408100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16029</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>554706</td>\n",
       "      <td>554903</td>\n",
       "      <td>YCR102W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.954</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00000020</td>\n",
       "      <td>chrXII_tig00000020554700554900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16047</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>557160</td>\n",
       "      <td>557459</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00000020</td>\n",
       "      <td>chrXII_tig00000020557160557450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17486</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>726</td>\n",
       "      <td>980</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004932</td>\n",
       "      <td>chrXI_tig00004932720980</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19203</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>743</td>\n",
       "      <td>997</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.941</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000040</td>\n",
       "      <td>chrXVI-2_tig00000040740990</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20927</th>\n",
       "      <td>chrXVI_tig00000013</td>\n",
       "      <td>719024</td>\n",
       "      <td>719337</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.895</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000013</td>\n",
       "      <td>chrXVI_tig00000013719020019330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20995</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>24888</td>\n",
       "      <td>25351</td>\n",
       "      <td>YIL171W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "      <td>chrXV_tig000000072400025350</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22468</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>4721</td>\n",
       "      <td>5826</td>\n",
       "      <td>YCR102C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative quinone oxidoreductase%3B overexpress...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "      <td>chrX_tig0000109847205820</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22481</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>13824</td>\n",
       "      <td>14287</td>\n",
       "      <td>YIL171W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.952</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "      <td>chrX_tig000010981382014280</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35 rows Ã— 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      seq_id   start     end       Name  \\\n",
       "9        chrII-2_tig00004927   14242   14700    YGR293C   \n",
       "39      chrIII-2_tig00004935    1323    2063    YCL076W   \n",
       "45      chrIII-2_tig00004935    6426    8271    YCL073C   \n",
       "59        chrIII_tig00004936    1316    2055    YCL076W   \n",
       "770       chrIII_tig00004936  310848  311102    YNL337W   \n",
       "2290       chrII_tig00000010  789332  789467  YHL048C-A   \n",
       "2313       chrIV_tig00000001    1219    1356  YHL048C-A   \n",
       "5183       chrIX_tig00000023    3329    3525  YCR102W-A   \n",
       "6045        chrI_tig00000033     676     975    YDR543C   \n",
       "6506        chrI_tig00000033  238815  239136    YAL069W   \n",
       "6778    chrVII-2_tig00000038     737    1036    YDR543C   \n",
       "8550     chrVIII_tig00000017  522253  522633  YAL064W-B   \n",
       "10168     chrVII_tig00004928  805604  806749    YGR295C   \n",
       "10171     chrVII_tig00004928  807356  807492  YHL048C-A   \n",
       "10179     chrVII_tig00004928  807660  807956  YER188C-A   \n",
       "10707      chrVI_tig00004938  298356  298655    YDR543C   \n",
       "12156     chrX-2_tig00001099     642     835    YCR108C   \n",
       "12167     chrX-2_tig00001099    4754    5859    YCR102C   \n",
       "12170     chrX-2_tig00001099    7160   11768    YJL222W   \n",
       "12174     chrX-2_tig00001099   12417   14186    YIL172C   \n",
       "12202    chrXI-2_tig00004931   21483   23330    YCL073C   \n",
       "12223   chrXII-2_tig00004934     699     999    YDR543C   \n",
       "13078   chrXII-3_tig00004933    6551    6679  YBL108C-A   \n",
       "13081   chrXII-3_tig00004933    7121    7351    YGL260W   \n",
       "13118  chrXIII-2_tig00000029    1228    1363  YHL048C-A   \n",
       "14840    chrXIII_tig00000018  467347  467484  YHL048C-A   \n",
       "14851    chrXIII_tig00000018  467923  468106  YOR394C-A   \n",
       "16029     chrXII_tig00000020  554706  554903  YCR102W-A   \n",
       "16047     chrXII_tig00000020  557160  557459  YER188C-A   \n",
       "17486      chrXI_tig00004932     726     980    YNL337W   \n",
       "19203   chrXVI-2_tig00000040     743     997    YNL337W   \n",
       "20927     chrXVI_tig00000013  719024  719337    YAL069W   \n",
       "20995      chrXV_tig00000007   24888   25351  YIL171W-A   \n",
       "22468       chrX_tig00001098    4721    5826    YCR102C   \n",
       "22481       chrX_tig00001098   13824   14287  YIL171W-A   \n",
       "\n",
       "                                          Alias  \\\n",
       "9                                          None   \n",
       "39                                         None   \n",
       "45                 GEX1,glutathione%20exchanger   \n",
       "59                                         None   \n",
       "770                                        None   \n",
       "2290                                       None   \n",
       "2313                                       None   \n",
       "5183                                       None   \n",
       "6045                                       None   \n",
       "6506                                       None   \n",
       "6778                                       None   \n",
       "8550                                       None   \n",
       "10168                                      COS6   \n",
       "10171                                      None   \n",
       "10179                                      None   \n",
       "10707                                      None   \n",
       "12156                                      None   \n",
       "12167                                      None   \n",
       "12170  VTH2,signal%20sequence-binding%20protein   \n",
       "12174       IMA3,oligo-1%2C6-glucosidase%20IMA3   \n",
       "12202              GEX1,glutathione%20exchanger   \n",
       "12223                                      None   \n",
       "13078                  PAU9,seripauperin%20PAU9   \n",
       "13081                                      None   \n",
       "13118                                      None   \n",
       "14840                                      None   \n",
       "14851                                      None   \n",
       "16029                                      None   \n",
       "16047                                      None   \n",
       "17486                                      None   \n",
       "19203                                      None   \n",
       "20927                                      None   \n",
       "20995                                      None   \n",
       "22468                                      None   \n",
       "22481                                      None   \n",
       "\n",
       "                                                    Note  \\\n",
       "9      Dubious open reading frame%3B unlikely to enco...   \n",
       "39     Dubious open reading frame%3B unlikely to enco...   \n",
       "45     Proton:glutathione antiporter%3B localized to ...   \n",
       "59     Dubious open reading frame%3B unlikely to enco...   \n",
       "770    Dubious open reading frame%3B unlikely to enco...   \n",
       "2290   Putative protein of unknown function%3B identi...   \n",
       "2313   Putative protein of unknown function%3B identi...   \n",
       "5183   Dubious open reading frame%3B unlikely to enco...   \n",
       "6045   Dubious open reading frame%3B unlikely to enco...   \n",
       "6506   Dubious open reading frame%3B unlikely to enco...   \n",
       "6778   Dubious open reading frame%3B unlikely to enco...   \n",
       "8550         Fungal-specific protein of unknown function   \n",
       "10168  Endosomal protein involved in turnover of plas...   \n",
       "10171  Putative protein of unknown function%3B identi...   \n",
       "10179               Putative protein of unknown function   \n",
       "10707  Dubious open reading frame%3B unlikely to enco...   \n",
       "12156  Putative protein of unknown function%3B identi...   \n",
       "12167  Putative quinone oxidoreductase%3B overexpress...   \n",
       "12170  Putative membrane glycoprotein%3B has strong s...   \n",
       "12174  Alpha-glucosidase%3B weak%2C but broad substra...   \n",
       "12202  Proton:glutathione antiporter%3B localized to ...   \n",
       "12223  Dubious open reading frame%3B unlikely to enco...   \n",
       "13078  Protein of unknown function%3B member of the s...   \n",
       "13081  Putative protein of unknown function%3B transc...   \n",
       "13118  Putative protein of unknown function%3B identi...   \n",
       "14840  Putative protein of unknown function%3B identi...   \n",
       "14851  Protein of unknown function%3B identified by g...   \n",
       "16029  Dubious open reading frame%3B unlikely to enco...   \n",
       "16047               Putative protein of unknown function   \n",
       "17486  Dubious open reading frame%3B unlikely to enco...   \n",
       "19203  Dubious open reading frame%3B unlikely to enco...   \n",
       "20927  Dubious open reading frame%3B unlikely to enco...   \n",
       "20995  Dubious open reading frame%3B unlikely to enco...   \n",
       "22468  Putative quinone oxidoreductase%3B overexpress...   \n",
       "22481  Dubious open reading frame%3B unlikely to enco...   \n",
       "\n",
       "                                           Ontology_term score strand phase  \\\n",
       "9            GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "39           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "45     GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...     .      -     .   \n",
       "59           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "770          GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "2290         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "2313         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "5183         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "6045         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6506         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "6778         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "8550         GO:0003674,GO:0005783,GO:0008150,SO:0000704     .      -     .   \n",
       "10168  GO:0000324,GO:0000324,GO:0003674,GO:0005768,GO...     .      -     .   \n",
       "10171        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "10179        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "10707        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "12156        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "12167  GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...     .      +     .   \n",
       "12170  GO:0000324,GO:0005048,GO:0005048,GO:0006623,GO...     .      +     .   \n",
       "12174  GO:0004574,GO:0004574,GO:0004575,GO:0005737,GO...     .      -     .   \n",
       "12202  GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...     .      +     .   \n",
       "12223        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "13078        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "13081        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "13118        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "14840        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "14851        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16029        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "16047        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "17486        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "19203        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20927        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "20995        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "22468  GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...     .      +     .   \n",
       "22481        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "\n",
       "       ... extra_copy_number  gene low_identity orf_classification  \\\n",
       "9      ...                 2  None         None            Dubious   \n",
       "39     ...                 2  None         None            Dubious   \n",
       "45     ...                 3  GEX1         None           Verified   \n",
       "59     ...                 3  None         None            Dubious   \n",
       "770    ...                 4  None         None            Dubious   \n",
       "2290   ...                 6  None         None    Uncharacterized   \n",
       "2313   ...                 4  None         None    Uncharacterized   \n",
       "5183   ...                 3  None         None            Dubious   \n",
       "6045   ...                 3  None         None            Dubious   \n",
       "6506   ...                 2  None         None            Dubious   \n",
       "6778   ...                 2  None         None            Dubious   \n",
       "8550   ...                 2  None         None    Uncharacterized   \n",
       "10168  ...                 2  COS6         None           Verified   \n",
       "10171  ...                 5  None         None    Uncharacterized   \n",
       "10179  ...                 3  None         None    Uncharacterized   \n",
       "10707  ...                 4  None         None            Dubious   \n",
       "12156  ...                 2  None         None    Uncharacterized   \n",
       "12167  ...                 2  None         None    Uncharacterized   \n",
       "12170  ...                 2  VTH2         None           Verified   \n",
       "12174  ...                 2  IMA3         None           Verified   \n",
       "12202  ...                 2  GEX1         None           Verified   \n",
       "12223  ...                 5  None         None            Dubious   \n",
       "13078  ...                 2  PAU9         None           Verified   \n",
       "13081  ...                 2  None         None    Uncharacterized   \n",
       "13118  ...                 2  None         None    Uncharacterized   \n",
       "14840  ...                 3  None         None    Uncharacterized   \n",
       "14851  ...                 2  None         None    Uncharacterized   \n",
       "16029  ...                 2  None         None            Dubious   \n",
       "16047  ...                 2  None         None    Uncharacterized   \n",
       "17486  ...                 2  None         None            Dubious   \n",
       "19203  ...                 3  None         None            Dubious   \n",
       "20927  ...                 3  None         None            Dubious   \n",
       "20995  ...                 2  None         None            Dubious   \n",
       "22468  ...                 3  None         None    Uncharacterized   \n",
       "22481  ...                 3  None         None            Dubious   \n",
       "\n",
       "      partial_mapping sequence_ID valid_ORFs Chromosome       Contig  \\\n",
       "9                None       0.989          0          2  tig00004927   \n",
       "39               None       0.981          0          3  tig00004935   \n",
       "45               None       0.990          0          3  tig00004935   \n",
       "59               None       0.979          0          3  tig00004936   \n",
       "770              None       0.949          1          3  tig00004936   \n",
       "2290             None       0.890          0          2  tig00000010   \n",
       "2313             None       0.920          0          4  tig00000001   \n",
       "5183             None       0.949          0          9  tig00000023   \n",
       "6045             None       0.946          0          1  tig00000033   \n",
       "6506             None       0.867          0          1  tig00000033   \n",
       "6778             None       0.936          0          7  tig00000038   \n",
       "8550             None       0.934          1          8  tig00000017   \n",
       "10168            None       0.961          1          7  tig00004928   \n",
       "10171            None       0.927          0          7  tig00004928   \n",
       "10179            None       0.930          1          7  tig00004928   \n",
       "10707            None       0.936          0          6  tig00004938   \n",
       "12156            None       0.917          0         10  tig00001099   \n",
       "12167            None       0.981          0         10  tig00001099   \n",
       "12170            None       0.972          0         10  tig00001099   \n",
       "12174            None       0.989          1         10  tig00001099   \n",
       "12202            None       0.991          1         11  tig00004931   \n",
       "12223            None       0.926          0         12  tig00004934   \n",
       "13078            None       1.000          1         12  tig00004933   \n",
       "13081            None       0.948          1         12  tig00004933   \n",
       "13118            None       0.948          0         13  tig00000029   \n",
       "14840            None       0.920          0         13  tig00000018   \n",
       "14851            None       0.822          0         13  tig00000018   \n",
       "16029            None       0.954          0         12  tig00000020   \n",
       "16047            None       0.956          1         12  tig00000020   \n",
       "17486            None       0.956          1         11  tig00004932   \n",
       "19203            None       0.941          1         16  tig00000040   \n",
       "20927            None       0.895          0         16  tig00000013   \n",
       "20995            None       0.956          0         15  tig00000007   \n",
       "22468            None       0.979          0         10  tig00001098   \n",
       "22481            None       0.952          0         10  tig00001098   \n",
       "\n",
       "                            overlap id  \n",
       "9        chrII-2_tig000049271404014700  \n",
       "39        chrIII-2_tig0000493510202060  \n",
       "45        chrIII-2_tig0000493504208270  \n",
       "59          chrIII_tig0000493613102000  \n",
       "770     chrIII_tig00004936310040311100  \n",
       "2290     chrII_tig00000010789330089460  \n",
       "2313         chrIV_tig0000000112101350  \n",
       "5183         chrIX_tig0000002333203020  \n",
       "6045            chrI_tig00000033070970  \n",
       "6506      chrI_tig00000033238810239130  \n",
       "6778       chrVII-2_tig000000380301030  \n",
       "8550   chrVIII_tig00000017522250522600  \n",
       "10168   chrVII_tig00004928805600806740  \n",
       "10171   chrVII_tig00004928807350807490  \n",
       "10179   chrVII_tig00004928807660807950  \n",
       "10707    chrVI_tig00004938298350298600  \n",
       "12156         chrX-2_tig00001099640830  \n",
       "12167       chrX-2_tig0000109907505850  \n",
       "12170      chrX-2_tig00001099716011760  \n",
       "12174     chrX-2_tig000010991241014180  \n",
       "12202    chrXI-2_tig000049312148023330  \n",
       "12223       chrXII-2_tig00004934600000  \n",
       "13078     chrXII-3_tig0000493365506670  \n",
       "13081     chrXII-3_tig0000493370207350  \n",
       "13118    chrXIII-2_tig0000002912201060  \n",
       "14840  chrXIII_tig00000018460340067080  \n",
       "14851  chrXIII_tig00000018467920408100  \n",
       "16029   chrXII_tig00000020554700554900  \n",
       "16047   chrXII_tig00000020557160557450  \n",
       "17486          chrXI_tig00004932720980  \n",
       "19203       chrXVI-2_tig00000040740990  \n",
       "20927   chrXVI_tig00000013719020019330  \n",
       "20995      chrXV_tig000000072400025350  \n",
       "22468         chrX_tig0000109847205820  \n",
       "22481       chrX_tig000010981382014280  \n",
       "\n",
       "[35 rows x 24 columns]"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[gene['extra_copy_number'] > '1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "id": "73d236f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl = gene[gene.duplicated(subset=['Name'], keep=False) == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "id": "f1876fa9",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl = dupl.sort_values(by = 'Name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "1f204f21",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl.to_excel('duplicated_verified_genes.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d155e624",
   "metadata": {},
   "source": [
    "Duplicated genes:\n",
    "\n",
    "ATP8/Q0080: Subunit 8 of the F0 sector of mitochondrial F1F0 ATP synthase; encoded on the mitochondrial genome; ATP8 and ATP6 mRNAs are not translated in the absence of the F1 sector of ATPase\n",
    "\n",
    "OLI1/Q0130: F0-ATP synthase subunit c (ATPase-associated proteolipid); encoded on the mitochondrial genome; mutation confers oligomycin resistance; expression is specifically dependent on the nuclear genes AEP1 and AEP2 1 2 3 4\n",
    "Name Description\n",
    "OLIgomycin resistance 1\n",
    "\n",
    "Both on mitochondria so not that relevant to my project.\n",
    "\n",
    "PAU9/YBL108C-A: Protein of unknown function; member of the seripauperin multigene family encoded mainly in subtelomeric regions; SWAT-GFP and mCherry fusion proteins localize to the endoplasmic reticulum and vacuole respectively\n",
    "\n",
    "YHR213W-A: Putative protein of unknown function; identified by gene-trapping, microarray-based expression analysis, and genome-wide homology searching "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e0c10ee",
   "metadata": {},
   "source": [
    "# Analyse unvalid genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82bdf79b",
   "metadata": {},
   "source": [
    "6 partial mapping genes:\n",
    "    YDL227C near telomere\n",
    "    YMR106C middle of chromosome arm\n",
    "    YNL322C near telomere and missing genes, suggesting GAP but could also be a fragment deletion.\n",
    "    YOR244W middle of chromosome arm\n",
    "    YOR245C middle of chromosome arm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64b75c73",
   "metadata": {},
   "source": [
    "### merge mRNA and gene data to get information on the type of issue in ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "3ad463f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'ID', 'Name', 'Parent', 'conditions', 'dbxref',\n",
       "       'extra_copy_number', 'inframe_stop_codon', 'matches_ref_protein',\n",
       "       'missing_start_codon', 'missing_stop_codon', 'transcript_id',\n",
       "       'valid_ORF'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mRNA_df = annotation.filter_feature_of_type(['mRNA'])\n",
    "mRNA_attr = mRNA_df.attributes_to_columns()\n",
    "mRNA_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "id": "2c97f343",
   "metadata": {},
   "outputs": [],
   "source": [
    "mRNA = mRNA_attr[['seq_id', 'start', 'end', 'ID', 'Name', 'Parent',\n",
    "                  'conditions', 'extra_copy_number', 'inframe_stop_codon', \n",
    "                  'matches_ref_protein', 'missing_start_codon', 'missing_stop_codon', \n",
    "                  'transcript_id', 'valid_ORF']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "bb1021da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'Ontology_term',\n",
       "       'copy_num_ID', 'coverage', 'curie', 'dbxref', 'display',\n",
       "       'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
       "       'partial_mapping', 'sequence_ID', 'valid_ORFs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "ad003156",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene2 = gene_attr[['seq_id', 'start', 'end',\n",
    "       'Alias', 'ID', 'Name', 'copy_num_ID', 'coverage', 'display',\n",
    "       'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
    "       'partial_mapping', 'sequence_ID', 'valid_ORFs']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "2cbd0c78",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Alias</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>MAL31,MAL3T,MALT,maltose%20permease</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>YBR298C</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltose%20permease</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL31</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>None</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>YBR298C-A</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>0.990</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>YBR299W</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltase%20%28alpha-D-glucosidase%29</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL32</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>None</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>0.993</td>\n",
       "      <td>Dubious%20open%20reading%20frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>PAU12,seripauperin%20PAU12</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>YGR294W</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>PAU12</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seq_id  start    end  \\\n",
       "0   chrII-2_tig00004927   8297  10119   \n",
       "3   chrII-2_tig00004927  10675  10894   \n",
       "6   chrII-2_tig00004927  10985  12717   \n",
       "9   chrII-2_tig00004927  14242  14700   \n",
       "12  chrII-2_tig00004927  14660  15022   \n",
       "\n",
       "                                         Alias           ID       Name  \\\n",
       "0          MAL31,MAL3T,MALT,maltose%20permease    YBR298C_1    YBR298C   \n",
       "3                                         None  YBR298C-A_1  YBR298C-A   \n",
       "6   MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32    YBR299W_1    YBR299W   \n",
       "9                                         None    YGR293C_2    YGR293C   \n",
       "12                  PAU12,seripauperin%20PAU12    YGR294W_1    YGR294W   \n",
       "\n",
       "    copy_num_ID coverage                                       display  \\\n",
       "0     YBR298C_1    0.986                            Maltose%20permease   \n",
       "3   YBR298C-A_1    0.990  Putative%20protein%20of%20unknown%20function   \n",
       "6     YBR299W_1    0.986           Maltase%20%28alpha-D-glucosidase%29   \n",
       "9     YGR293C_2    0.993              Dubious%20open%20reading%20frame   \n",
       "12    YGR294W_1      1.0             Protein%20of%20unknown%20function   \n",
       "\n",
       "   extra_copy_number   gene low_identity orf_classification partial_mapping  \\\n",
       "0                  1  MAL31         None           Verified            None   \n",
       "3                  1   None         None    Uncharacterized            None   \n",
       "6                  1  MAL32         None           Verified            None   \n",
       "9                  2   None         None            Dubious            None   \n",
       "12                 1  PAU12         None           Verified            None   \n",
       "\n",
       "   sequence_ID valid_ORFs  \n",
       "0        0.946          0  \n",
       "3        0.981          0  \n",
       "6        0.981          0  \n",
       "9        0.989          0  \n",
       "12       0.983          1  "
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "c22a435a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Parent</th>\n",
       "      <th>conditions</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>inframe_stop_codon</th>\n",
       "      <th>matches_ref_protein</th>\n",
       "      <th>missing_start_codon</th>\n",
       "      <th>missing_stop_codon</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>valid_ORF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>YBR298C_id001_1</td>\n",
       "      <td>YBR298C_id001</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288108_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>YBR298C-A_mRNA_1</td>\n",
       "      <td>YBR298C-A_mRNA</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>RefSeq:NM_001184588.1_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>YBR299W_id001_1</td>\n",
       "      <td>YBR299W_id001</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288109_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>YGR293C_mRNA_2</td>\n",
       "      <td>YGR293C_mRNA</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>YGR294W_id001_1</td>\n",
       "      <td>YGR294W_id001</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289696_1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seq_id  start    end                ID            Name  \\\n",
       "1   chrII-2_tig00004927   8297  10119   YBR298C_id001_1   YBR298C_id001   \n",
       "4   chrII-2_tig00004927  10675  10894  YBR298C-A_mRNA_1  YBR298C-A_mRNA   \n",
       "7   chrII-2_tig00004927  10985  12717   YBR299W_id001_1   YBR299W_id001   \n",
       "10  chrII-2_tig00004927  14242  14700    YGR293C_mRNA_2    YGR293C_mRNA   \n",
       "13  chrII-2_tig00004927  14660  15022   YGR294W_id001_1   YGR294W_id001   \n",
       "\n",
       "         Parent conditions extra_copy_number inframe_stop_codon  \\\n",
       "1     YBR298C_1        GAL                 1               None   \n",
       "4   YBR298C-A_1       None                 1               None   \n",
       "7     YBR299W_1        GAL                 1               None   \n",
       "10    YGR293C_2       None                 2               True   \n",
       "13    YGR294W_1        GAL                 1               None   \n",
       "\n",
       "   matches_ref_protein missing_start_codon missing_stop_codon  \\\n",
       "1                False                None               True   \n",
       "4                False                None               True   \n",
       "7                False                None               True   \n",
       "10               False                None               None   \n",
       "13                True                None               None   \n",
       "\n",
       "              transcript_id valid_ORF  \n",
       "1          SGD:S000288108_1     False  \n",
       "4   RefSeq:NM_001184588.1_1     False  \n",
       "7          SGD:S000288109_1     False  \n",
       "10                     None     False  \n",
       "13         SGD:S000289696_1      True  "
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mRNA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "28cc6ac2",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA = pd.merge(gene2, mRNA,  how='right', left_on='ID', right_on = 'Parent')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "2c098e59",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     6167\n",
       "False     524\n",
       "Name: valid_ORF, dtype: int64"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA['valid_ORF'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "085f85aa",
   "metadata": {},
   "source": [
    "## Inframe stop codon analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "5c996414",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\4266080126.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  inframe_288c['lenght'] = inframe_288c['end'] - inframe_288c['start']\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\4266080126.py:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  inframe_stop['lenght'] = inframe_stop['end_x'] - inframe_stop['start_x']\n"
     ]
    }
   ],
   "source": [
    "inframe_stop = gene_mRNA[gene_mRNA['inframe_stop_codon'] == 'True']\n",
    "inframe_g = inframe_stop['Name_x']\n",
    "inframe_288c = gene_s288c[gene_attr_s288c['Name'].isin(inframe_g)]\n",
    "inframe_288c['lenght'] = inframe_288c['end'] - inframe_288c['start']\n",
    "inframe_stop['lenght'] = inframe_stop['end_x'] - inframe_stop['start_x']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "d46d13f7",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframe_stop = inframe_stop[['Name_x','copy_num_ID', 'coverage', 'lenght', 'orf_classification']]\n",
    "inframe_288c = inframe_288c[['Name', 'lenght']]\n",
    "\n",
    "inframes_comp = pd.merge(inframe_288c, inframe_stop,  how='right', left_on='Name', right_on = 'Name_x')\n",
    "inframes_comp['diff'] = inframes_comp['lenght_x'] - inframes_comp['lenght_y']\n",
    "\n",
    "inframes_comp = inframes_comp.sort_values(by='diff')\n",
    "inframes_comp = inframes_comp[['Name', 'lenght_x', 'lenght_y', 'diff']]\n",
    "inframes_comp.rename(columns = {'lenght_x': 'length s288c gene', 'lenght_y': 'length yps gene'}, inplace = True)\n",
    "inframes_comp.to_excel('inframes.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "49d2c172",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>length s288c gene</th>\n",
       "      <th>length yps gene</th>\n",
       "      <th>diff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>YKR102W</td>\n",
       "      <td>3509.0</td>\n",
       "      <td>4118</td>\n",
       "      <td>-609.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>YPR012W</td>\n",
       "      <td>254.0</td>\n",
       "      <td>281</td>\n",
       "      <td>-27.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>YBR134W</td>\n",
       "      <td>401.0</td>\n",
       "      <td>410</td>\n",
       "      <td>-9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>YNL269W</td>\n",
       "      <td>395.0</td>\n",
       "      <td>401</td>\n",
       "      <td>-6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>YNL205C</td>\n",
       "      <td>422.0</td>\n",
       "      <td>425</td>\n",
       "      <td>-3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Q0105</td>\n",
       "      <td>7107.0</td>\n",
       "      <td>7068</td>\n",
       "      <td>39.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>Q0105</td>\n",
       "      <td>7107.0</td>\n",
       "      <td>7059</td>\n",
       "      <td>48.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>YJL222W</td>\n",
       "      <td>4649.0</td>\n",
       "      <td>4571</td>\n",
       "      <td>78.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>Q0050</td>\n",
       "      <td>2504.0</td>\n",
       "      <td>1901</td>\n",
       "      <td>603.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>Q0045</td>\n",
       "      <td>12883.0</td>\n",
       "      <td>9003</td>\n",
       "      <td>3880.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>96 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Name  length s288c gene  length yps gene    diff\n",
       "52  YKR102W             3509.0             4118  -609.0\n",
       "83  YPR012W              254.0              281   -27.0\n",
       "6   YBR134W              401.0              410    -9.0\n",
       "73  YNL269W              395.0              401    -6.0\n",
       "76  YNL205C              422.0              425    -3.0\n",
       "..      ...                ...              ...     ...\n",
       "22    Q0105             7107.0             7068    39.0\n",
       "32    Q0105             7107.0             7059    48.0\n",
       "94  YJL222W             4649.0             4571    78.0\n",
       "30    Q0050             2504.0             1901   603.0\n",
       "29    Q0045            12883.0             9003  3880.0\n",
       "\n",
       "[96 rows x 4 columns]"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inframes_comp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b2e0186d",
   "metadata": {},
   "source": [
    "## Add genes with inframe stop codon as having a valid ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "b1474b11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id_x</th>\n",
       "      <th>start_x</th>\n",
       "      <th>end_x</th>\n",
       "      <th>Alias</th>\n",
       "      <th>ID_x</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number_x</th>\n",
       "      <th>...</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Parent</th>\n",
       "      <th>conditions</th>\n",
       "      <th>extra_copy_number_y</th>\n",
       "      <th>inframe_stop_codon</th>\n",
       "      <th>matches_ref_protein</th>\n",
       "      <th>missing_start_codon</th>\n",
       "      <th>missing_stop_codon</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>valid_ORF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>MAL31,MAL3T,MALT,maltose%20permease</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>YBR298C</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltose%20permease</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YBR298C_id001</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288108_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>None</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>YBR298C-A</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>0.990</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YBR298C-A_mRNA</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>RefSeq:NM_001184588.1_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>YBR299W</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltase%20%28alpha-D-glucosidase%29</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YBR299W_id001</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288109_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>None</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>0.993</td>\n",
       "      <td>Dubious%20open%20reading%20frame</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>YGR293C_mRNA</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>PAU12,seripauperin%20PAU12</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>YGR294W</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YGR294W_id001</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289696_1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              seq_id_x  start_x  end_x  \\\n",
       "0  chrII-2_tig00004927     8297  10119   \n",
       "1  chrII-2_tig00004927    10675  10894   \n",
       "2  chrII-2_tig00004927    10985  12717   \n",
       "3  chrII-2_tig00004927    14242  14700   \n",
       "4  chrII-2_tig00004927    14660  15022   \n",
       "\n",
       "                                        Alias         ID_x     Name_x  \\\n",
       "0         MAL31,MAL3T,MALT,maltose%20permease    YBR298C_1    YBR298C   \n",
       "1                                        None  YBR298C-A_1  YBR298C-A   \n",
       "2  MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32    YBR299W_1    YBR299W   \n",
       "3                                        None    YGR293C_2    YGR293C   \n",
       "4                  PAU12,seripauperin%20PAU12    YGR294W_1    YGR294W   \n",
       "\n",
       "   copy_num_ID coverage                                       display  \\\n",
       "0    YBR298C_1    0.986                            Maltose%20permease   \n",
       "1  YBR298C-A_1    0.990  Putative%20protein%20of%20unknown%20function   \n",
       "2    YBR299W_1    0.986           Maltase%20%28alpha-D-glucosidase%29   \n",
       "3    YGR293C_2    0.993              Dubious%20open%20reading%20frame   \n",
       "4    YGR294W_1      1.0             Protein%20of%20unknown%20function   \n",
       "\n",
       "  extra_copy_number_x  ...          Name_y       Parent conditions  \\\n",
       "0                   1  ...   YBR298C_id001    YBR298C_1        GAL   \n",
       "1                   1  ...  YBR298C-A_mRNA  YBR298C-A_1       None   \n",
       "2                   1  ...   YBR299W_id001    YBR299W_1        GAL   \n",
       "3                   2  ...    YGR293C_mRNA    YGR293C_2       None   \n",
       "4                   1  ...   YGR294W_id001    YGR294W_1        GAL   \n",
       "\n",
       "  extra_copy_number_y inframe_stop_codon matches_ref_protein  \\\n",
       "0                   1               None               False   \n",
       "1                   1               None               False   \n",
       "2                   1               None               False   \n",
       "3                   2               True               False   \n",
       "4                   1               None                True   \n",
       "\n",
       "  missing_start_codon  missing_stop_codon            transcript_id valid_ORF  \n",
       "0                None                True         SGD:S000288108_1     False  \n",
       "1                None                True  RefSeq:NM_001184588.1_1     False  \n",
       "2                None                True         SGD:S000288109_1     False  \n",
       "3                None                None                     None     False  \n",
       "4                None                None         SGD:S000289696_1      True  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "b35ffa77",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA.loc[gene_mRNA['inframe_stop_codon'] == 'True', 'valid_ORF'] = \"True\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "41cb0743",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     6263\n",
       "False     428\n",
       "Name: valid_ORF, dtype: int64"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA['valid_ORF'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "142cd805",
   "metadata": {},
   "source": [
    "# kick out duplicated genes who have a valid version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 158,
   "id": "d7d26f8f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "39"
      ]
     },
     "execution_count": 158,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_valid = gene_mRNA[gene_mRNA['valid_ORF'] == 'True' ]\n",
    "gene_mRNA_unvalid = gene_mRNA[gene_mRNA['valid_ORF'] == 'False']\n",
    "\n",
    "valid_list = set(gene_mRNA_valid['Name_x'])\n",
    "unvalid_list = set(gene_mRNA_unvalid['Name_x'])\n",
    "dupl_val_unval = valid_list & unvalid_list\n",
    "\n",
    "# 39 genes have 1 duplicated unvalid version I can forget about\n",
    "len(dupl_val_unval)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "id": "f7ae02ae",
   "metadata": {},
   "outputs": [],
   "source": [
    "# We are kicking out 39 genes \n",
    "gene_mRNA_unvalid = gene_mRNA_unvalid[~gene_mRNA_unvalid['Name_x'].isin(list(dupl_val_unval))]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9f5832cc",
   "metadata": {},
   "source": [
    "### Focus on Verified unvalid (missing start or stop codon)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "f1a4503c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified = gene_mRNA[gene_mRNA['orf_classification']== 'Verified']\n",
    "gene_mRNA_verified_unvalid = gene_mRNA_unvalid[gene_mRNA_unvalid['orf_classification']== 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "69dcda27",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\4212071888.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gene_mRNA_verified_unvalid.drop_duplicates(subset=['Name_x'], inplace=True)\n"
     ]
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid.drop_duplicates(subset=['Name_x'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "id": "b94ac26f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    85\n",
       "Name: missing_stop_codon, dtype: int64"
      ]
     },
     "execution_count": 164,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_stop_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "id": "8b863735",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    17\n",
       "Name: missing_start_codon, dtype: int64"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_start_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "14c71a0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified_unvalid[gene_mRNA_verified_unvalid['missing_start_codon'] == 'True'].to_excel('missing_start.xlsx')\n",
    "gene_mRNA_verified_unvalid[gene_mRNA_verified_unvalid['missing_stop_codon'] == 'True'].to_excel('missing_stop.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "id": "3a54ed6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    85\n",
       "Name: missing_stop_codon, dtype: int64"
      ]
     },
     "execution_count": 168,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_stop_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "id": "995f580d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    17\n",
       "Name: missing_start_codon, dtype: int64"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_start_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76d9fb3e",
   "metadata": {},
   "source": [
    "## Unvalid ORFs to add to valid genes (because essential or valid ORF in y22 Liftoff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "id": "40c642bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_add = []\n",
    "with open('missing_stop_start_genes_to_add.txt', 'r')as file:\n",
    "    for l in file:\n",
    "        gene_to_add.append(l.strip('\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "id": "44043281",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA.loc[gene_mRNA['Name_x'].isin(gene_to_add), 'valid_ORF'] = \"True\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0d45060a",
   "metadata": {},
   "source": [
    "# Add Missing genes that have illumina sequencing values\n",
    "We insured that those are still on the same chromosome by insuring that disomes RPKM values are two times higher in the rest of the strains. More details: C:\\Users\\user\\Documents\\yps1009_nanopore_genomeSeq\\LIFTOFF\\IGV\\James_illumiaReads."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "id": "248464ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "miss_illumina = pd.read_excel('missing_genes_v5_illuminaReads.xlsx', index_col=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "5457d019",
   "metadata": {},
   "outputs": [],
   "source": [
    "del miss_illumina['Unnamed: 0.1']\n",
    "del miss_illumina['Unnamed: 21']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "fe05c73f",
   "metadata": {},
   "outputs": [],
   "source": [
    "miss_illumina = miss_illumina[miss_illumina['AVG rpmk Disome'] > 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "ddb349b4",
   "metadata": {},
   "outputs": [],
   "source": [
    "miss_illumina = lat_to_arabic_S288(miss_illumina)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "1fa590c3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'start', 'end', 'Name', 'gene', 'Alias', 'Note',\n",
       "       'Ontology_term', 'strand', 'phase', 'curie', 'orf_classification',\n",
       "       'display', 'Chromosome', 'Blast', 'rpkm sum', 'no illumina reads',\n",
       "       'AVG rpmk Disome', 'AVG rpkm all but Dis'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miss_illumina.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "93b21dd5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id_x', 'start_x', 'end_x', 'Alias', 'ID_x', 'Name_x',\n",
       "       'copy_num_ID', 'coverage', 'display', 'extra_copy_number_x', 'gene',\n",
       "       'low_identity', 'orf_classification', 'partial_mapping', 'sequence_ID',\n",
       "       'valid_ORFs', 'seq_id_y', 'start_y', 'end_y', 'ID_y', 'Name_y',\n",
       "       'Parent', 'conditions', 'extra_copy_number_y', 'inframe_stop_codon',\n",
       "       'matches_ref_protein', 'missing_start_codon', 'missing_stop_codon',\n",
       "       'transcript_id', 'valid_ORF'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "cbd3b424",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Recall all valid genes\n",
    "gene_mRNA_valid = gene_mRNA[gene_mRNA['valid_ORF'] == 'True']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "id": "865cf971",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_valid = gene_mRNA_valid[['seq_id_x', 'start_x', 'end_x', 'Alias', 'ID_x', 'Name_x',\n",
    "       'copy_num_ID', 'coverage', 'display', 'extra_copy_number_x', 'gene', \n",
    "       'low_identity', 'orf_classification', 'partial_mapping', 'sequence_ID', 'valid_ORF']]\n",
    "gene_mRNA_valid.columns = gene_mRNA_valid.columns.str.strip('_x')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "id": "ccea1181",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_valid = lat_to_arabic(gene_mRNA_valid)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "c2a637d9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>gene</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>curie</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>display</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Blast</th>\n",
       "      <th>rpkm sum</th>\n",
       "      <th>no illumina reads</th>\n",
       "      <th>AVG rpmk Disome</th>\n",
       "      <th>AVG rpkm all but Dis</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9420</th>\n",
       "      <td>chrVII</td>\n",
       "      <td>249869</td>\n",
       "      <td>252738</td>\n",
       "      <td>YGL137W</td>\n",
       "      <td>SEC27</td>\n",
       "      <td>SEC27,coatomer%20subunit%20beta'</td>\n",
       "      <td>Essential beta'-coat protein of the COPI coato...</td>\n",
       "      <td>GO:0006888,GO:0006890,GO:0008298,GO:0030126,GO...</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SGD:S000003105</td>\n",
       "      <td>Verified</td>\n",
       "      <td>Essential beta'-coat protein of the COPI coatomer</td>\n",
       "      <td>7</td>\n",
       "      <td>No hits</td>\n",
       "      <td>1839.993915</td>\n",
       "      <td>False</td>\n",
       "      <td>75.560709</td>\n",
       "      <td>34.172246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9426</th>\n",
       "      <td>chrVII</td>\n",
       "      <td>252897</td>\n",
       "      <td>253859</td>\n",
       "      <td>YGL136C</td>\n",
       "      <td>MRM2</td>\n",
       "      <td>MRM2,21S%20rRNA%20%28uridine2791-2'-O%29%20met...</td>\n",
       "      <td>Mitochondrial 2' O-ribose methyltransferase%3B...</td>\n",
       "      <td>GO:0005739,GO:0008650,GO:0008650,GO:0008650,GO...</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>SGD:S000003104</td>\n",
       "      <td>Verified</td>\n",
       "      <td>Mitochondrial 2' O-ribose methyltransferase</td>\n",
       "      <td>7</td>\n",
       "      <td>No hits</td>\n",
       "      <td>2084.380818</td>\n",
       "      <td>False</td>\n",
       "      <td>58.191421</td>\n",
       "      <td>41.147003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9434</th>\n",
       "      <td>chrVII</td>\n",
       "      <td>255663</td>\n",
       "      <td>256964</td>\n",
       "      <td>YGL134W</td>\n",
       "      <td>PCL10</td>\n",
       "      <td>PCL10</td>\n",
       "      <td>Pho85p cyclin%3B recruits%2C activates%2C and ...</td>\n",
       "      <td>GO:0000079,GO:0000307,GO:0016538,GO:0045719,GO...</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SGD:S000003102</td>\n",
       "      <td>Verified</td>\n",
       "      <td>Pho85p cyclin</td>\n",
       "      <td>7</td>\n",
       "      <td>No hits</td>\n",
       "      <td>1916.747909</td>\n",
       "      <td>False</td>\n",
       "      <td>64.940069</td>\n",
       "      <td>36.821947</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9438</th>\n",
       "      <td>chrVII</td>\n",
       "      <td>257707</td>\n",
       "      <td>261501</td>\n",
       "      <td>YGL133W</td>\n",
       "      <td>ITC1</td>\n",
       "      <td>ITC1</td>\n",
       "      <td>Subunit of ATP-dependent Isw2p-Itc1p chromatin...</td>\n",
       "      <td>GO:0003674,GO:0005634,GO:0006348,GO:0008623,GO...</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SGD:S000003101</td>\n",
       "      <td>Verified</td>\n",
       "      <td>Subunit of ATP-dependent Isw2p-Itc1p chromatin...</td>\n",
       "      <td>7</td>\n",
       "      <td>No hits</td>\n",
       "      <td>1882.690663</td>\n",
       "      <td>False</td>\n",
       "      <td>71.756721</td>\n",
       "      <td>35.459195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9441</th>\n",
       "      <td>chrVII</td>\n",
       "      <td>261580</td>\n",
       "      <td>261915</td>\n",
       "      <td>YGL132W</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>SGD:S000003100</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>7</td>\n",
       "      <td>No hits</td>\n",
       "      <td>2015.775295</td>\n",
       "      <td>False</td>\n",
       "      <td>59.302206</td>\n",
       "      <td>39.523699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            seq_id   start     end     Name   gene  \\\n",
       "Unnamed: 0                                           \n",
       "9420        chrVII  249869  252738  YGL137W  SEC27   \n",
       "9426        chrVII  252897  253859  YGL136C   MRM2   \n",
       "9434        chrVII  255663  256964  YGL134W  PCL10   \n",
       "9438        chrVII  257707  261501  YGL133W   ITC1   \n",
       "9441        chrVII  261580  261915  YGL132W    NaN   \n",
       "\n",
       "                                                        Alias  \\\n",
       "Unnamed: 0                                                      \n",
       "9420                         SEC27,coatomer%20subunit%20beta'   \n",
       "9426        MRM2,21S%20rRNA%20%28uridine2791-2'-O%29%20met...   \n",
       "9434                                                    PCL10   \n",
       "9438                                                     ITC1   \n",
       "9441                                                      NaN   \n",
       "\n",
       "                                                         Note  \\\n",
       "Unnamed: 0                                                      \n",
       "9420        Essential beta'-coat protein of the COPI coato...   \n",
       "9426        Mitochondrial 2' O-ribose methyltransferase%3B...   \n",
       "9434        Pho85p cyclin%3B recruits%2C activates%2C and ...   \n",
       "9438        Subunit of ATP-dependent Isw2p-Itc1p chromatin...   \n",
       "9441        Dubious open reading frame%3B unlikely to enco...   \n",
       "\n",
       "                                                Ontology_term strand phase  \\\n",
       "Unnamed: 0                                                                   \n",
       "9420        GO:0006888,GO:0006890,GO:0008298,GO:0030126,GO...      +     .   \n",
       "9426        GO:0005739,GO:0008650,GO:0008650,GO:0008650,GO...      -     .   \n",
       "9434        GO:0000079,GO:0000307,GO:0016538,GO:0045719,GO...      +     .   \n",
       "9438        GO:0003674,GO:0005634,GO:0006348,GO:0008623,GO...      +     .   \n",
       "9441              GO:0003674,GO:0005575,GO:0008150,SO:0000704      +     .   \n",
       "\n",
       "                     curie orf_classification  \\\n",
       "Unnamed: 0                                      \n",
       "9420        SGD:S000003105           Verified   \n",
       "9426        SGD:S000003104           Verified   \n",
       "9434        SGD:S000003102           Verified   \n",
       "9438        SGD:S000003101           Verified   \n",
       "9441        SGD:S000003100            Dubious   \n",
       "\n",
       "                                                      display  Chromosome  \\\n",
       "Unnamed: 0                                                                  \n",
       "9420        Essential beta'-coat protein of the COPI coatomer           7   \n",
       "9426              Mitochondrial 2' O-ribose methyltransferase           7   \n",
       "9434                                            Pho85p cyclin           7   \n",
       "9438        Subunit of ATP-dependent Isw2p-Itc1p chromatin...           7   \n",
       "9441                               Dubious open reading frame           7   \n",
       "\n",
       "              Blast     rpkm sum  no illumina reads  AVG rpmk Disome  \\\n",
       "Unnamed: 0                                                             \n",
       "9420        No hits  1839.993915              False        75.560709   \n",
       "9426        No hits  2084.380818              False        58.191421   \n",
       "9434        No hits  1916.747909              False        64.940069   \n",
       "9438        No hits  1882.690663              False        71.756721   \n",
       "9441        No hits  2015.775295              False        59.302206   \n",
       "\n",
       "            AVG rpkm all but Dis  \n",
       "Unnamed: 0                        \n",
       "9420                   34.172246  \n",
       "9426                   41.147003  \n",
       "9434                   36.821947  \n",
       "9438                   35.459195  \n",
       "9441                   39.523699  "
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "miss_illumina.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "c59d5855",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\3215699702.py:1: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  gene_mRNA_valid = gene_mRNA_valid.append(miss_illumina)\n"
     ]
    }
   ],
   "source": [
    "gene_mRNA_valid = gene_mRNA_valid.append(miss_illumina)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "id": "4c8e1bc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_valid = gene_mRNA_valid[['seq_id', 'start', 'end', 'Alias', 'ID', 'Name', 'copy_num_ID', 'orf_classification',\n",
    "       'coverage', 'display', 'extra_copy_number', 'gene', 'low_identity', 'Chromosome', 'valid_ORF', 'sequence_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "id": "bff920f2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id_x', 'start_x', 'end_x', 'Alias', 'ID_x', 'Name_x',\n",
       "       'copy_num_ID', 'coverage', 'display', 'extra_copy_number_x', 'gene',\n",
       "       'low_identity', 'orf_classification', 'partial_mapping', 'sequence_ID',\n",
       "       'valid_ORFs', 'seq_id_y', 'start_y', 'end_y', 'ID_y', 'Name_y',\n",
       "       'Parent', 'conditions', 'extra_copy_number_y', 'inframe_stop_codon',\n",
       "       'matches_ref_protein', 'missing_start_codon', 'missing_stop_codon',\n",
       "       'transcript_id', 'valid_ORF'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "ff1c5164",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6321, 16)"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_valid.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "f3efe767",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Verified           5175\n",
       "Uncharacterized     578\n",
       "Dubious             566\n",
       "Name: orf_classification, dtype: int64"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_valid['orf_classification'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "8662934b",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_valid_sh = gene_mRNA_valid[['Name', 'gene', 'Chromosome', 'orf_classification']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "id": "d6c0222f",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_valid_sh.to_csv('yps1009_genes_final.txt', sep = '\\t', index= False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7b84d1c",
   "metadata": {},
   "source": [
    "# Count Verified, Dubious and Uncharacterized Genes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "5932f7ef",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified = gene_mRNA_valid[gene_mRNA_valid['orf_classification'] == 'Verified']\n",
    "gene_mRNA_dubious = gene_mRNA_valid[gene_mRNA_valid['orf_classification'] == 'Dubious']\n",
    "gene_mRNA_uncharacterized = gene_mRNA_valid[gene_mRNA_valid['orf_classification'] == 'Uncharacterized']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "0d9bdf0d",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_count = gene_mRNA_verified['Chromosome'].value_counts().to_frame()\n",
    "verified_count = verified_count.rename(columns = {'Chromosome': 'yps1009 verified genes'})\n",
    "\n",
    "dubious_count = gene_mRNA_dubious['Chromosome'].value_counts().to_frame()\n",
    "dubious_count = dubious_count.rename(columns = {'Chromosome': 'yps1009 dubious genes'})\n",
    "\n",
    "uncharacterized_count = gene_mRNA_uncharacterized['Chromosome'].value_counts().to_frame()\n",
    "uncharacterized_count = uncharacterized_count.rename(columns = {'Chromosome': 'yps1009 uncharacterized genes'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "431d3bd2",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS = FEATS.join(verified_count)\n",
    "FEATS = FEATS.join(dubious_count)\n",
    "FEATS = FEATS.join(uncharacterized_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "8f73ee66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['S288C ARS', 'yps1009 ARS', 'S288C ltr', 'yps1009 ltr', 'S288C ncRNA',\n",
       "       'yps1009 ncRNA', 'S288C snRNA', 'yps1009 snRNA', 'yps1009 LTR RetroP',\n",
       "       'S288C snoRNA', 'yps1009 snoRNA', 'S288C tRNA', 'yps1009 tRNA',\n",
       "       'yps1009 verified genes', 'yps1009 dubious genes',\n",
       "       'yps1009 uncharacterized genes'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FEATS.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "id": "7ddc68e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS_yps = FEATS[['yps1009 verified genes', 'yps1009 dubious genes',\n",
    "       'yps1009 uncharacterized genes', 'yps1009 ARS', 'yps1009 ltr', 'yps1009 ncRNA', 'yps1009 LTR RetroP', 'yps1009 snoRNA',\n",
    "        'yps1009 tRNA', 'yps1009 rRNA', 'yps1009 snRNA']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "id": "620b89d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS_yps.columns = FEATS_yps.columns.str.replace('yps1009 ', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 257,
   "id": "561a5ee4",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\4281676088.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  FEATS_yps['Total Features'] = FEATS_yps.sum(axis=1)\n"
     ]
    }
   ],
   "source": [
    "FEATS_yps['Total Features'] = FEATS_yps.sum(axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "fd648cca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['verified genes', 'dubious genes', 'uncharacterized genes', 'ARS',\n",
       "       'ltr', 'ncRNA', 'LTR RetroP', 'snoRNA', 'tRNA', 'rRNA', 'snRNA',\n",
       "       'Total Features'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 258,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "FEATS_yps.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "id": "0db43b59",
   "metadata": {},
   "outputs": [],
   "source": [
    "FEATS_yps.to_csv('yps1009_v5_features_per_chromosome.csv', index = True)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e11b406a",
   "metadata": {},
   "source": [
    "# Check small contigs that looks like duplicated region from the same chromosome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "id": "bfc2d971",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id_x</th>\n",
       "      <th>start_x</th>\n",
       "      <th>end_x</th>\n",
       "      <th>Alias</th>\n",
       "      <th>ID_x</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number_x</th>\n",
       "      <th>...</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Parent</th>\n",
       "      <th>conditions</th>\n",
       "      <th>extra_copy_number_y</th>\n",
       "      <th>inframe_stop_codon</th>\n",
       "      <th>matches_ref_protein</th>\n",
       "      <th>missing_start_codon</th>\n",
       "      <th>missing_stop_codon</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>valid_ORF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>MAL31,MAL3T,MALT,maltose%20permease</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>YBR298C</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltose%20permease</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YBR298C_id001</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288108_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>None</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>YBR298C-A</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>0.990</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YBR298C-A_mRNA</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>RefSeq:NM_001184588.1_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>YBR299W</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltase%20%28alpha-D-glucosidase%29</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YBR299W_id001</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288109_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>None</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>0.993</td>\n",
       "      <td>Dubious%20open%20reading%20frame</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>YGR293C_mRNA</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>PAU12,seripauperin%20PAU12</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>YGR294W</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YGR294W_id001</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289696_1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>15948</td>\n",
       "      <td>17085</td>\n",
       "      <td>COS6</td>\n",
       "      <td>YGR295C_1</td>\n",
       "      <td>YGR295C</td>\n",
       "      <td>YGR295C_1</td>\n",
       "      <td>0.993</td>\n",
       "      <td>Endosomal%20protein%20involved%20in%20turnover...</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YGR295C_mRNA</td>\n",
       "      <td>YGR295C_1</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>RefSeq:NM_001181424.1_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>17693</td>\n",
       "      <td>17829</td>\n",
       "      <td>None</td>\n",
       "      <td>YHL048C-A_1</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>YHL048C-A_1</td>\n",
       "      <td>0.992</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>YHL048C-A_mRNA</td>\n",
       "      <td>YHL048C-A_1</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>RefSeq:NM_001184650.1_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7 rows Ã— 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              seq_id_x  start_x  end_x  \\\n",
       "0  chrII-2_tig00004927     8297  10119   \n",
       "1  chrII-2_tig00004927    10675  10894   \n",
       "2  chrII-2_tig00004927    10985  12717   \n",
       "3  chrII-2_tig00004927    14242  14700   \n",
       "4  chrII-2_tig00004927    14660  15022   \n",
       "5  chrII-2_tig00004927    15948  17085   \n",
       "6  chrII-2_tig00004927    17693  17829   \n",
       "\n",
       "                                        Alias         ID_x     Name_x  \\\n",
       "0         MAL31,MAL3T,MALT,maltose%20permease    YBR298C_1    YBR298C   \n",
       "1                                        None  YBR298C-A_1  YBR298C-A   \n",
       "2  MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32    YBR299W_1    YBR299W   \n",
       "3                                        None    YGR293C_2    YGR293C   \n",
       "4                  PAU12,seripauperin%20PAU12    YGR294W_1    YGR294W   \n",
       "5                                        COS6    YGR295C_1    YGR295C   \n",
       "6                                        None  YHL048C-A_1  YHL048C-A   \n",
       "\n",
       "   copy_num_ID coverage                                            display  \\\n",
       "0    YBR298C_1    0.986                                 Maltose%20permease   \n",
       "1  YBR298C-A_1    0.990       Putative%20protein%20of%20unknown%20function   \n",
       "2    YBR299W_1    0.986                Maltase%20%28alpha-D-glucosidase%29   \n",
       "3    YGR293C_2    0.993                   Dubious%20open%20reading%20frame   \n",
       "4    YGR294W_1      1.0                  Protein%20of%20unknown%20function   \n",
       "5    YGR295C_1    0.993  Endosomal%20protein%20involved%20in%20turnover...   \n",
       "6  YHL048C-A_1    0.992       Putative%20protein%20of%20unknown%20function   \n",
       "\n",
       "  extra_copy_number_x  ...          Name_y       Parent conditions  \\\n",
       "0                   1  ...   YBR298C_id001    YBR298C_1        GAL   \n",
       "1                   1  ...  YBR298C-A_mRNA  YBR298C-A_1       None   \n",
       "2                   1  ...   YBR299W_id001    YBR299W_1        GAL   \n",
       "3                   2  ...    YGR293C_mRNA    YGR293C_2       None   \n",
       "4                   1  ...   YGR294W_id001    YGR294W_1        GAL   \n",
       "5                   1  ...    YGR295C_mRNA    YGR295C_1       None   \n",
       "6                   1  ...  YHL048C-A_mRNA  YHL048C-A_1       None   \n",
       "\n",
       "  extra_copy_number_y inframe_stop_codon matches_ref_protein  \\\n",
       "0                   1               None               False   \n",
       "1                   1               None               False   \n",
       "2                   1               None               False   \n",
       "3                   2               True               False   \n",
       "4                   1               None                True   \n",
       "5                   1               None               False   \n",
       "6                   1               None               False   \n",
       "\n",
       "  missing_start_codon  missing_stop_codon            transcript_id valid_ORF  \n",
       "0                None                True         SGD:S000288108_1     False  \n",
       "1                None                True  RefSeq:NM_001184588.1_1     False  \n",
       "2                None                True         SGD:S000288109_1     False  \n",
       "3                None                None                     None      True  \n",
       "4                None                None         SGD:S000289696_1      True  \n",
       "5                None                True  RefSeq:NM_001181424.1_1     False  \n",
       "6                None                True  RefSeq:NM_001184650.1_1     False  \n",
       "\n",
       "[7 rows x 30 columns]"
      ]
     },
     "execution_count": 262,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA[gene_mRNA['seq_id_x'] == 'chrII-2_tig00004927']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f5b45e13",
   "metadata": {},
   "source": [
    "# Loading illumina seq RPKM from James \n",
    "The idea is to see whether RPKM of supposedly duplicated region have a higher read counts "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "id": "ef35d158",
   "metadata": {},
   "outputs": [],
   "source": [
    "rpkm = pd.read_csv('James_220427_KC5T7-RPKM.results', sep = '\\t', index_col=0)\n",
    "samples = pd.read_csv('YPS1009-Disomes_GenomicSeq.txt', sep = '\\t')\n",
    "dic = {}\n",
    "for ind, row in samples.iterrows():\n",
    "    code = row['Next-Gen Gasch #'].replace(' ', '-')\n",
    "    dic[code] = row['Strain']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "id": "20f857f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "for c in list(rpkm.columns):\n",
    "    cs, t = c.split('_')\n",
    "    rpkm = rpkm.rename(columns={c: dic[cs]})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 265,
   "id": "a9f0f236",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_16892\\2898711814.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  rpkm_Eupl['Median'] = rpkm_Eupl.median(axis=1)\n"
     ]
    }
   ],
   "source": [
    "rpkm_Eupl = rpkm.filter(like=\"Euploid\", axis=1)\n",
    "rpkm_Eupl['Median'] = rpkm_Eupl.median(axis=1)\n",
    "#filter 0 and low detection genes\n",
    "rpkm_Eupl = rpkm_Eupl[rpkm_Eupl['Median'] > 5]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "eb60670c",
   "metadata": {},
   "source": [
    "## now check these contigs that are partially mapped to other contigs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "8a1e9477",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "4858be8c",
   "metadata": {},
   "outputs": [],
   "source": [
    "chr2_2_genes = list(gene_mRNA[gene_mRNA['seq_id_x'] == 'chrII-2_tig00004927']['Name_x'])\n",
    "chr2_genes = list(gene_mRNA[gene_mRNA['seq_id_x'] == 'chrII_tig00000010']['Name_x'])\n",
    "\n",
    "ch2_rp = rpkm_Eupl[rpkm_Eupl.index.isin(chr2_genes)]\n",
    "ch2_dupl_rp = rpkm_Eupl[rpkm_Eupl.index.isin(chr2_2_genes)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "73ec4a5e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "39.75553585341481"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.median(ch2_rp['Median'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "51a51024",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "35.73805739723673"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.median(ch2_dupl_rp['Median'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32d9d613",
   "metadata": {},
   "source": [
    "## Chromosome 12 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 275,
   "id": "104fe58a",
   "metadata": {},
   "outputs": [],
   "source": [
    "chr12_2_genes = list(gene_mRNA[gene_mRNA['seq_id_x'] == 'chrXII-2_tig00004934']['Name_x'])\n",
    "chr12_genes = list(gene_mRNA[gene_mRNA['seq_id_x'] == 'chrXII_tig00000020']['Name_x'])\n",
    "\n",
    "ch12_rp = rpkm_Eupl[rpkm_Eupl.index.isin(chr12_genes)]\n",
    "ch12_dupl_rp = rpkm_Eupl[rpkm_Eupl.index.isin(chr12_2_genes)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "id": "b5368ca6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chr12:  39.05192975088535\n",
      "Chr12 short contig:  38.5105917963216\n"
     ]
    }
   ],
   "source": [
    "print('Chr12: ', np.median(ch12_rp['Median']))\n",
    "print('Chr12 short contig: ', np.median(ch12_dupl_rp['Median']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "076d7dc4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
