{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fbef926e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from BCBio.GFF import GFFExaminer"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "18ceb6d5",
   "metadata": {},
   "source": [
    "https://gffpandas.readthedocs.io/en/latest/tutorial.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "id": "88084c43",
   "metadata": {},
   "outputs": [],
   "source": [
    "import gffpandas.gffpandas as gffpd\n",
    "import pandas as pd\n",
    "# Import Biopython tools for running local BLASTX\n",
    "from Bio.Blast.Applications import NcbiblastnCommandline\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0e4c4aab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\anaconda3\\lib\\site-packages\\gffpandas\\gffpandas.py:32: DtypeWarning: Columns (1,2,5,6,7,8) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  self.df = pd.read_table(self._gff_file, comment='#',\n"
     ]
    }
   ],
   "source": [
    "annotation = gffpd.read_gff3('YPS1009_S288C_v5_polished.gff')\n",
    "annot_s288c = gffpd.read_gff3('saccharomyces_cerevisiae_R64-3.gff')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "05ec8c7a",
   "metadata": {},
   "outputs": [],
   "source": [
    "features = annotation.df['type'].value_counts().to_frame()\n",
    "features = features.rename(columns = {'type': 'yps1009'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1f380f29",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "features = features.join(annot_s288c.df['type'].value_counts().to_frame())\n",
    "features = features.rename(columns={'type': 'S288C'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "10aa4474",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>yps1009</th>\n",
       "      <th>S288C</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CDS</th>\n",
       "      <td>7164</td>\n",
       "      <td>7066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <td>6691</td>\n",
       "      <td>6607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mRNA</th>\n",
       "      <td>6691</td>\n",
       "      <td>11119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARS</th>\n",
       "      <td>554</td>\n",
       "      <td>543</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>noncoding_exon</th>\n",
       "      <td>512</td>\n",
       "      <td>489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>long_terminal_repeat</th>\n",
       "      <td>361</td>\n",
       "      <td>384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA</th>\n",
       "      <td>323</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tRNA_gene</th>\n",
       "      <td>323</td>\n",
       "      <td>299</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARS_consensus_sequence</th>\n",
       "      <td>199</td>\n",
       "      <td>196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomere</th>\n",
       "      <td>98</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transposable_element</th>\n",
       "      <td>95</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>transposable_element_gene</th>\n",
       "      <td>95</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>region</th>\n",
       "      <td>82</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snoRNA</th>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snoRNA_gene</th>\n",
       "      <td>77</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LTR_retrotransposon</th>\n",
       "      <td>53</td>\n",
       "      <td>50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>plus_1_translational_frameshift</th>\n",
       "      <td>47</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomeric_repeat</th>\n",
       "      <td>42</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_element</th>\n",
       "      <td>35</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_element_combinatorial_repeat</th>\n",
       "      <td>33</td>\n",
       "      <td>28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere</th>\n",
       "      <td>30</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>pseudogene</th>\n",
       "      <td>25</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>five_prime_UTR_intron</th>\n",
       "      <td>24</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA_gene</th>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ncRNA</th>\n",
       "      <td>23</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA_gene</th>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rRNA</th>\n",
       "      <td>20</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>chromosome</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y_prime_element</th>\n",
       "      <td>17</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere_DNA_Element_II</th>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere_DNA_Element_III</th>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>centromere_DNA_Element_I</th>\n",
       "      <td>15</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>blocked_reading_frame</th>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>origin_of_replication</th>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>external_transcribed_spacer_region</th>\n",
       "      <td>6</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>uORF</th>\n",
       "      <td>6</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snRNA_gene</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>snRNA</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>silent_mating_type_cassette_array</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>internal_transcribed_spacer_region</th>\n",
       "      <td>4</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Z1_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>non_transcribed_region</th>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>intein_encoding_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>W_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mating_type_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Z2_region</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomerase_RNA</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>telomerase_RNA_gene</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>recombination_enhancer</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    yps1009  S288C\n",
       "CDS                                    7164   7066\n",
       "gene                                   6691   6607\n",
       "mRNA                                   6691  11119\n",
       "ARS                                     554    543\n",
       "noncoding_exon                          512    489\n",
       "long_terminal_repeat                    361    384\n",
       "tRNA                                    323    299\n",
       "tRNA_gene                               323    299\n",
       "ARS_consensus_sequence                  199    196\n",
       "telomere                                 98     64\n",
       "transposable_element                     95     91\n",
       "transposable_element_gene                95     91\n",
       "region                                   82     78\n",
       "snoRNA                                   77     77\n",
       "snoRNA_gene                              77     77\n",
       "LTR_retrotransposon                      53     50\n",
       "plus_1_translational_frameshift          47     47\n",
       "telomeric_repeat                         42     31\n",
       "X_element                                35     32\n",
       "X_element_combinatorial_repeat           33     28\n",
       "centromere                               30     32\n",
       "pseudogene                               25     12\n",
       "five_prime_UTR_intron                    24     24\n",
       "ncRNA_gene                               23     23\n",
       "ncRNA                                    23     23\n",
       "rRNA_gene                                20     24\n",
       "rRNA                                     20     24\n",
       "chromosome                               17     17\n",
       "Y_prime_element                          17     19\n",
       "centromere_DNA_Element_II                15     16\n",
       "centromere_DNA_Element_III               15     16\n",
       "centromere_DNA_Element_I                 15     16\n",
       "blocked_reading_frame                    12     12\n",
       "origin_of_replication                    10      8\n",
       "external_transcribed_spacer_region        6      8\n",
       "uORF                                      6     11\n",
       "snRNA_gene                                6      6\n",
       "snRNA                                     6      6\n",
       "silent_mating_type_cassette_array         4      4\n",
       "internal_transcribed_spacer_region        4      8\n",
       "X_region                                  3      3\n",
       "Y_region                                  3      3\n",
       "Z1_region                                 3      3\n",
       "non_transcribed_region                    3      3\n",
       "intein_encoding_region                    2      2\n",
       "W_region                                  2      2\n",
       "mating_type_region                        2      2\n",
       "Z2_region                                 2      2\n",
       "telomerase_RNA                            1      1\n",
       "telomerase_RNA_gene                       1      1\n",
       "recombination_enhancer                    1      1"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "76666a89",
   "metadata": {},
   "source": [
    "# Coding sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "id": "64b45cf1",
   "metadata": {},
   "outputs": [],
   "source": [
    "CDS_df = annotation.filter_feature_of_type(['CDS'])\n",
    "CDS_attr = CDS_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "id": "784abc3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "CDS_attr['Name short'] = CDS_attr['Name'].str.replace('_CDS', '')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 310,
   "id": "b0dbf958",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6565"
      ]
     },
     "execution_count": 310,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(gene['Name']) & set(CDS_attr['Name short']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 327,
   "id": "8db38089",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7164, 16)"
      ]
     },
     "execution_count": 327,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CDS_attr.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 330,
   "id": "5c3e2fff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6691, 23)"
      ]
     },
     "execution_count": 330,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 331,
   "id": "5720ef29",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    6563\n",
       "1      91\n",
       "2      19\n",
       "3      10\n",
       "4       3\n",
       "5       2\n",
       "6       1\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 331,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 329,
   "id": "c5399817",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    7020\n",
       "1     107\n",
       "2      20\n",
       "3      11\n",
       "4       3\n",
       "5       2\n",
       "6       1\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 329,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CDS_attr['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "id": "59cdd40a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "97"
      ]
     },
     "execution_count": 312,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(CDS_attr['Name short']) - set(gene['Name']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "id": "80a66c76",
   "metadata": {},
   "outputs": [],
   "source": [
    "diff = set(CDS_attr['Name short']) - set(gene['Name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 315,
   "id": "5e1aaf1a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>attributes</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Parent</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>Name short</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>410</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>140850</td>\n",
       "      <td>142166</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_140_1;Parent=YBL100W-A_transposable_ele...</td>\n",
       "      <td>CDS_140_1</td>\n",
       "      <td>YBL100W-A_CDS</td>\n",
       "      <td>YBL100W-A_transposable_element_1</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:Q12260_1</td>\n",
       "      <td>YBL100W-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1128</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>185379</td>\n",
       "      <td>186695</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_1322;Parent=YDR210W-A_transposable_elem...</td>\n",
       "      <td>CDS_1322</td>\n",
       "      <td>YDR210W-A_CDS</td>\n",
       "      <td>YDR210W-A_transposable_element</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:Q03483</td>\n",
       "      <td>YDR210W-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1131</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>185379</td>\n",
       "      <td>186695</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_1380;Parent=YDR261W-A_transposable_elem...</td>\n",
       "      <td>CDS_1380</td>\n",
       "      <td>YDR261W-A_CDS</td>\n",
       "      <td>YDR261W-A_transposable_element</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:Q99303</td>\n",
       "      <td>YDR261W-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>185379</td>\n",
       "      <td>186695</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_2668;Parent=YGR161W-A_transposable_elem...</td>\n",
       "      <td>CDS_2668</td>\n",
       "      <td>YGR161W-A_CDS</td>\n",
       "      <td>YGR161W-A_transposable_element</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:P0CX62</td>\n",
       "      <td>YGR161W-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3667</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>700770</td>\n",
       "      <td>701156</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_1238;Parent=YDR134C;Name=YDR134C_CDS;pr...</td>\n",
       "      <td>CDS_1238</td>\n",
       "      <td>YDR134C_CDS</td>\n",
       "      <td>YDR134C</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:Q7LHD1</td>\n",
       "      <td>YDR134C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19307</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>58988</td>\n",
       "      <td>60304</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_668;Parent=YCL020W_transposable_element...</td>\n",
       "      <td>CDS_668</td>\n",
       "      <td>YCL020W_CDS</td>\n",
       "      <td>YCL020W_transposable_element</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:P25383</td>\n",
       "      <td>YCL020W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21015</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>37854</td>\n",
       "      <td>39599</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_5850;Parent=YOL153C;Name=YOL153C_CDS;pr...</td>\n",
       "      <td>CDS_5850</td>\n",
       "      <td>YOL153C_CDS</td>\n",
       "      <td>YOL153C</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:P0C155</td>\n",
       "      <td>YOL153C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21165</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>112411</td>\n",
       "      <td>113727</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_140;Parent=YBL100W-A_transposable_eleme...</td>\n",
       "      <td>CDS_140</td>\n",
       "      <td>YBL100W-A_CDS</td>\n",
       "      <td>YBL100W-A_transposable_element</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:Q12260</td>\n",
       "      <td>YBL100W-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21698</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>396557</td>\n",
       "      <td>396766</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_6054;Parent=YOR031W;Name=YOR031W_CDS;pr...</td>\n",
       "      <td>CDS_6054</td>\n",
       "      <td>YOR031W_CDS</td>\n",
       "      <td>YOR031W</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:P41902</td>\n",
       "      <td>YOR031W</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23238</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>CDS</td>\n",
       "      <td>406970</td>\n",
       "      <td>408286</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CDS_1126_1;Parent=YDR034C-C_transposable_el...</td>\n",
       "      <td>CDS_1126_1</td>\n",
       "      <td>YDR034C-C_CDS</td>\n",
       "      <td>YDR034C-C_transposable_element_1</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>UniProtKB:Q12392_1</td>\n",
       "      <td>YDR034C-C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>145 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     seq_id   source type   start     end score strand phase  \\\n",
       "410      chrIII_tig00004936  Liftoff  CDS  140850  142166     .      -     .   \n",
       "1128      chrII_tig00000010  Liftoff  CDS  185379  186695     .      +     .   \n",
       "1131      chrII_tig00000010  Liftoff  CDS  185379  186695     .      +     .   \n",
       "1134      chrII_tig00000010  Liftoff  CDS  185379  186695     .      +     .   \n",
       "3667      chrIV_tig00000001  Liftoff  CDS  700770  701156     .      -     .   \n",
       "...                     ...      ...  ...     ...     ...   ...    ...   ...   \n",
       "19307  chrXVI-2_tig00000040  Liftoff  CDS   58988   60304     .      -     .   \n",
       "21015     chrXV_tig00000007  Liftoff  CDS   37854   39599     .      -     .   \n",
       "21165     chrXV_tig00000007  Liftoff  CDS  112411  113727     .      +     .   \n",
       "21698     chrXV_tig00000007  Liftoff  CDS  396557  396766     .      +     .   \n",
       "23238      chrX_tig00001098  Liftoff  CDS  406970  408286     .      +     .   \n",
       "\n",
       "                                              attributes          ID  \\\n",
       "410    ID=CDS_140_1;Parent=YBL100W-A_transposable_ele...   CDS_140_1   \n",
       "1128   ID=CDS_1322;Parent=YDR210W-A_transposable_elem...    CDS_1322   \n",
       "1131   ID=CDS_1380;Parent=YDR261W-A_transposable_elem...    CDS_1380   \n",
       "1134   ID=CDS_2668;Parent=YGR161W-A_transposable_elem...    CDS_2668   \n",
       "3667   ID=CDS_1238;Parent=YDR134C;Name=YDR134C_CDS;pr...    CDS_1238   \n",
       "...                                                  ...         ...   \n",
       "19307  ID=CDS_668;Parent=YCL020W_transposable_element...     CDS_668   \n",
       "21015  ID=CDS_5850;Parent=YOL153C;Name=YOL153C_CDS;pr...    CDS_5850   \n",
       "21165  ID=CDS_140;Parent=YBL100W-A_transposable_eleme...     CDS_140   \n",
       "21698  ID=CDS_6054;Parent=YOR031W;Name=YOR031W_CDS;pr...    CDS_6054   \n",
       "23238  ID=CDS_1126_1;Parent=YDR034C-C_transposable_el...  CDS_1126_1   \n",
       "\n",
       "                Name                            Parent extra_copy_number  \\\n",
       "410    YBL100W-A_CDS  YBL100W-A_transposable_element_1                 1   \n",
       "1128   YDR210W-A_CDS    YDR210W-A_transposable_element                 0   \n",
       "1131   YDR261W-A_CDS    YDR261W-A_transposable_element                 0   \n",
       "1134   YGR161W-A_CDS    YGR161W-A_transposable_element                 0   \n",
       "3667     YDR134C_CDS                           YDR134C                 0   \n",
       "...              ...                               ...               ...   \n",
       "19307    YCL020W_CDS      YCL020W_transposable_element                 0   \n",
       "21015    YOL153C_CDS                           YOL153C                 0   \n",
       "21165  YBL100W-A_CDS    YBL100W-A_transposable_element                 0   \n",
       "21698    YOR031W_CDS                           YOR031W                 0   \n",
       "23238  YDR034C-C_CDS  YDR034C-C_transposable_element_1                 1   \n",
       "\n",
       "      orf_classification          protein_id Name short  \n",
       "410                 None  UniProtKB:Q12260_1  YBL100W-A  \n",
       "1128                None    UniProtKB:Q03483  YDR210W-A  \n",
       "1131                None    UniProtKB:Q99303  YDR261W-A  \n",
       "1134                None    UniProtKB:P0CX62  YGR161W-A  \n",
       "3667                None    UniProtKB:Q7LHD1    YDR134C  \n",
       "...                  ...                 ...        ...  \n",
       "19307               None    UniProtKB:P25383    YCL020W  \n",
       "21015               None    UniProtKB:P0C155    YOL153C  \n",
       "21165               None    UniProtKB:Q12260  YBL100W-A  \n",
       "21698               None    UniProtKB:P41902    YOR031W  \n",
       "23238               None  UniProtKB:Q12392_1  YDR034C-C  \n",
       "\n",
       "[145 rows x 16 columns]"
      ]
     },
     "execution_count": 315,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CDS_attr[CDS_attr['Name short'].isin(diff)].to_excel('CDS_not_gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57f812ed",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c6a7aea",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e888216",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "26784560",
   "metadata": {},
   "source": [
    "# Centromere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2cafc06",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "1b9e2beb",
   "metadata": {},
   "outputs": [],
   "source": [
    "CEN_df = annotation.filter_feature_of_type(['centromere'])\n",
    "CEN_attr = CEN_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "695e9913",
   "metadata": {},
   "outputs": [],
   "source": [
    "CEN_attr = CEN_attr[CEN_attr['Alias'].notna()]\n",
    "CEN_attr = CEN_attr.sort_values(by=['Alias'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "4f4d0c43",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>attributes</th>\n",
       "      <th>Alias</th>\n",
       "      <th>...</th>\n",
       "      <th>Note</th>\n",
       "      <th>Parent</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>curie</th>\n",
       "      <th>dbxref</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>sequence_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6319</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>148004</td>\n",
       "      <td>148121</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN1;Name=CEN1;gene=CEN1;Alias=CEN1;Note=Ch...</td>\n",
       "      <td>CEN1</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20I%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN1_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006463</td>\n",
       "      <td>SGD:S000006463</td>\n",
       "      <td>Chromosome%20I%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN1</td>\n",
       "      <td>0.957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23281</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>432174</td>\n",
       "      <td>432290</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN10;Name=CEN10;gene=CEN10;Alias=CEN10;Not...</td>\n",
       "      <td>CEN10</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20X%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN10_0</td>\n",
       "      <td>0.983</td>\n",
       "      <td>SGD:S000006471</td>\n",
       "      <td>SGD:S000006471</td>\n",
       "      <td>Chromosome%20X%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN10</td>\n",
       "      <td>0.957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18292</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>440039</td>\n",
       "      <td>440156</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN11;Name=CEN11;gene=CEN11;Alias=CEN11;Not...</td>\n",
       "      <td>CEN11</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20XI%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN11_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006472</td>\n",
       "      <td>SGD:S000006472</td>\n",
       "      <td>Chromosome%20XI%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN11</td>\n",
       "      <td>0.983</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12444</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>144863</td>\n",
       "      <td>144981</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN12;Name=CEN12;gene=CEN12;Alias=CEN12;Not...</td>\n",
       "      <td>CEN12</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20XII%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN12_0</td>\n",
       "      <td>0.991</td>\n",
       "      <td>SGD:S000006473</td>\n",
       "      <td>SGD:S000006473</td>\n",
       "      <td>Chromosome%20XII%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN12</td>\n",
       "      <td>0.958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13618</th>\n",
       "      <td>chrXIII-2_tig00000029</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>251334</td>\n",
       "      <td>251453</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN13;Name=CEN13;gene=CEN13;Alias=CEN13;Not...</td>\n",
       "      <td>CEN13</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20XIII%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN13_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006474</td>\n",
       "      <td>SGD:S000006474</td>\n",
       "      <td>Chromosome%20XIII%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN13</td>\n",
       "      <td>0.966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17197</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>576570</td>\n",
       "      <td>576687</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN14;Name=CEN14;gene=CEN14;Alias=CEN14;Not...</td>\n",
       "      <td>CEN14</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20XIV%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN14_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006475</td>\n",
       "      <td>SGD:S000006475</td>\n",
       "      <td>Chromosome%20XIV%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN14</td>\n",
       "      <td>0.966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21577</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>333875</td>\n",
       "      <td>333993</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN15;Name=CEN15;gene=CEN15;Alias=CEN15;Not...</td>\n",
       "      <td>CEN15</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20XV%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN15_0</td>\n",
       "      <td>0.991</td>\n",
       "      <td>SGD:S000006476</td>\n",
       "      <td>SGD:S000006476</td>\n",
       "      <td>Chromosome%20XV%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN15</td>\n",
       "      <td>0.941</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20176</th>\n",
       "      <td>chrXVI_tig00000013</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>351056</td>\n",
       "      <td>351172</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN16;Name=CEN16;gene=CEN16;Alias=CEN16;Not...</td>\n",
       "      <td>CEN16</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20XVI%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN16_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006477</td>\n",
       "      <td>SGD:S000006477</td>\n",
       "      <td>Chromosome%20XVI%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN16</td>\n",
       "      <td>0.974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1200</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>226275</td>\n",
       "      <td>226391</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN2;Name=CEN2;gene=CEN2;Alias=CEN2;Note=Ch...</td>\n",
       "      <td>CEN2</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20II%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN2_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006464</td>\n",
       "      <td>SGD:S000006464</td>\n",
       "      <td>Chromosome%20II%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN2</td>\n",
       "      <td>0.965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>108185</td>\n",
       "      <td>108301</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN3;Name=CEN3;gene=CEN3;Alias=CEN3;Note=Ch...</td>\n",
       "      <td>CEN3</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20III%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN3_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006465</td>\n",
       "      <td>SGD:S000006465</td>\n",
       "      <td>Chromosome%20III%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN3</td>\n",
       "      <td>0.957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3184</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>450223</td>\n",
       "      <td>450341</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN4;Name=CEN4;gene=CEN4;Alias=CEN4;Note=Ch...</td>\n",
       "      <td>CEN4</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20IV%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN4_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006466</td>\n",
       "      <td>SGD:S000006466</td>\n",
       "      <td>Chromosome%20IV%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN4</td>\n",
       "      <td>0.911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11296</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>152380</td>\n",
       "      <td>152497</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN5;Name=CEN5;gene=CEN5;Alias=CEN5;Note=Ch...</td>\n",
       "      <td>CEN5</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20V%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN5_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006467</td>\n",
       "      <td>SGD:S000006467</td>\n",
       "      <td>Chromosome%20V%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN5</td>\n",
       "      <td>0.966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9095</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>235936</td>\n",
       "      <td>236054</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN7;Name=CEN7;gene=CEN7;Alias=CEN7;Note=Ch...</td>\n",
       "      <td>CEN7</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20VII%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN7_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006468</td>\n",
       "      <td>SGD:S000006468</td>\n",
       "      <td>Chromosome%20VII%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN7</td>\n",
       "      <td>0.966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7690</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>94751</td>\n",
       "      <td>94868</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN8;Name=CEN8;gene=CEN8;Alias=CEN8;Note=Ch...</td>\n",
       "      <td>CEN8</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20VIII%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN8_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006469</td>\n",
       "      <td>SGD:S000006469</td>\n",
       "      <td>Chromosome%20VIII%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN8</td>\n",
       "      <td>0.974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5832</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>Liftoff</td>\n",
       "      <td>centromere</td>\n",
       "      <td>345678</td>\n",
       "      <td>345794</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=CEN9;Name=CEN9;gene=CEN9;Alias=CEN9;Note=Ch...</td>\n",
       "      <td>CEN9</td>\n",
       "      <td>...</td>\n",
       "      <td>Chromosome%20IX%20centromere</td>\n",
       "      <td>None</td>\n",
       "      <td>CEN9_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SGD:S000006470</td>\n",
       "      <td>SGD:S000006470</td>\n",
       "      <td>Chromosome%20IX%20centromere</td>\n",
       "      <td>0</td>\n",
       "      <td>CEN9</td>\n",
       "      <td>0.965</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      seq_id   source        type   start     end score  \\\n",
       "6319        chrI_tig00000033  Liftoff  centromere  148004  148121     .   \n",
       "23281       chrX_tig00001098  Liftoff  centromere  432174  432290     .   \n",
       "18292      chrXI_tig00004932  Liftoff  centromere  440039  440156     .   \n",
       "12444   chrXII-2_tig00004934  Liftoff  centromere  144863  144981     .   \n",
       "13618  chrXIII-2_tig00000029  Liftoff  centromere  251334  251453     .   \n",
       "17197     chrXIV_tig00004929  Liftoff  centromere  576570  576687     .   \n",
       "21577      chrXV_tig00000007  Liftoff  centromere  333875  333993     .   \n",
       "20176     chrXVI_tig00000013  Liftoff  centromere  351056  351172     .   \n",
       "1200       chrII_tig00000010  Liftoff  centromere  226275  226391     .   \n",
       "339       chrIII_tig00004936  Liftoff  centromere  108185  108301     .   \n",
       "3184       chrIV_tig00000001  Liftoff  centromere  450223  450341     .   \n",
       "11296       chrV_tig00000015  Liftoff  centromere  152380  152497     .   \n",
       "9095      chrVII_tig00004928  Liftoff  centromere  235936  236054     .   \n",
       "7690     chrVIII_tig00000017  Liftoff  centromere   94751   94868     .   \n",
       "5832       chrIX_tig00000023  Liftoff  centromere  345678  345794     .   \n",
       "\n",
       "      strand phase                                         attributes  Alias  \\\n",
       "6319       +     .  ID=CEN1;Name=CEN1;gene=CEN1;Alias=CEN1;Note=Ch...   CEN1   \n",
       "23281      -     .  ID=CEN10;Name=CEN10;gene=CEN10;Alias=CEN10;Not...  CEN10   \n",
       "18292      -     .  ID=CEN11;Name=CEN11;gene=CEN11;Alias=CEN11;Not...  CEN11   \n",
       "12444      -     .  ID=CEN12;Name=CEN12;gene=CEN12;Alias=CEN12;Not...  CEN12   \n",
       "13618      +     .  ID=CEN13;Name=CEN13;gene=CEN13;Alias=CEN13;Not...  CEN13   \n",
       "17197      +     .  ID=CEN14;Name=CEN14;gene=CEN14;Alias=CEN14;Not...  CEN14   \n",
       "21577      -     .  ID=CEN15;Name=CEN15;gene=CEN15;Alias=CEN15;Not...  CEN15   \n",
       "20176      +     .  ID=CEN16;Name=CEN16;gene=CEN16;Alias=CEN16;Not...  CEN16   \n",
       "1200       +     .  ID=CEN2;Name=CEN2;gene=CEN2;Alias=CEN2;Note=Ch...   CEN2   \n",
       "339        +     .  ID=CEN3;Name=CEN3;gene=CEN3;Alias=CEN3;Note=Ch...   CEN3   \n",
       "3184       +     .  ID=CEN4;Name=CEN4;gene=CEN4;Alias=CEN4;Note=Ch...   CEN4   \n",
       "11296      +     .  ID=CEN5;Name=CEN5;gene=CEN5;Alias=CEN5;Note=Ch...   CEN5   \n",
       "9095       -     .  ID=CEN7;Name=CEN7;gene=CEN7;Alias=CEN7;Note=Ch...   CEN7   \n",
       "7690       -     .  ID=CEN8;Name=CEN8;gene=CEN8;Alias=CEN8;Note=Ch...   CEN8   \n",
       "5832       +     .  ID=CEN9;Name=CEN9;gene=CEN9;Alias=CEN9;Note=Ch...   CEN9   \n",
       "\n",
       "       ...                            Note Parent copy_num_ID coverage  \\\n",
       "6319   ...     Chromosome%20I%20centromere   None      CEN1_0      1.0   \n",
       "23281  ...     Chromosome%20X%20centromere   None     CEN10_0    0.983   \n",
       "18292  ...    Chromosome%20XI%20centromere   None     CEN11_0      1.0   \n",
       "12444  ...   Chromosome%20XII%20centromere   None     CEN12_0    0.991   \n",
       "13618  ...  Chromosome%20XIII%20centromere   None     CEN13_0      1.0   \n",
       "17197  ...   Chromosome%20XIV%20centromere   None     CEN14_0      1.0   \n",
       "21577  ...    Chromosome%20XV%20centromere   None     CEN15_0    0.991   \n",
       "20176  ...   Chromosome%20XVI%20centromere   None     CEN16_0      1.0   \n",
       "1200   ...    Chromosome%20II%20centromere   None      CEN2_0      1.0   \n",
       "339    ...   Chromosome%20III%20centromere   None      CEN3_0      1.0   \n",
       "3184   ...    Chromosome%20IV%20centromere   None      CEN4_0      1.0   \n",
       "11296  ...     Chromosome%20V%20centromere   None      CEN5_0      1.0   \n",
       "9095   ...   Chromosome%20VII%20centromere   None      CEN7_0      1.0   \n",
       "7690   ...  Chromosome%20VIII%20centromere   None      CEN8_0      1.0   \n",
       "5832   ...    Chromosome%20IX%20centromere   None      CEN9_0      1.0   \n",
       "\n",
       "                curie          dbxref                         display  \\\n",
       "6319   SGD:S000006463  SGD:S000006463     Chromosome%20I%20centromere   \n",
       "23281  SGD:S000006471  SGD:S000006471     Chromosome%20X%20centromere   \n",
       "18292  SGD:S000006472  SGD:S000006472    Chromosome%20XI%20centromere   \n",
       "12444  SGD:S000006473  SGD:S000006473   Chromosome%20XII%20centromere   \n",
       "13618  SGD:S000006474  SGD:S000006474  Chromosome%20XIII%20centromere   \n",
       "17197  SGD:S000006475  SGD:S000006475   Chromosome%20XIV%20centromere   \n",
       "21577  SGD:S000006476  SGD:S000006476    Chromosome%20XV%20centromere   \n",
       "20176  SGD:S000006477  SGD:S000006477   Chromosome%20XVI%20centromere   \n",
       "1200   SGD:S000006464  SGD:S000006464    Chromosome%20II%20centromere   \n",
       "339    SGD:S000006465  SGD:S000006465   Chromosome%20III%20centromere   \n",
       "3184   SGD:S000006466  SGD:S000006466    Chromosome%20IV%20centromere   \n",
       "11296  SGD:S000006467  SGD:S000006467     Chromosome%20V%20centromere   \n",
       "9095   SGD:S000006468  SGD:S000006468   Chromosome%20VII%20centromere   \n",
       "7690   SGD:S000006469  SGD:S000006469  Chromosome%20VIII%20centromere   \n",
       "5832   SGD:S000006470  SGD:S000006470    Chromosome%20IX%20centromere   \n",
       "\n",
       "      extra_copy_number   gene sequence_ID  \n",
       "6319                  0   CEN1       0.957  \n",
       "23281                 0  CEN10       0.957  \n",
       "18292                 0  CEN11       0.983  \n",
       "12444                 0  CEN12       0.958  \n",
       "13618                 0  CEN13       0.966  \n",
       "17197                 0  CEN14       0.966  \n",
       "21577                 0  CEN15       0.941  \n",
       "20176                 0  CEN16       0.974  \n",
       "1200                  0   CEN2       0.965  \n",
       "339                   0   CEN3       0.957  \n",
       "3184                  0   CEN4       0.911  \n",
       "11296                 0   CEN5       0.966  \n",
       "9095                  0   CEN7       0.966  \n",
       "7690                  0   CEN8       0.974  \n",
       "5832                  0   CEN9       0.965  \n",
       "\n",
       "[15 rows x 22 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "CEN_attr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6709dcde",
   "metadata": {},
   "source": [
    "# Telomere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "bf10befd",
   "metadata": {},
   "outputs": [],
   "source": [
    "telomere_df = annotation.filter_feature_of_type(['telomere'])\n",
    "telomere_attr = telomere_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "db8ab677",
   "metadata": {},
   "outputs": [],
   "source": [
    "telomere_attr = telomere_attr[telomere_attr['Parent'].notna()]\n",
    "telomere_attr = telomere_attr.sort_values(by=['ID', 'start'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1009c702",
   "metadata": {},
   "source": [
    "## Make df by features YPS1009"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "45985f8d",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_df = annotation.filter_feature_of_type(['transposable_element_gene'])\n",
    "sno_df = annotation.filter_feature_of_type(['snoRNA_gene'])\n",
    "tRNA_df = annotation.filter_feature_of_type(['tRNA_gene'])\n",
    "gene_df = annotation.filter_feature_of_type(['gene'])\n",
    "LTR_retroP_df = annotation.filter_feature_of_type(['LTR_retrotransposon'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "aa90ef94",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_attr = gene_df.attributes_to_columns()\n",
    "gene = gene_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
    "       'partial_mapping', 'sequence_ID', 'valid_ORFs']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "65fb5bab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'Ontology_term',\n",
       "       'copy_num_ID', 'coverage', 'curie', 'dbxref', 'display',\n",
       "       'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
       "       'partial_mapping', 'sequence_ID', 'valid_ORFs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "da9223db",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    6165\n",
       "0     524\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr['valid_ORFs'].value_counts()\n",
    "# so valid_ORFs in gene does count itself and I should look carefully at the 439 genes with no valid ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "id": "fba3c9ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6565"
      ]
     },
     "execution_count": 301,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# number of annotated genes without potential duplicates\n",
    "# Doesn't works, we want duplicates\n",
    "len(set(gene['Name']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 332,
   "id": "7526d8ec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(6691, 23)"
      ]
     },
     "execution_count": 332,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "a2feefd3",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_attr = TE_df.attributes_to_columns()\n",
    "sno_attr = sno_df.attributes_to_columns()\n",
    "tRNA_attr = tRNA_df.attributes_to_columns()\n",
    "LTR_retroP_attr = LTR_retroP_df.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "12d4dea2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'copy_num_ID', 'coverage',\n",
       "       'curie', 'dbxref', 'display', 'extra_copy_number', 'low_identity',\n",
       "       'sequence_ID'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LTR_retroP_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "8d70a21e",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE = TE_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'sequence_ID', 'valid_ORFs']]\n",
    "\n",
    "sno = sno_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'sequence_ID']]\n",
    "\n",
    "tRNA = tRNA_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'gene', 'sequence_ID']]\n",
    "\n",
    "LTR_retroP = LTR_retroP_attr[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'score', 'strand', 'phase',\n",
    "       'copy_num_ID', 'coverage', 'curie', 'display', 'extra_copy_number', 'sequence_ID']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "4b9d5296",
   "metadata": {},
   "outputs": [],
   "source": [
    "# This code just convert the chromosome numbering to integer and remove mitochondrial genes.\n",
    "def lat_to_arabic(DF):\n",
    "    Latin_to_arabic = {'I': 1, 'II':2, 'III':3, 'IV':4, 'V':5, 'VI':6, 'VII':7, 'VIII':8, 'IX':9, 'X':10,\n",
    "                      'XI':11, 'XII':12, 'XIII':13, 'XIV':14, 'XV':15, 'XVI':16, 'Mito': 'mitoch', 'MANY': 7}\n",
    "    seq_id = DF['seq_id']\n",
    "    ch = []\n",
    "    contig = []\n",
    "    for i in seq_id:\n",
    "        chrom, cont = i.split('_')\n",
    "        contig.append(cont)\n",
    "        chrom = chrom.replace('chr', '')\n",
    "        if '-' in chrom:\n",
    "            chrom, e = chrom.split('-')\n",
    "        ch.append(Latin_to_arabic[chrom])\n",
    "    DF['Chromosome'] = ch\n",
    "    DF['Contig'] = contig\n",
    "    return DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "47a1360f",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:15: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\3764198105.py:16: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Contig'] = contig\n"
     ]
    }
   ],
   "source": [
    "gene = lat_to_arabic(gene)\n",
    "TE = lat_to_arabic(TE)\n",
    "sno = lat_to_arabic(sno)\n",
    "tRNA = lat_to_arabic(tRNA)\n",
    "LTR_retroP = lat_to_arabic(LTR_retroP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "88d25b7d",
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean20(DF):\n",
    "    note = DF['Note']\n",
    "    dis = DF['display']\n",
    "    clean = []\n",
    "    display = []\n",
    "    for n in note:\n",
    "        clean.append(n.replace('%20', ' '))\n",
    "    for d in dis:\n",
    "        display.append(str(d).replace('%20', ' '))\n",
    "    DF['Note'] = clean\n",
    "    DF['display'] = display\n",
    "    return DF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "50940e10",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\741321415.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Note'] = clean\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\741321415.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['display'] = display\n"
     ]
    }
   ],
   "source": [
    "gene = clean20(gene)\n",
    "TE = clean20(TE)\n",
    "sno = clean20(sno)\n",
    "tRNA = clean20(tRNA)\n",
    "LTR_retroP = clean20(LTR_retroP)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "259832d6",
   "metadata": {},
   "source": [
    "## Df by attribute for S288C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "685b570a",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE_df_s288c = annot_s288c.filter_feature_of_type(['transposable_element_gene'])\n",
    "sno_df_s288c = annot_s288c.filter_feature_of_type(['snoRNA_gene'])\n",
    "tRNA_df_s288c = annot_s288c.filter_feature_of_type(['tRNA_gene'])\n",
    "gene_df_s288c = annot_s288c.filter_feature_of_type(['gene'])\n",
    "LTR_retroP_df_s288c = annot_s288c.filter_feature_of_type(['LTR_retrotransposon'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "98713fc0",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_attr_s288c = gene_df_s288c.attributes_to_columns()\n",
    "TE_attr_s288c = TE_df_s288c.attributes_to_columns()\n",
    "sno_attr_s288c = sno_df_s288c.attributes_to_columns()\n",
    "tRNA_attr_s288c = tRNA_df_s288c.attributes_to_columns()\n",
    "LTR_retroP_attr_s288c = LTR_retroP_df_s288c.attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "4dc8b278",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'curie', 'dbxref',\n",
       "       'display'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LTR_retroP_attr_s288c.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "a65bdbc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_s288c = gene_attr_s288c[['seq_id', 'start', 'end', 'Name', 'gene',  'Alias',  'Note', 'Ontology_term', 'strand', 'phase', 'curie',\n",
    "        'orf_classification', 'display']]\n",
    "\n",
    "TE_s288c = TE_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "sno_s288c = sno_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "tRNA_s288c = tRNA_attr_s288c[['seq_id', 'start', 'end', 'Name',  'Alias',  'Note', 'Ontology_term', 'score', 'strand', 'phase',\n",
    "      'curie', 'display']]\n",
    "\n",
    "LTR_retroP_s288c = LTR_retroP_attr_s288c[['seq_id', 'start', 'end',  'ID', 'Name',  'Alias', 'score', 'strand', 'phase',\n",
    "       'attributes', 'Note', 'curie', 'display']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "d230909b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def lat_to_arabic_S288(df):\n",
    "    # This code just convert the chromosome numbering to integer and remove mitochondrial genes.\n",
    "    Latin_to_arabic = {'I': 1, 'II':2, 'III':3, 'IV':4, 'V':5, 'VI':6, 'VII':7, 'VIII':8, 'IX':9, 'X':10,\n",
    "                      'XI':11, 'XII':12, 'XIII':13, 'XIV':14, 'XV':15, 'XVI':16, 'Mito': 'mitoch'}\n",
    "    chLat = df['seq_id']\n",
    "    ch = []\n",
    "    for chrom in chLat:\n",
    "        chrom = chrom.replace('chr', '')\n",
    "        ch.append(Latin_to_arabic[chrom])\n",
    "    df['Chromosome'] = ch\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "4ef36a0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\760545625.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['Chromosome'] = ch\n"
     ]
    }
   ],
   "source": [
    "gene_s288c = lat_to_arabic_S288(gene_s288c)\n",
    "TE_s288c = lat_to_arabic_S288(TE_s288c)\n",
    "sno_s288c = lat_to_arabic_S288(sno_s288c)\n",
    "tRNA_s288c = lat_to_arabic_S288(tRNA_s288c)\n",
    "LTR_retroP_s288c = lat_to_arabic_S288(LTR_retroP_s288c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "ea5b328c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\741321415.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['Note'] = clean\n",
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\741321415.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  DF['display'] = display\n"
     ]
    }
   ],
   "source": [
    "gene_s288c = clean20(gene_s288c)\n",
    "TE_s288c = clean20(TE_s288c)\n",
    "sno_s288c = clean20(sno_s288c)\n",
    "tRNA_s288c = clean20(tRNA_s288c)\n",
    "LTR_retroP_s288c = clean20(LTR_retroP_s288c)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ba9450b",
   "metadata": {},
   "source": [
    "# TE \n",
    "\n",
    "### IMPORTANT:\n",
    "\n",
    "There are several TE mapped to the exact same location. Probably because map to 2 genes that have a very similar sequence!\n",
    "That's bad, I need to remove them. DO MANUALLY. --> Only 17 unique TE found!!!!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "id": "cd662573",
   "metadata": {},
   "outputs": [],
   "source": [
    "TE.to_excel('TE_liftoff_v5_yps1009.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cbb09b1c",
   "metadata": {},
   "source": [
    "## LTR Retrotransposons"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "e41ea165",
   "metadata": {},
   "outputs": [],
   "source": [
    "LTR_RP_count = LTR_retroP['Chromosome'].value_counts().to_frame()\n",
    "LTR_RP_count = LTR_RP_count.rename(columns = {'Chromosome' : 'yps1009 LTR retrotransposon'})\n",
    "LTR_RP_count_s288 = LTR_retroP_s288c['Chromosome'].value_counts().to_frame()\n",
    "LTR_RP_count_s288 = LTR_RP_count_s288.rename(columns= {'Chromosome': 'S288C LTR retroposon'})\n",
    "LTR_RP_count = LTR_RP_count_s288.join(LTR_RP_count)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b102e3b1",
   "metadata": {},
   "source": [
    "# IMPORTANT \n",
    "Overlapping features, to filter manually "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "id": "dc2f134c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>curie</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6374</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177542</td>\n",
       "      <td>183707</td>\n",
       "      <td>YDRWTy2-3</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy2-3_0</td>\n",
       "      <td>0.862</td>\n",
       "      <td>SGD:S000006885</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.835</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6375</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177544</td>\n",
       "      <td>183707</td>\n",
       "      <td>YGRWTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRWTy2-2_0</td>\n",
       "      <td>0.861</td>\n",
       "      <td>SGD:S000006983</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.834</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6376</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>177545</td>\n",
       "      <td>183707</td>\n",
       "      <td>YDRWTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy2-2_0</td>\n",
       "      <td>0.860</td>\n",
       "      <td>SGD:S000006884</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>423</td>\n",
       "      <td>4934</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_1</td>\n",
       "      <td>0.993</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.948</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>790</td>\n",
       "      <td>4554</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_0</td>\n",
       "      <td>0.997</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.954</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>407</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>136337</td>\n",
       "      <td>142463</td>\n",
       "      <td>YBLWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YBLWTy2-1_0</td>\n",
       "      <td>0.856</td>\n",
       "      <td>SGD:S000006809</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.823</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4750</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297667</td>\n",
       "      <td>1303627</td>\n",
       "      <td>YLRWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRWTy2-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007100</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.812</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4751</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297674</td>\n",
       "      <td>1309617</td>\n",
       "      <td>YNLWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YNLWTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007142</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.426</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4752</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1303627</td>\n",
       "      <td>YDRCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy2-1_0</td>\n",
       "      <td>0.832</td>\n",
       "      <td>SGD:S000006862</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.812</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4753</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1304036</td>\n",
       "      <td>YFLWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YFLWTy2-1_0</td>\n",
       "      <td>0.866</td>\n",
       "      <td>SGD:S000006930</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.826</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4754</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1303630</td>\n",
       "      <td>YNLCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YNLCTy2-1_0</td>\n",
       "      <td>0.935</td>\n",
       "      <td>SGD:S000007135</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.913</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4755</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1297677</td>\n",
       "      <td>1303627</td>\n",
       "      <td>YORWTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YORWTy2-2_0</td>\n",
       "      <td>0.831</td>\n",
       "      <td>SGD:S000007183</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.810</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4806</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1303049</td>\n",
       "      <td>1309617</td>\n",
       "      <td>YJRWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YJRWTy1-1_0</td>\n",
       "      <td>0.864</td>\n",
       "      <td>SGD:S000007048</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.757</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4808</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1303305</td>\n",
       "      <td>1309617</td>\n",
       "      <td>YPLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YPLWTy1-1_0</td>\n",
       "      <td>0.877</td>\n",
       "      <td>SGD:S000007199</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.808</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12092</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>543496</td>\n",
       "      <td>548928</td>\n",
       "      <td>YILWTy3-1</td>\n",
       "      <td>Ty3</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YILWTy3-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007020</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>5</td>\n",
       "      <td>tig00000015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12097</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>543497</td>\n",
       "      <td>548928</td>\n",
       "      <td>YGRWTy3-1</td>\n",
       "      <td>Ty3</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRWTy3-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000006984</td>\n",
       "      <td>Ty3 element%2C LTR retrotransposon of the Gyps...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.818</td>\n",
       "      <td>5</td>\n",
       "      <td>tig00000015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6975</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106301</td>\n",
       "      <td>112481</td>\n",
       "      <td>YERCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YERCTy1-1_0</td>\n",
       "      <td>0.868</td>\n",
       "      <td>SGD:S000006910</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.807</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6980</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106477</td>\n",
       "      <td>112460</td>\n",
       "      <td>YMRCTy1-4</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YMRCTy1-4_0</td>\n",
       "      <td>0.837</td>\n",
       "      <td>SGD:S000007120</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.790</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6981</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106488</td>\n",
       "      <td>112460</td>\n",
       "      <td>YDRCTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy1-3_0</td>\n",
       "      <td>0.839</td>\n",
       "      <td>SGD:S000006861</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.733</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6982</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106565</td>\n",
       "      <td>112460</td>\n",
       "      <td>YDRCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy1-1_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000006859</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.757</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6984</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106565</td>\n",
       "      <td>112485</td>\n",
       "      <td>YPRCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YPRCTy1-2_0</td>\n",
       "      <td>0.831</td>\n",
       "      <td>SGD:S000007208</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.761</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6986</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106566</td>\n",
       "      <td>112460</td>\n",
       "      <td>YBRWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YBRWTy1-2_0</td>\n",
       "      <td>0.829</td>\n",
       "      <td>SGD:S000006821</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.792</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6987</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106567</td>\n",
       "      <td>112460</td>\n",
       "      <td>YPRWTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YPRWTy1-3_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000007220</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.776</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6989</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YDRWTy1-4</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy1-4_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000006882</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.776</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6990</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YGRCTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRCTy1-3_0</td>\n",
       "      <td>0.829</td>\n",
       "      <td>SGD:S000006963</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.793</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6991</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YHRCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YHRCTy1-1_0</td>\n",
       "      <td>0.813</td>\n",
       "      <td>SGD:S000007006</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.769</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6992</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112459</td>\n",
       "      <td>YMLWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YMLWTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007108</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.771</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6995</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106569</td>\n",
       "      <td>112460</td>\n",
       "      <td>YORWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YORWTy1-2_0</td>\n",
       "      <td>0.829</td>\n",
       "      <td>SGD:S000007182</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.787</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6996</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YARCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YARCTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006792</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.769</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6998</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112481</td>\n",
       "      <td>YDRWTy1-5</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRWTy1-5_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006883</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.766</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7001</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112460</td>\n",
       "      <td>YGRCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRCTy1-2_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000006962</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.759</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7002</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YGRWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRWTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006982</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.793</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7004</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YJRWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YJRWTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007049</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.787</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7006</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YLRCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRCTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007081</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.772</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7008</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112460</td>\n",
       "      <td>YLRWTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRWTy1-3_0</td>\n",
       "      <td>0.827</td>\n",
       "      <td>SGD:S000007099</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.763</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7010</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112460</td>\n",
       "      <td>YMLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YMLWTy1-1_0</td>\n",
       "      <td>0.828</td>\n",
       "      <td>SGD:S000007107</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.778</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7011</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112481</td>\n",
       "      <td>YNLCTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YNLCTy1-1_0</td>\n",
       "      <td>0.832</td>\n",
       "      <td>SGD:S000007134</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.814</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7013</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YOLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YOLWTy1-1_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007161</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.777</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7015</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>106570</td>\n",
       "      <td>112480</td>\n",
       "      <td>YPRCTy1-4</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YPRCTy1-4_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000007209</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.779</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7397</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>212802</td>\n",
       "      <td>219077</td>\n",
       "      <td>YHLWTy4-1</td>\n",
       "      <td>Ty4</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YHLWTy4-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000006991</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.821</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7399</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>212802</td>\n",
       "      <td>219075</td>\n",
       "      <td>YJLWTy4-1</td>\n",
       "      <td>Ty4</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YJLWTy4-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007039</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.819</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7401</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>212802</td>\n",
       "      <td>219075</td>\n",
       "      <td>YPLCTy4-1</td>\n",
       "      <td>Ty4</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YPLCTy4-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007188</td>\n",
       "      <td>Ty4 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.821</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8302</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>376739</td>\n",
       "      <td>382705</td>\n",
       "      <td>YORCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YORCTy2-1_0</td>\n",
       "      <td>0.834</td>\n",
       "      <td>SGD:S000007168</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.815</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12205</th>\n",
       "      <td>chrXI-2_tig00004931</td>\n",
       "      <td>24798</td>\n",
       "      <td>29311</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_3</td>\n",
       "      <td>0.993</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>3</td>\n",
       "      <td>0.945</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18660</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>664456</td>\n",
       "      <td>668970</td>\n",
       "      <td>YCLWTy5-1</td>\n",
       "      <td>Ty5</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy5-1_2</td>\n",
       "      <td>0.994</td>\n",
       "      <td>SGD:S000006831</td>\n",
       "      <td>Ty5 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>2</td>\n",
       "      <td>0.945</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12925</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414716</td>\n",
       "      <td>420627</td>\n",
       "      <td>YBLWTy1-1</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YBLWTy1-1_0</td>\n",
       "      <td>0.832</td>\n",
       "      <td>SGD:S000006808</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.826</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12926</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414716</td>\n",
       "      <td>420627</td>\n",
       "      <td>YMRCTy1-3</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YMRCTy1-3_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000007119</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.825</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12927</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414736</td>\n",
       "      <td>420627</td>\n",
       "      <td>YDRCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YDRCTy1-2_0</td>\n",
       "      <td>0.826</td>\n",
       "      <td>SGD:S000006860</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.755</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12928</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>414910</td>\n",
       "      <td>420627</td>\n",
       "      <td>YERCTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YERCTy1-2_0</td>\n",
       "      <td>0.830</td>\n",
       "      <td>SGD:S000006911</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.791</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21161</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>112119</td>\n",
       "      <td>118082</td>\n",
       "      <td>YGRCTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YGRCTy2-1_0</td>\n",
       "      <td>0.833</td>\n",
       "      <td>SGD:S000006964</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.813</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21162</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>112119</td>\n",
       "      <td>118084</td>\n",
       "      <td>YLRCTy2-2</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRCTy2-2_0</td>\n",
       "      <td>0.834</td>\n",
       "      <td>SGD:S000007082</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.751</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21359</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>228916</td>\n",
       "      <td>235127</td>\n",
       "      <td>YLRWTy1-2</td>\n",
       "      <td>Ty1</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>YLRWTy1-2_0</td>\n",
       "      <td>0.871</td>\n",
       "      <td>SGD:S000007098</td>\n",
       "      <td>Ty1 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.793</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19303</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>54637</td>\n",
       "      <td>60826</td>\n",
       "      <td>YCLWTy2-1</td>\n",
       "      <td>Ty2</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>YCLWTy2-1_0</td>\n",
       "      <td>0.865</td>\n",
       "      <td>SGD:S000006830</td>\n",
       "      <td>Ty2 element%2C LTR retrotransposon of the Copi...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.843</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000040</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     seq_id    start      end       Name Alias  \\\n",
       "6374       chrI_tig00000033   177542   183707  YDRWTy2-3   Ty2   \n",
       "6375       chrI_tig00000033   177544   183707  YGRWTy2-2   Ty2   \n",
       "6376       chrI_tig00000033   177545   183707  YDRWTy2-2   Ty2   \n",
       "37     chrIII-2_tig00004935      423     4934  YCLWTy5-1   Ty5   \n",
       "57       chrIII_tig00004936      790     4554  YCLWTy5-1   Ty5   \n",
       "407      chrIII_tig00004936   136337   142463  YBLWTy2-1   Ty2   \n",
       "4750      chrIV_tig00000001  1297667  1303627  YLRWTy2-1   Ty2   \n",
       "4751      chrIV_tig00000001  1297674  1309617  YNLWTy1-2   Ty1   \n",
       "4752      chrIV_tig00000001  1297677  1303627  YDRCTy2-1   Ty2   \n",
       "4753      chrIV_tig00000001  1297677  1304036  YFLWTy2-1   Ty2   \n",
       "4754      chrIV_tig00000001  1297677  1303630  YNLCTy2-1   Ty2   \n",
       "4755      chrIV_tig00000001  1297677  1303627  YORWTy2-2   Ty2   \n",
       "4806      chrIV_tig00000001  1303049  1309617  YJRWTy1-1   Ty1   \n",
       "4808      chrIV_tig00000001  1303305  1309617  YPLWTy1-1   Ty1   \n",
       "12092      chrV_tig00000015   543496   548928  YILWTy3-1   Ty3   \n",
       "12097      chrV_tig00000015   543497   548928  YGRWTy3-1   Ty3   \n",
       "6975   chrVII-2_tig00000038   106301   112481  YERCTy1-1   Ty1   \n",
       "6980   chrVII-2_tig00000038   106477   112460  YMRCTy1-4   Ty1   \n",
       "6981   chrVII-2_tig00000038   106488   112460  YDRCTy1-3   Ty1   \n",
       "6982   chrVII-2_tig00000038   106565   112460  YDRCTy1-1   Ty1   \n",
       "6984   chrVII-2_tig00000038   106565   112485  YPRCTy1-2   Ty1   \n",
       "6986   chrVII-2_tig00000038   106566   112460  YBRWTy1-2   Ty1   \n",
       "6987   chrVII-2_tig00000038   106567   112460  YPRWTy1-3   Ty1   \n",
       "6989   chrVII-2_tig00000038   106569   112460  YDRWTy1-4   Ty1   \n",
       "6990   chrVII-2_tig00000038   106569   112460  YGRCTy1-3   Ty1   \n",
       "6991   chrVII-2_tig00000038   106569   112460  YHRCTy1-1   Ty1   \n",
       "6992   chrVII-2_tig00000038   106569   112459  YMLWTy1-2   Ty1   \n",
       "6995   chrVII-2_tig00000038   106569   112460  YORWTy1-2   Ty1   \n",
       "6996   chrVII-2_tig00000038   106570   112480  YARCTy1-1   Ty1   \n",
       "6998   chrVII-2_tig00000038   106570   112481  YDRWTy1-5   Ty1   \n",
       "7001   chrVII-2_tig00000038   106570   112460  YGRCTy1-2   Ty1   \n",
       "7002   chrVII-2_tig00000038   106570   112480  YGRWTy1-1   Ty1   \n",
       "7004   chrVII-2_tig00000038   106570   112480  YJRWTy1-2   Ty1   \n",
       "7006   chrVII-2_tig00000038   106570   112480  YLRCTy1-1   Ty1   \n",
       "7008   chrVII-2_tig00000038   106570   112460  YLRWTy1-3   Ty1   \n",
       "7010   chrVII-2_tig00000038   106570   112460  YMLWTy1-1   Ty1   \n",
       "7011   chrVII-2_tig00000038   106570   112481  YNLCTy1-1   Ty1   \n",
       "7013   chrVII-2_tig00000038   106570   112480  YOLWTy1-1   Ty1   \n",
       "7015   chrVII-2_tig00000038   106570   112480  YPRCTy1-4   Ty1   \n",
       "7397   chrVII-2_tig00000038   212802   219077  YHLWTy4-1   Ty4   \n",
       "7399   chrVII-2_tig00000038   212802   219075  YJLWTy4-1   Ty4   \n",
       "7401   chrVII-2_tig00000038   212802   219075  YPLCTy4-1   Ty4   \n",
       "8302    chrVIII_tig00000017   376739   382705  YORCTy2-1   Ty2   \n",
       "12205   chrXI-2_tig00004931    24798    29311  YCLWTy5-1   Ty5   \n",
       "18660     chrXI_tig00004932   664456   668970  YCLWTy5-1   Ty5   \n",
       "12925  chrXII-2_tig00004934   414716   420627  YBLWTy1-1   Ty1   \n",
       "12926  chrXII-2_tig00004934   414716   420627  YMRCTy1-3   Ty1   \n",
       "12927  chrXII-2_tig00004934   414736   420627  YDRCTy1-2   Ty1   \n",
       "12928  chrXII-2_tig00004934   414910   420627  YERCTy1-2   Ty1   \n",
       "21161     chrXV_tig00000007   112119   118082  YGRCTy2-1   Ty2   \n",
       "21162     chrXV_tig00000007   112119   118084  YLRCTy2-2   Ty2   \n",
       "21359     chrXV_tig00000007   228916   235127  YLRWTy1-2   Ty1   \n",
       "19303  chrXVI-2_tig00000040    54637    60826  YCLWTy2-1   Ty2   \n",
       "\n",
       "                                                    Note score strand phase  \\\n",
       "6374   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6375   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6376   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "37     Ty5 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "57     Ty5 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "407    Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4750   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4751   Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4752   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4753   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4754   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4755   Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4806   Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "4808   Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12092  Ty3 element%2C LTR retrotransposon of the Gyps...     .      +     .   \n",
       "12097  Ty3 element%2C LTR retrotransposon of the Gyps...     .      +     .   \n",
       "6975   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6980   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6981   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6982   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6984   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6986   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6987   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6989   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6990   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6991   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6992   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6995   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6996   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "6998   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7001   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7002   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7004   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7006   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7008   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7010   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7011   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7013   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7015   Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "7397   Ty4 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "7399   Ty4 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "7401   Ty4 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "8302   Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "12205  Ty5 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "18660  Ty5 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12925  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12926  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12927  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "12928  Ty1 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "21161  Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "21162  Ty2 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "21359  Ty1 element%2C LTR retrotransposon of the Copi...     .      +     .   \n",
       "19303  Ty2 element%2C LTR retrotransposon of the Copi...     .      -     .   \n",
       "\n",
       "       copy_num_ID coverage           curie  \\\n",
       "6374   YDRWTy2-3_0    0.862  SGD:S000006885   \n",
       "6375   YGRWTy2-2_0    0.861  SGD:S000006983   \n",
       "6376   YDRWTy2-2_0    0.860  SGD:S000006884   \n",
       "37     YCLWTy5-1_1    0.993  SGD:S000006831   \n",
       "57     YCLWTy5-1_0    0.997  SGD:S000006831   \n",
       "407    YBLWTy2-1_0    0.856  SGD:S000006809   \n",
       "4750   YLRWTy2-1_0    0.833  SGD:S000007100   \n",
       "4751   YNLWTy1-2_0    0.830  SGD:S000007142   \n",
       "4752   YDRCTy2-1_0    0.832  SGD:S000006862   \n",
       "4753   YFLWTy2-1_0    0.866  SGD:S000006930   \n",
       "4754   YNLCTy2-1_0    0.935  SGD:S000007135   \n",
       "4755   YORWTy2-2_0    0.831  SGD:S000007183   \n",
       "4806   YJRWTy1-1_0    0.864  SGD:S000007048   \n",
       "4808   YPLWTy1-1_0    0.877  SGD:S000007199   \n",
       "12092  YILWTy3-1_0    0.833  SGD:S000007020   \n",
       "12097  YGRWTy3-1_0    0.833  SGD:S000006984   \n",
       "6975   YERCTy1-1_0    0.868  SGD:S000006910   \n",
       "6980   YMRCTy1-4_0    0.837  SGD:S000007120   \n",
       "6981   YDRCTy1-3_0    0.839  SGD:S000006861   \n",
       "6982   YDRCTy1-1_0    0.827  SGD:S000006859   \n",
       "6984   YPRCTy1-2_0    0.831  SGD:S000007208   \n",
       "6986   YBRWTy1-2_0    0.829  SGD:S000006821   \n",
       "6987   YPRWTy1-3_0    0.827  SGD:S000007220   \n",
       "6989   YDRWTy1-4_0    0.827  SGD:S000006882   \n",
       "6990   YGRCTy1-3_0    0.829  SGD:S000006963   \n",
       "6991   YHRCTy1-1_0    0.813  SGD:S000007006   \n",
       "6992   YMLWTy1-2_0    0.830  SGD:S000007108   \n",
       "6995   YORWTy1-2_0    0.829  SGD:S000007182   \n",
       "6996   YARCTy1-1_0    0.830  SGD:S000006792   \n",
       "6998   YDRWTy1-5_0    0.830  SGD:S000006883   \n",
       "7001   YGRCTy1-2_0    0.827  SGD:S000006962   \n",
       "7002   YGRWTy1-1_0    0.830  SGD:S000006982   \n",
       "7004   YJRWTy1-2_0    0.830  SGD:S000007049   \n",
       "7006   YLRCTy1-1_0    0.830  SGD:S000007081   \n",
       "7008   YLRWTy1-3_0    0.827  SGD:S000007099   \n",
       "7010   YMLWTy1-1_0    0.828  SGD:S000007107   \n",
       "7011   YNLCTy1-1_0    0.832  SGD:S000007134   \n",
       "7013   YOLWTy1-1_0    0.830  SGD:S000007161   \n",
       "7015   YPRCTy1-4_0    0.830  SGD:S000007209   \n",
       "7397   YHLWTy4-1_0    0.833  SGD:S000006991   \n",
       "7399   YJLWTy4-1_0    0.833  SGD:S000007039   \n",
       "7401   YPLCTy4-1_0    0.833  SGD:S000007188   \n",
       "8302   YORCTy2-1_0    0.834  SGD:S000007168   \n",
       "12205  YCLWTy5-1_3    0.993  SGD:S000006831   \n",
       "18660  YCLWTy5-1_2    0.994  SGD:S000006831   \n",
       "12925  YBLWTy1-1_0    0.832  SGD:S000006808   \n",
       "12926  YMRCTy1-3_0    0.833  SGD:S000007119   \n",
       "12927  YDRCTy1-2_0    0.826  SGD:S000006860   \n",
       "12928  YERCTy1-2_0    0.830  SGD:S000006911   \n",
       "21161  YGRCTy2-1_0    0.833  SGD:S000006964   \n",
       "21162  YLRCTy2-2_0    0.834  SGD:S000007082   \n",
       "21359  YLRWTy1-2_0    0.871  SGD:S000007098   \n",
       "19303  YCLWTy2-1_0    0.865  SGD:S000006830   \n",
       "\n",
       "                                                 display extra_copy_number  \\\n",
       "6374   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6375   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6376   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "37     Ty5 element%2C LTR retrotransposon of the Copi...                 1   \n",
       "57     Ty5 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "407    Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4750   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4751   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4752   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4753   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4754   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4755   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4806   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "4808   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12092  Ty3 element%2C LTR retrotransposon of the Gyps...                 0   \n",
       "12097  Ty3 element%2C LTR retrotransposon of the Gyps...                 0   \n",
       "6975   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6980   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6981   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6982   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6984   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6986   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6987   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6989   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6990   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6991   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6992   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6995   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6996   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "6998   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7001   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7002   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7004   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7006   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7008   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7010   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7011   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7013   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7015   Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7397   Ty4 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7399   Ty4 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "7401   Ty4 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "8302   Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12205  Ty5 element%2C LTR retrotransposon of the Copi...                 3   \n",
       "18660  Ty5 element%2C LTR retrotransposon of the Copi...                 2   \n",
       "12925  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12926  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12927  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "12928  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "21161  Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "21162  Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "21359  Ty1 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "19303  Ty2 element%2C LTR retrotransposon of the Copi...                 0   \n",
       "\n",
       "      sequence_ID  Chromosome       Contig  \n",
       "6374        0.835           1  tig00000033  \n",
       "6375        0.834           1  tig00000033  \n",
       "6376        0.833           1  tig00000033  \n",
       "37          0.948           3  tig00004935  \n",
       "57          0.954           3  tig00004936  \n",
       "407         0.823           3  tig00004936  \n",
       "4750        0.812           4  tig00000001  \n",
       "4751        0.426           4  tig00000001  \n",
       "4752        0.812           4  tig00000001  \n",
       "4753        0.826           4  tig00000001  \n",
       "4754        0.913           4  tig00000001  \n",
       "4755        0.810           4  tig00000001  \n",
       "4806        0.757           4  tig00000001  \n",
       "4808        0.808           4  tig00000001  \n",
       "12092       0.827           5  tig00000015  \n",
       "12097       0.818           5  tig00000015  \n",
       "6975        0.807           7  tig00000038  \n",
       "6980        0.790           7  tig00000038  \n",
       "6981        0.733           7  tig00000038  \n",
       "6982        0.757           7  tig00000038  \n",
       "6984        0.761           7  tig00000038  \n",
       "6986        0.792           7  tig00000038  \n",
       "6987        0.776           7  tig00000038  \n",
       "6989        0.776           7  tig00000038  \n",
       "6990        0.793           7  tig00000038  \n",
       "6991        0.769           7  tig00000038  \n",
       "6992        0.771           7  tig00000038  \n",
       "6995        0.787           7  tig00000038  \n",
       "6996        0.769           7  tig00000038  \n",
       "6998        0.766           7  tig00000038  \n",
       "7001        0.759           7  tig00000038  \n",
       "7002        0.793           7  tig00000038  \n",
       "7004        0.787           7  tig00000038  \n",
       "7006        0.772           7  tig00000038  \n",
       "7008        0.763           7  tig00000038  \n",
       "7010        0.778           7  tig00000038  \n",
       "7011        0.814           7  tig00000038  \n",
       "7013        0.777           7  tig00000038  \n",
       "7015        0.779           7  tig00000038  \n",
       "7397        0.821           7  tig00000038  \n",
       "7399        0.819           7  tig00000038  \n",
       "7401        0.821           7  tig00000038  \n",
       "8302        0.815           8  tig00000017  \n",
       "12205       0.945          11  tig00004931  \n",
       "18660       0.945          11  tig00004932  \n",
       "12925       0.826          12  tig00004934  \n",
       "12926       0.825          12  tig00004934  \n",
       "12927       0.755          12  tig00004934  \n",
       "12928       0.791          12  tig00004934  \n",
       "21161       0.813          15  tig00000007  \n",
       "21162       0.751          15  tig00000007  \n",
       "21359       0.793          15  tig00000007  \n",
       "19303       0.843          16  tig00000040  "
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LTR_retroP.sort_values(by=['Chromosome', 'start'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "09d2206b",
   "metadata": {},
   "outputs": [],
   "source": [
    "LTR_retroP.to_excel('Liftoff_LTR_retrop.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b245257b",
   "metadata": {},
   "source": [
    "How do I reconciliate these results? it somehow feels not right\n",
    "compare to the TE Ultimate results. See TE_annotation jupyter notebook "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "id": "77ade57e",
   "metadata": {},
   "outputs": [],
   "source": [
    "filteredRetroP = pd.read_excel('Liftoff_LTR_retrop_filtered.xlsx', index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 268,
   "id": "037367e7",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = filteredRetroP['Chromosome'].value_counts().to_frame()\n",
    "RetroP_count = RetroP_count.rename(columns = {'Chromosome': 'yps1009 RetroP'})\n",
    "RetroP_count_s = LTR_retroP_s288c['Chromosome'].value_counts().to_frame()\n",
    "RetroP_count_s = RetroP_count_s.rename(columns = {'Chromosome': 'S288C RetroP'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 271,
   "id": "c51ec222",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = pd.concat((RetroP_count_s, RetroP_count), axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 272,
   "id": "a2faa007",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C RetroP</th>\n",
       "      <th>yps1009 RetroP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C RetroP  yps1009 RetroP\n",
       "1            1.0             1.0\n",
       "2            3.0             NaN\n",
       "3            2.0             3.0\n",
       "4            8.0             2.0\n",
       "5            2.0             1.0\n",
       "6            1.0             NaN\n",
       "7            6.0             2.0\n",
       "8            2.0             1.0\n",
       "9            1.0             NaN\n",
       "10           3.0             NaN\n",
       "11           NaN             2.0\n",
       "12           5.0             1.0\n",
       "13           4.0             NaN\n",
       "14           3.0             NaN\n",
       "15           4.0             2.0\n",
       "16           5.0             1.0"
      ]
     },
     "execution_count": 272,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "id": "3059b9a2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S288C RetroP      50.0\n",
       "yps1009 RetroP    16.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6f48ebcb",
   "metadata": {},
   "source": [
    "# Compared to the filtered ReasonaTE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "id": "f2f195f1",
   "metadata": {},
   "outputs": [],
   "source": [
    "Reasona = pd.read_csv('ReasonaTE_filteredblast.txt', sep = '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "e51c3306",
   "metadata": {},
   "outputs": [],
   "source": [
    "Reasona_count = Reasona['chromosome'].value_counts().to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "d6dee124",
   "metadata": {},
   "outputs": [],
   "source": [
    "Reasona_count.rename(columns={'chromosome': 'ReasonaTE yps RetroP'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "e39c507b",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = RetroP_count.join(Reasona_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "id": "58cdb2a5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C RetroP</th>\n",
       "      <th>yps1009 RetroP</th>\n",
       "      <th>ReasonaTE yps RetroP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C RetroP  yps1009 RetroP  ReasonaTE yps RetroP\n",
       "4            8.0             2.0                   4.0\n",
       "7            6.0             2.0                   2.0\n",
       "12           5.0             1.0                   1.0\n",
       "16           5.0             1.0                   1.0\n",
       "13           4.0             NaN                   NaN\n",
       "15           4.0             2.0                   2.0\n",
       "2            3.0             NaN                   1.0\n",
       "10           3.0             NaN                   1.0\n",
       "14           3.0             NaN                   NaN\n",
       "3            2.0             3.0                   1.0\n",
       "5            2.0             1.0                   2.0\n",
       "8            2.0             1.0                   1.0\n",
       "1            1.0             1.0                   2.0\n",
       "6            1.0             NaN                   2.0\n",
       "9            1.0             NaN                   NaN\n",
       "11           NaN             2.0                   NaN"
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "id": "9163115b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S288C RetroP            50.0\n",
       "yps1009 RetroP          16.0\n",
       "ReasonaTE yps RetroP    20.0\n",
       "dtype: float64"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 286,
   "id": "02e67922",
   "metadata": {},
   "outputs": [],
   "source": [
    "combiTE = pd.read_csv('TE_liftoffv5_reasonaTE_combined.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "750009e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "combi_count = combiTE['chromosome'].value_counts().to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "id": "1bbb0cfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "combi_count.rename(columns={'chromosome': 'Reasona+Liftoff'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "id": "a7c9d0fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "RetroP_count = RetroP_count.join(combi_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "id": "7fcefaab",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C RetroP</th>\n",
       "      <th>yps1009 RetroP</th>\n",
       "      <th>ReasonaTE yps RetroP</th>\n",
       "      <th>Reasona+Liftoff</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C RetroP  yps1009 RetroP  ReasonaTE yps RetroP  Reasona+Liftoff\n",
       "1            1.0             1.0                   2.0              2.0\n",
       "2            3.0             NaN                   1.0              1.0\n",
       "3            2.0             3.0                   1.0              3.0\n",
       "4            8.0             2.0                   4.0              4.0\n",
       "5            2.0             1.0                   2.0              2.0\n",
       "6            1.0             NaN                   2.0              2.0\n",
       "7            6.0             2.0                   2.0              2.0\n",
       "8            2.0             1.0                   1.0              1.0\n",
       "9            1.0             NaN                   NaN              NaN\n",
       "10           3.0             NaN                   1.0              1.0\n",
       "11           NaN             2.0                   NaN              2.0\n",
       "12           5.0             1.0                   1.0              1.0\n",
       "13           4.0             NaN                   NaN              NaN\n",
       "14           3.0             NaN                   NaN              NaN\n",
       "15           4.0             2.0                   2.0              2.0\n",
       "16           5.0             1.0                   1.0              1.0"
      ]
     },
     "execution_count": 293,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RetroP_count.sort_index()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ec032b45",
   "metadata": {},
   "source": [
    "# snoRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "7726d381",
   "metadata": {},
   "outputs": [],
   "source": [
    "sno_count = sno['Chromosome'].value_counts().to_frame()\n",
    "sno_count = sno_count.rename(columns = {'Chromosome': 'yps1009 snoRNA'})\n",
    "sno_count_s = sno_s288c['Chromosome'].value_counts().to_frame()\n",
    "sno_count_s = sno_count_s.rename(columns = {'Chromosome': 'S288C snoRNA'})\n",
    "sno_count = sno_count_s.join(sno_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "0d0ff52b",
   "metadata": {},
   "outputs": [],
   "source": [
    "sno_count.to_excel('snoRNA_count_v5.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "5bd9c523",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>S288C snoRNA</th>\n",
       "      <th>yps1009 snoRNA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    S288C snoRNA  yps1009 snoRNA\n",
       "13            13              13\n",
       "15            11              11\n",
       "12             7               7\n",
       "7              6               6\n",
       "16             6               6\n",
       "5              5               5\n",
       "10             5               5\n",
       "11             5               5\n",
       "14             5               5\n",
       "3              4               4\n",
       "4              4               4\n",
       "2              2               2\n",
       "8              2               2\n",
       "1              1               1\n",
       "9              1               1"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sno_count"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "530e98aa",
   "metadata": {},
   "source": [
    "# tRNA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "id": "4962e83e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    299\n",
       "1     24\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "id": "bbef7664",
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_count = tRNA['Chromosome'].value_counts().to_frame()\n",
    "tRNA_count = tRNA_count.rename(columns = {'Chromosome': 'yps1009 tRNA'})\n",
    "tRNA_count_s = tRNA_s288c['Chromosome'].value_counts().to_frame()\n",
    "tRNA_count_s = tRNA_count_s.rename(columns = {'Chromosome': 'S288C tRNA'})\n",
    "tRNA_count = tRNA_count_s.join(tRNA_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "bda5f137",
   "metadata": {},
   "outputs": [],
   "source": [
    "tRNA_count.to_excel('tRNA_count_v5.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 319,
   "id": "5db14254",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(323, 19)"
      ]
     },
     "execution_count": 319,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 321,
   "id": "d5702ecd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(299, 13)"
      ]
     },
     "execution_count": 321,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tRNA_s288c.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8effbe52",
   "metadata": {},
   "source": [
    "# GENE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "c117842e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Verified           5250\n",
       "Uncharacterized     731\n",
       "Dubious             708\n",
       "Name: orf_classification, dtype: int64"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene['orf_classification'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "6b91315f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    518\n",
       "0    190\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dubious = gene[gene['orf_classification'] == 'Dubious']\n",
    "dubious['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "fd97eac3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    559\n",
       "0    172\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uncharacterized = gene[gene['orf_classification'] == 'Uncharacterized']\n",
    "uncharacterized['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 333,
   "id": "95381574",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_valid = gene[gene['valid_ORFs'] == '1']\n",
    "gene_unvalid = gene[gene['valid_ORFs'] == '0']\n",
    "\n",
    "gene_valid_count = gene_valid['Chromosome'].value_counts().to_frame()\n",
    "gene_valid_count = gene_valid_count.rename(columns={'Chromosome': 'Number valid genes'})\n",
    "gene_unvalid_count = gene_unvalid['Chromosome'].value_counts().to_frame()\n",
    "gene_unvalid_count = gene_unvalid_count.rename(columns={'Chromosome': 'Number unvalid genes'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 334,
   "id": "6a166057",
   "metadata": {},
   "outputs": [],
   "source": [
    "unval_dic = {}\n",
    "for ind, row in gene_unvalid.iterrows():\n",
    "    if row['Chromosome'] in unval_dic:\n",
    "        unval_dic[row['Chromosome']] = unval_dic[row['Chromosome']] + [row['Name']]\n",
    "    else:\n",
    "        unval_dic[row['Chromosome']] = [row['Name']]\n",
    "        \n",
    "gene_sum = gene_valid_count.join(gene_unvalid_count)\n",
    "tmp = []\n",
    "for ind in gene_sum.index:\n",
    "    tmp.append(unval_dic[ind])\n",
    "gene_sum['unvalid genes'] = tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 335,
   "id": "b48ac901",
   "metadata": {},
   "outputs": [],
   "source": [
    "missed = []\n",
    "moved = []\n",
    "count_miss = []\n",
    "count_move = []\n",
    "for Ch in gene_sum.index:\n",
    "    s288 = gene_s288c[gene_s288c['Chromosome'] == Ch]\n",
    "    yps = gene[gene['Chromosome'] == Ch]\n",
    "    tmp = list(set(s288['Name']) - set(yps['Name']))\n",
    "    miss = []\n",
    "    mov = []\n",
    "    others = gene[gene['Chromosome'] != Ch]\n",
    "    for g in tmp:\n",
    "        search = others[others['Name'] == g]\n",
    "        if search.shape[0] >= 1:\n",
    "            mov.append((g, list(search['Chromosome']), list(search['valid_ORFs'])))\n",
    "        else:\n",
    "            miss.append(g)  \n",
    "    moved.append(mov)\n",
    "    missed.append(miss)\n",
    "    count_miss.append(len(miss))\n",
    "    count_move.append(len(mov))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 336,
   "id": "ca916bcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_sum['Number of missing genes'] = count_miss\n",
    "gene_sum['Missing genes'] = missed\n",
    "gene_sum['Number of moved genes'] = count_move\n",
    "gene_sum['Moved genes, new chromosome(s), orf validity'] = moved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "id": "c6c2f892",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Number valid genes                                                                           6165\n",
       "Number unvalid genes                                                                          524\n",
       "unvalid genes                                   [YFL063W, YHL048C-A, YDL247W, YJR157W, YNL331C...\n",
       "Number of missing genes                                                                        42\n",
       "Missing genes                                   [YDR034C-A, YOR255W, YOR248W, YOR246C, YOR247W...\n",
       "Number of moved genes                                                                         142\n",
       "Moved genes, new chromosome(s), orf validity    [(YDR544C, [11], [0]), (YDR545W, [5], [1]), (Y...\n",
       "dtype: object"
      ]
     },
     "execution_count": 349,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_sum.sum()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 347,
   "id": "8d191a2b",
   "metadata": {},
   "outputs": [],
   "source": [
    "flat_list = [item for sublist in missed for item in sublist]\n",
    "missing_genes = gene_s288c[gene_s288c['Name'].isin(flat_list)]\n",
    "missing_genes.to_excel('missing_genes_v5.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "id": "28371a69",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Blast\n",
    "ids = []\n",
    "seq = []\n",
    "for seq_record in SeqIO.parse(\"fasta_file/orf_coding_all_R64-3-1_20210421.fasta\", \"fasta\"):\n",
    "    ids.append(str(seq_record.id))\n",
    "    seq.append(str(seq_record.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 345,
   "id": "96f3e5f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "for g in flat_list:\n",
    "    name = g + '.txt'\n",
    "    with open(name , \"w\") as file:\n",
    "        file.write('> ' + g + '\\n')\n",
    "        file.write(seq[ids.index(g)])\n",
    "    outfile = 'Blast_results/' + g + '.tab'\n",
    "    cmd_blastn = NcbiblastnCommandline(query= name, out= outfile, evalue=0.001, db= 'YPS1009_DB/YPS1009_DB')\n",
    "    stdout, stderr = cmd_blastn()\n",
    "    #print(\"STDOUT: %s\" % stdout)\n",
    "    #print(\"STDERR: %s\" % stderr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 346,
   "id": "b9efab82",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "42"
      ]
     },
     "execution_count": 346,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(flat_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8b062b2",
   "metadata": {},
   "source": [
    "# Verified ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "faddeacc",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified = gene[gene['orf_classification'] == 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "7b49021f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1    5088\n",
       "0     162\n",
       "Name: valid_ORFs, dtype: int64"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified['valid_ORFs'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "bd6300ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_s288c = gene_s288c[gene_s288c['orf_classification'] == 'Verified']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "id": "d019b4f8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5216, 14)"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_s288c.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "6a766625",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid = verified[verified['valid_ORFs'] == '1']\n",
    "verified_unvalid = verified[verified['valid_ORFs'] == '0']\n",
    "\n",
    "verified_valid_count = verified_valid['Chromosome'].value_counts().to_frame()\n",
    "verified_valid_count = verified_valid_count.rename(columns={'Chromosome': 'Number valid genes'})\n",
    "verified_unvalid_count = verified_unvalid['Chromosome'].value_counts().to_frame()\n",
    "verified_unvalid_count = verified_unvalid_count.rename(columns={'Chromosome': 'Number unvalid genes'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "32c1a04f",
   "metadata": {},
   "outputs": [],
   "source": [
    "unval_dic = {}\n",
    "for ind, row in verified_unvalid.iterrows():\n",
    "    if row['Chromosome'] in unval_dic:\n",
    "        unval_dic[row['Chromosome']] = unval_dic[row['Chromosome']] + [row['Name']]\n",
    "    else:\n",
    "        unval_dic[row['Chromosome']] = [row['Name']]    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "63ab8ca2",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum = verified_valid_count.join(verified_unvalid_count)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "52c99402",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmp = []\n",
    "for ind in verified_sum.index:\n",
    "    tmp.append(unval_dic[ind])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "ec34117a",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum['unvalid genes'] = tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "e7ad4015",
   "metadata": {},
   "outputs": [],
   "source": [
    "missed = []\n",
    "moved = []\n",
    "count_miss = []\n",
    "count_move = []\n",
    "for Ch in verified_sum.index:\n",
    "    s288 = verified_s288c[verified_s288c['Chromosome'] == Ch]\n",
    "    yps = verified[verified['Chromosome'] == Ch]\n",
    "    tmp = list(set(s288['Name']) - set(yps['Name']))\n",
    "    miss = []\n",
    "    mov = []\n",
    "    others = verified[verified['Chromosome'] != Ch]\n",
    "    for g in tmp:\n",
    "        search = others[others['Name'] == g]\n",
    "        if search.shape[0] >= 1:\n",
    "            mov.append((g, list(search['Chromosome']), list(search['valid_ORFs'])))\n",
    "        else:\n",
    "            miss.append(g)  \n",
    "    moved.append(mov)\n",
    "    missed.append(miss)\n",
    "    count_miss.append(len(miss))\n",
    "    count_move.append(len(mov))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "0e731ec9",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum['Number of missing genes'] = count_miss\n",
    "verified_sum['Missing genes'] = missed\n",
    "verified_sum['Number of moved genes'] = count_move\n",
    "verified_sum['Moved genes, new chromosome(s), orf validity'] = moved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "9ad232b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_sum.to_excel('Liftoff_v5_verified_genes_summary.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3a1a07a6",
   "metadata": {},
   "source": [
    "# Verified missing genes dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "146d74ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "flat_list = [item for sublist in missed for item in sublist]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "b9fa3baf",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_genes = verified_s288c[verified_s288c['Name'].isin(flat_list)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "2532382c",
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_genes.to_excel('missing_verified_genes_v5.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4c6a1c7b",
   "metadata": {},
   "source": [
    "# Duplicated genes\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "c130b50c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0    6563\n",
       "1      94\n",
       "2      22\n",
       "3      12\n",
       "4       7\n",
       "5       5\n",
       "6       2\n",
       "Name: extra_copy_number, dtype: int64"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Many duplicated genes in this version\n",
    "gene['extra_copy_number'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "bda8b613",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Name</th>\n",
       "      <th>Alias</th>\n",
       "      <th>Note</th>\n",
       "      <th>Ontology_term</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>...</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Contig</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>17994</td>\n",
       "      <td>18290</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.943</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00004927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>767</td>\n",
       "      <td>1021</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>1323</td>\n",
       "      <td>2063</td>\n",
       "      <td>YCL076W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>chrIII-2_tig00004935</td>\n",
       "      <td>6426</td>\n",
       "      <td>8271</td>\n",
       "      <td>YCL073C</td>\n",
       "      <td>GEX1,glutathione%20exchanger</td>\n",
       "      <td>Proton:glutathione antiporter%3B localized to ...</td>\n",
       "      <td>GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Proton:glutathione antiporter</td>\n",
       "      <td>2</td>\n",
       "      <td>GEX1</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.990</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>760</td>\n",
       "      <td>1014</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>1316</td>\n",
       "      <td>2055</td>\n",
       "      <td>YCL076W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>310848</td>\n",
       "      <td>311102</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>tig00004936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2097</th>\n",
       "      <td>chrII_tig00000010</td>\n",
       "      <td>789332</td>\n",
       "      <td>789467</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.890</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>tig00000010</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2106</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>1219</td>\n",
       "      <td>1356</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>tig00000001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4835</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>2653</td>\n",
       "      <td>2790</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4838</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>2759</td>\n",
       "      <td>2986</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.947</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4841</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>3329</td>\n",
       "      <td>3525</td>\n",
       "      <td>YCR102W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>tig00000023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5644</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>676</td>\n",
       "      <td>975</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6054</th>\n",
       "      <td>chrI_tig00000033</td>\n",
       "      <td>238815</td>\n",
       "      <td>239136</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.867</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>tig00000033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6057</th>\n",
       "      <td>chrMANY_tig00000038</td>\n",
       "      <td>737</td>\n",
       "      <td>1036</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6063</th>\n",
       "      <td>chrMANY_tig00000038</td>\n",
       "      <td>2728</td>\n",
       "      <td>2958</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00000038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6917</th>\n",
       "      <td>chrVI-2_tig00004937</td>\n",
       "      <td>12581</td>\n",
       "      <td>12879</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.943</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>tig00004937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6920</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>680</td>\n",
       "      <td>934</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7906</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>522253</td>\n",
       "      <td>522633</td>\n",
       "      <td>YAL064W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Fungal-specific protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005783,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Fungal-specific protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.934</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>tig00000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9422</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>805604</td>\n",
       "      <td>806749</td>\n",
       "      <td>YGR295C</td>\n",
       "      <td>COS6</td>\n",
       "      <td>Endosomal protein involved in turnover of plas...</td>\n",
       "      <td>GO:0000324,GO:0000324,GO:0003674,GO:0005768,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Endosomal protein involved in turnover of plas...</td>\n",
       "      <td>2</td>\n",
       "      <td>COS6</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.961</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9425</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>807356</td>\n",
       "      <td>807492</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.927</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9428</th>\n",
       "      <td>chrVII_tig00004928</td>\n",
       "      <td>807660</td>\n",
       "      <td>807956</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.93</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>tig00004928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9888</th>\n",
       "      <td>chrVI_tig00004938</td>\n",
       "      <td>298356</td>\n",
       "      <td>298655</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.936</td>\n",
       "      <td>0</td>\n",
       "      <td>6</td>\n",
       "      <td>tig00004938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11170</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>642</td>\n",
       "      <td>835</td>\n",
       "      <td>YCR108C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11173</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>2619</td>\n",
       "      <td>2756</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.949</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11182</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>4754</td>\n",
       "      <td>5859</td>\n",
       "      <td>YCR102C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative quinone oxidoreductase%3B overexpress...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative quinone oxidoreductase</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11185</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>7160</td>\n",
       "      <td>11768</td>\n",
       "      <td>YJL222W</td>\n",
       "      <td>VTH2,signal%20sequence-binding%20protein</td>\n",
       "      <td>Putative membrane glycoprotein%3B has strong s...</td>\n",
       "      <td>GO:0000324,GO:0005048,GO:0005048,GO:0006623,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative membrane glycoprotein</td>\n",
       "      <td>2</td>\n",
       "      <td>VTH2</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.972</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11188</th>\n",
       "      <td>chrX-2_tig00001099</td>\n",
       "      <td>12417</td>\n",
       "      <td>14186</td>\n",
       "      <td>YIL172C</td>\n",
       "      <td>IMA3,oligo-1%2C6-glucosidase%20IMA3</td>\n",
       "      <td>Alpha-glucosidase%3B weak%2C but broad substra...</td>\n",
       "      <td>GO:0004574,GO:0004574,GO:0004575,GO:0005737,GO...</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Alpha-glucosidase</td>\n",
       "      <td>2</td>\n",
       "      <td>IMA3</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11221</th>\n",
       "      <td>chrXI-2_tig00004931</td>\n",
       "      <td>28713</td>\n",
       "      <td>28967</td>\n",
       "      <td>YNR077C</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function%2C abundance chang...</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.918</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004931</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11224</th>\n",
       "      <td>chrXII-2_tig00004934</td>\n",
       "      <td>699</td>\n",
       "      <td>999</td>\n",
       "      <td>YDR543C</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.926</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12027</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>6551</td>\n",
       "      <td>6679</td>\n",
       "      <td>YBL108C-A</td>\n",
       "      <td>PAU9,seripauperin%20PAU9</td>\n",
       "      <td>Protein of unknown function%3B member of the s...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>PAU9</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12030</th>\n",
       "      <td>chrXII-3_tig00004933</td>\n",
       "      <td>7121</td>\n",
       "      <td>7351</td>\n",
       "      <td>YGL260W</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B transc...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.948</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00004933</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12060</th>\n",
       "      <td>chrXIII-2_tig00000029</td>\n",
       "      <td>1228</td>\n",
       "      <td>1363</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.948</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13703</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>467347</td>\n",
       "      <td>467484</td>\n",
       "      <td>YHL048C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function%3B identi...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.920</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13709</th>\n",
       "      <td>chrXIII_tig00000018</td>\n",
       "      <td>467923</td>\n",
       "      <td>468106</td>\n",
       "      <td>YOR394C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Protein of unknown function%3B identified by g...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.822</td>\n",
       "      <td>0</td>\n",
       "      <td>13</td>\n",
       "      <td>tig00000018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14775</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>554706</td>\n",
       "      <td>554903</td>\n",
       "      <td>YCR102W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.954</td>\n",
       "      <td>0</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00000020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14790</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>557160</td>\n",
       "      <td>557459</td>\n",
       "      <td>YER188C-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative protein of unknown function</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>tig00000020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16142</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>728468</td>\n",
       "      <td>728695</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.929</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>tig00004929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16145</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>728662</td>\n",
       "      <td>728801</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.921</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>tig00004929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16158</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>726</td>\n",
       "      <td>980</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>tig00004932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17809</th>\n",
       "      <td>chrXVI-2_tig00000040</td>\n",
       "      <td>743</td>\n",
       "      <td>997</td>\n",
       "      <td>YNL337W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.941</td>\n",
       "      <td>1</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19446</th>\n",
       "      <td>chrXVI_tig00000013</td>\n",
       "      <td>719024</td>\n",
       "      <td>719337</td>\n",
       "      <td>YAL069W</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.895</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>tig00000013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19500</th>\n",
       "      <td>chrXV_tig00000007</td>\n",
       "      <td>24888</td>\n",
       "      <td>25351</td>\n",
       "      <td>YIL171W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>tig00000007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20876</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>2582</td>\n",
       "      <td>2719</td>\n",
       "      <td>YJL222W-B</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.956</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20879</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>2688</td>\n",
       "      <td>2915</td>\n",
       "      <td>YJL222W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.939</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20885</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>4721</td>\n",
       "      <td>5826</td>\n",
       "      <td>YCR102C</td>\n",
       "      <td>None</td>\n",
       "      <td>Putative quinone oxidoreductase%3B overexpress...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Putative quinone oxidoreductase</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.979</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20897</th>\n",
       "      <td>chrX_tig00001098</td>\n",
       "      <td>13824</td>\n",
       "      <td>14287</td>\n",
       "      <td>YIL171W-A</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious open reading frame%3B unlikely to enco...</td>\n",
       "      <td>GO:0003674,GO:0005575,GO:0008150,SO:0000704</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>...</td>\n",
       "      <td>Dubious open reading frame</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.952</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>tig00001098</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>48 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      seq_id   start     end       Name  \\\n",
       "9        chrII-2_tig00004927   14242   14700    YGR293C   \n",
       "21       chrII-2_tig00004927   17994   18290  YER188C-A   \n",
       "24      chrIII-2_tig00004935     767    1021    YNR077C   \n",
       "27      chrIII-2_tig00004935    1323    2063    YCL076W   \n",
       "30      chrIII-2_tig00004935    6426    8271    YCL073C   \n",
       "33        chrIII_tig00004936     760    1014    YNR077C   \n",
       "36        chrIII_tig00004936    1316    2055    YCL076W   \n",
       "651       chrIII_tig00004936  310848  311102    YNL337W   \n",
       "2097       chrII_tig00000010  789332  789467  YHL048C-A   \n",
       "2106       chrIV_tig00000001    1219    1356  YHL048C-A   \n",
       "4835       chrIX_tig00000023    2653    2790  YJL222W-B   \n",
       "4838       chrIX_tig00000023    2759    2986  YJL222W-A   \n",
       "4841       chrIX_tig00000023    3329    3525  YCR102W-A   \n",
       "5644        chrI_tig00000033     676     975    YDR543C   \n",
       "6054        chrI_tig00000033  238815  239136    YAL069W   \n",
       "6057     chrMANY_tig00000038     737    1036    YDR543C   \n",
       "6063     chrMANY_tig00000038    2728    2958  YJL222W-A   \n",
       "6917     chrVI-2_tig00004937   12581   12879    YDR543C   \n",
       "6920     chrVIII_tig00000017     680     934    YNR077C   \n",
       "7906     chrVIII_tig00000017  522253  522633  YAL064W-B   \n",
       "9422      chrVII_tig00004928  805604  806749    YGR295C   \n",
       "9425      chrVII_tig00004928  807356  807492  YHL048C-A   \n",
       "9428      chrVII_tig00004928  807660  807956  YER188C-A   \n",
       "9888       chrVI_tig00004938  298356  298655    YDR543C   \n",
       "11170     chrX-2_tig00001099     642     835    YCR108C   \n",
       "11173     chrX-2_tig00001099    2619    2756  YJL222W-B   \n",
       "11182     chrX-2_tig00001099    4754    5859    YCR102C   \n",
       "11185     chrX-2_tig00001099    7160   11768    YJL222W   \n",
       "11188     chrX-2_tig00001099   12417   14186    YIL172C   \n",
       "11221    chrXI-2_tig00004931   28713   28967    YNR077C   \n",
       "11224   chrXII-2_tig00004934     699     999    YDR543C   \n",
       "12027   chrXII-3_tig00004933    6551    6679  YBL108C-A   \n",
       "12030   chrXII-3_tig00004933    7121    7351    YGL260W   \n",
       "12060  chrXIII-2_tig00000029    1228    1363  YHL048C-A   \n",
       "13703    chrXIII_tig00000018  467347  467484  YHL048C-A   \n",
       "13709    chrXIII_tig00000018  467923  468106  YOR394C-A   \n",
       "14775     chrXII_tig00000020  554706  554903  YCR102W-A   \n",
       "14790     chrXII_tig00000020  557160  557459  YER188C-A   \n",
       "16142     chrXIV_tig00004929  728468  728695  YJL222W-A   \n",
       "16145     chrXIV_tig00004929  728662  728801  YJL222W-B   \n",
       "16158      chrXI_tig00004932     726     980    YNL337W   \n",
       "17809   chrXVI-2_tig00000040     743     997    YNL337W   \n",
       "19446     chrXVI_tig00000013  719024  719337    YAL069W   \n",
       "19500      chrXV_tig00000007   24888   25351  YIL171W-A   \n",
       "20876       chrX_tig00001098    2582    2719  YJL222W-B   \n",
       "20879       chrX_tig00001098    2688    2915  YJL222W-A   \n",
       "20885       chrX_tig00001098    4721    5826    YCR102C   \n",
       "20897       chrX_tig00001098   13824   14287  YIL171W-A   \n",
       "\n",
       "                                          Alias  \\\n",
       "9                                          None   \n",
       "21                                         None   \n",
       "24                                         None   \n",
       "27                                         None   \n",
       "30                 GEX1,glutathione%20exchanger   \n",
       "33                                         None   \n",
       "36                                         None   \n",
       "651                                        None   \n",
       "2097                                       None   \n",
       "2106                                       None   \n",
       "4835                                       None   \n",
       "4838                                       None   \n",
       "4841                                       None   \n",
       "5644                                       None   \n",
       "6054                                       None   \n",
       "6057                                       None   \n",
       "6063                                       None   \n",
       "6917                                       None   \n",
       "6920                                       None   \n",
       "7906                                       None   \n",
       "9422                                       COS6   \n",
       "9425                                       None   \n",
       "9428                                       None   \n",
       "9888                                       None   \n",
       "11170                                      None   \n",
       "11173                                      None   \n",
       "11182                                      None   \n",
       "11185  VTH2,signal%20sequence-binding%20protein   \n",
       "11188       IMA3,oligo-1%2C6-glucosidase%20IMA3   \n",
       "11221                                      None   \n",
       "11224                                      None   \n",
       "12027                  PAU9,seripauperin%20PAU9   \n",
       "12030                                      None   \n",
       "12060                                      None   \n",
       "13703                                      None   \n",
       "13709                                      None   \n",
       "14775                                      None   \n",
       "14790                                      None   \n",
       "16142                                      None   \n",
       "16145                                      None   \n",
       "16158                                      None   \n",
       "17809                                      None   \n",
       "19446                                      None   \n",
       "19500                                      None   \n",
       "20876                                      None   \n",
       "20879                                      None   \n",
       "20885                                      None   \n",
       "20897                                      None   \n",
       "\n",
       "                                                    Note  \\\n",
       "9      Dubious open reading frame%3B unlikely to enco...   \n",
       "21                  Putative protein of unknown function   \n",
       "24     Protein of unknown function%2C abundance chang...   \n",
       "27     Dubious open reading frame%3B unlikely to enco...   \n",
       "30     Proton:glutathione antiporter%3B localized to ...   \n",
       "33     Protein of unknown function%2C abundance chang...   \n",
       "36     Dubious open reading frame%3B unlikely to enco...   \n",
       "651    Dubious open reading frame%3B unlikely to enco...   \n",
       "2097   Putative protein of unknown function%3B identi...   \n",
       "2106   Putative protein of unknown function%3B identi...   \n",
       "4835   Dubious open reading frame%3B unlikely to enco...   \n",
       "4838   Dubious open reading frame%3B unlikely to enco...   \n",
       "4841   Dubious open reading frame%3B unlikely to enco...   \n",
       "5644   Dubious open reading frame%3B unlikely to enco...   \n",
       "6054   Dubious open reading frame%3B unlikely to enco...   \n",
       "6057   Dubious open reading frame%3B unlikely to enco...   \n",
       "6063   Dubious open reading frame%3B unlikely to enco...   \n",
       "6917   Dubious open reading frame%3B unlikely to enco...   \n",
       "6920   Protein of unknown function%2C abundance chang...   \n",
       "7906         Fungal-specific protein of unknown function   \n",
       "9422   Endosomal protein involved in turnover of plas...   \n",
       "9425   Putative protein of unknown function%3B identi...   \n",
       "9428                Putative protein of unknown function   \n",
       "9888   Dubious open reading frame%3B unlikely to enco...   \n",
       "11170  Putative protein of unknown function%3B identi...   \n",
       "11173  Dubious open reading frame%3B unlikely to enco...   \n",
       "11182  Putative quinone oxidoreductase%3B overexpress...   \n",
       "11185  Putative membrane glycoprotein%3B has strong s...   \n",
       "11188  Alpha-glucosidase%3B weak%2C but broad substra...   \n",
       "11221  Protein of unknown function%2C abundance chang...   \n",
       "11224  Dubious open reading frame%3B unlikely to enco...   \n",
       "12027  Protein of unknown function%3B member of the s...   \n",
       "12030  Putative protein of unknown function%3B transc...   \n",
       "12060  Putative protein of unknown function%3B identi...   \n",
       "13703  Putative protein of unknown function%3B identi...   \n",
       "13709  Protein of unknown function%3B identified by g...   \n",
       "14775  Dubious open reading frame%3B unlikely to enco...   \n",
       "14790               Putative protein of unknown function   \n",
       "16142  Dubious open reading frame%3B unlikely to enco...   \n",
       "16145  Dubious open reading frame%3B unlikely to enco...   \n",
       "16158  Dubious open reading frame%3B unlikely to enco...   \n",
       "17809  Dubious open reading frame%3B unlikely to enco...   \n",
       "19446  Dubious open reading frame%3B unlikely to enco...   \n",
       "19500  Dubious open reading frame%3B unlikely to enco...   \n",
       "20876  Dubious open reading frame%3B unlikely to enco...   \n",
       "20879  Dubious open reading frame%3B unlikely to enco...   \n",
       "20885  Putative quinone oxidoreductase%3B overexpress...   \n",
       "20897  Dubious open reading frame%3B unlikely to enco...   \n",
       "\n",
       "                                           Ontology_term score strand phase  \\\n",
       "9            GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "21           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "24           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "27           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "30     GO:0005774,GO:0005886,GO:0015299,GO:0034775,GO...     .      -     .   \n",
       "33           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "36           GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "651          GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "2097         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "2106         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "4835         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "4838         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "4841         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "5644         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6054         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "6057         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6063         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "6917         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "6920         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "7906         GO:0003674,GO:0005783,GO:0008150,SO:0000704     .      -     .   \n",
       "9422   GO:0000324,GO:0000324,GO:0003674,GO:0005768,GO...     .      -     .   \n",
       "9425         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "9428         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "9888         GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "11170        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "11173        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "11182  GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...     .      +     .   \n",
       "11185  GO:0000324,GO:0005048,GO:0005048,GO:0006623,GO...     .      +     .   \n",
       "11188  GO:0004574,GO:0004574,GO:0004575,GO:0005737,GO...     .      -     .   \n",
       "11221        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "11224        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "12027        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "12030        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "12060        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "13703        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "13709        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "14775        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "14790        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16142        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16145        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "16158        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "17809        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "19446        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      -     .   \n",
       "19500        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20876        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20879        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "20885  GO:0003674,GO:0005575,GO:0046688,GO:0071468,SO...     .      +     .   \n",
       "20897        GO:0003674,GO:0005575,GO:0008150,SO:0000704     .      +     .   \n",
       "\n",
       "       ...                                            display  \\\n",
       "9      ...                         Dubious open reading frame   \n",
       "21     ...               Putative protein of unknown function   \n",
       "24     ...  Protein of unknown function%2C abundance chang...   \n",
       "27     ...                         Dubious open reading frame   \n",
       "30     ...                      Proton:glutathione antiporter   \n",
       "33     ...  Protein of unknown function%2C abundance chang...   \n",
       "36     ...                         Dubious open reading frame   \n",
       "651    ...                         Dubious open reading frame   \n",
       "2097   ...               Putative protein of unknown function   \n",
       "2106   ...               Putative protein of unknown function   \n",
       "4835   ...                         Dubious open reading frame   \n",
       "4838   ...                         Dubious open reading frame   \n",
       "4841   ...                         Dubious open reading frame   \n",
       "5644   ...                         Dubious open reading frame   \n",
       "6054   ...                         Dubious open reading frame   \n",
       "6057   ...                         Dubious open reading frame   \n",
       "6063   ...                         Dubious open reading frame   \n",
       "6917   ...                         Dubious open reading frame   \n",
       "6920   ...  Protein of unknown function%2C abundance chang...   \n",
       "7906   ...        Fungal-specific protein of unknown function   \n",
       "9422   ...  Endosomal protein involved in turnover of plas...   \n",
       "9425   ...               Putative protein of unknown function   \n",
       "9428   ...               Putative protein of unknown function   \n",
       "9888   ...                         Dubious open reading frame   \n",
       "11170  ...               Putative protein of unknown function   \n",
       "11173  ...                         Dubious open reading frame   \n",
       "11182  ...                    Putative quinone oxidoreductase   \n",
       "11185  ...                     Putative membrane glycoprotein   \n",
       "11188  ...                                  Alpha-glucosidase   \n",
       "11221  ...  Protein of unknown function%2C abundance chang...   \n",
       "11224  ...                         Dubious open reading frame   \n",
       "12027  ...                        Protein of unknown function   \n",
       "12030  ...               Putative protein of unknown function   \n",
       "12060  ...               Putative protein of unknown function   \n",
       "13703  ...               Putative protein of unknown function   \n",
       "13709  ...                        Protein of unknown function   \n",
       "14775  ...                         Dubious open reading frame   \n",
       "14790  ...               Putative protein of unknown function   \n",
       "16142  ...                         Dubious open reading frame   \n",
       "16145  ...                         Dubious open reading frame   \n",
       "16158  ...                         Dubious open reading frame   \n",
       "17809  ...                         Dubious open reading frame   \n",
       "19446  ...                         Dubious open reading frame   \n",
       "19500  ...                         Dubious open reading frame   \n",
       "20876  ...                         Dubious open reading frame   \n",
       "20879  ...                         Dubious open reading frame   \n",
       "20885  ...                    Putative quinone oxidoreductase   \n",
       "20897  ...                         Dubious open reading frame   \n",
       "\n",
       "      extra_copy_number  gene low_identity orf_classification partial_mapping  \\\n",
       "9                     2  None         None            Dubious            None   \n",
       "21                    3  None         None    Uncharacterized            None   \n",
       "24                    3  None         None    Uncharacterized            None   \n",
       "27                    2  None         None            Dubious            None   \n",
       "30                    2  GEX1         None           Verified            None   \n",
       "33                    2  None         None    Uncharacterized            None   \n",
       "36                    3  None         None            Dubious            None   \n",
       "651                   4  None         None            Dubious            None   \n",
       "2097                  6  None         None    Uncharacterized            None   \n",
       "2106                  4  None         None    Uncharacterized            None   \n",
       "4835                  4  None         None            Dubious            None   \n",
       "4838                  2  None         None            Dubious            None   \n",
       "4841                  3  None         None            Dubious            None   \n",
       "5644                  4  None         None            Dubious            None   \n",
       "6054                  2  None         None            Dubious            None   \n",
       "6057                  2  None         None            Dubious            None   \n",
       "6063                  5  None         None            Dubious            None   \n",
       "6917                  3  None         None            Dubious            None   \n",
       "6920                  5  None         None    Uncharacterized            None   \n",
       "7906                  2  None         None    Uncharacterized            None   \n",
       "9422                  2  COS6         None           Verified            None   \n",
       "9425                  5  None         None    Uncharacterized            None   \n",
       "9428                  4  None         None    Uncharacterized            None   \n",
       "9888                  5  None         None            Dubious            None   \n",
       "11170                 2  None         None    Uncharacterized            None   \n",
       "11173                 5  None         None            Dubious            None   \n",
       "11182                 2  None         None    Uncharacterized            None   \n",
       "11185                 2  VTH2         None           Verified            None   \n",
       "11188                 2  IMA3         None           Verified            None   \n",
       "11221                 4  None         None    Uncharacterized            None   \n",
       "11224                 6  None         None            Dubious            None   \n",
       "12027                 2  PAU9         None           Verified            None   \n",
       "12030                 2  None         None    Uncharacterized            None   \n",
       "12060                 2  None         None    Uncharacterized            None   \n",
       "13703                 3  None         None    Uncharacterized            None   \n",
       "13709                 2  None         None    Uncharacterized            None   \n",
       "14775                 2  None         None            Dubious            None   \n",
       "14790                 2  None         None    Uncharacterized            None   \n",
       "16142                 3  None         None            Dubious            None   \n",
       "16145                 3  None         None            Dubious            None   \n",
       "16158                 2  None         None            Dubious            None   \n",
       "17809                 3  None         None            Dubious            None   \n",
       "19446                 3  None         None            Dubious            None   \n",
       "19500                 2  None         None            Dubious            None   \n",
       "20876                 2  None         None            Dubious            None   \n",
       "20879                 4  None         None            Dubious            None   \n",
       "20885                 3  None         None    Uncharacterized            None   \n",
       "20897                 3  None         None            Dubious            None   \n",
       "\n",
       "      sequence_ID valid_ORFs Chromosome       Contig  \n",
       "9           0.989          0          2  tig00004927  \n",
       "21          0.943          1          2  tig00004927  \n",
       "24          0.918          1          3  tig00004935  \n",
       "27          0.981          0          3  tig00004935  \n",
       "30          0.990          0          3  tig00004935  \n",
       "33          0.918          1          3  tig00004936  \n",
       "36          0.979          0          3  tig00004936  \n",
       "651         0.949          1          3  tig00004936  \n",
       "2097        0.890          0          2  tig00000010  \n",
       "2106        0.920          0          4  tig00000001  \n",
       "4835        0.942          1          9  tig00000023  \n",
       "4838        0.947          1          9  tig00000023  \n",
       "4841        0.949          0          9  tig00000023  \n",
       "5644        0.946          0          1  tig00000033  \n",
       "6054        0.867          0          1  tig00000033  \n",
       "6057        0.936          0          7  tig00000038  \n",
       "6063        0.917          0          7  tig00000038  \n",
       "6917        0.943          0          6  tig00004937  \n",
       "6920        0.917          0          8  tig00000017  \n",
       "7906        0.934          1          8  tig00000017  \n",
       "9422        0.961          1          7  tig00004928  \n",
       "9425        0.927          0          7  tig00004928  \n",
       "9428         0.93          1          7  tig00004928  \n",
       "9888        0.936          0          6  tig00004938  \n",
       "11170       0.917          0         10  tig00001099  \n",
       "11173       0.949          1         10  tig00001099  \n",
       "11182       0.981          0         10  tig00001099  \n",
       "11185       0.972          0         10  tig00001099  \n",
       "11188       0.989          1         10  tig00001099  \n",
       "11221       0.918          1         11  tig00004931  \n",
       "11224       0.926          0         12  tig00004934  \n",
       "12027         1.0          1         12  tig00004933  \n",
       "12030       0.948          1         12  tig00004933  \n",
       "12060       0.948          0         13  tig00000029  \n",
       "13703       0.920          0         13  tig00000018  \n",
       "13709       0.822          0         13  tig00000018  \n",
       "14775       0.954          0         12  tig00000020  \n",
       "14790       0.956          1         12  tig00000020  \n",
       "16142       0.929          1         14  tig00004929  \n",
       "16145       0.921          0         14  tig00004929  \n",
       "16158       0.956          1         11  tig00004932  \n",
       "17809       0.941          1         16  tig00000040  \n",
       "19446       0.895          0         16  tig00000013  \n",
       "19500       0.956          0         15  tig00000007  \n",
       "20876       0.956          1         10  tig00001098  \n",
       "20879       0.939          0         10  tig00001098  \n",
       "20885       0.979          0         10  tig00001098  \n",
       "20897       0.952          0         10  tig00001098  \n",
       "\n",
       "[48 rows x 23 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene[gene['extra_copy_number'] > '1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "id": "73d236f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl = gene[gene.duplicated(subset=['Name'], keep=False) == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "id": "f1876fa9",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl = dupl.sort_values(by = 'Name')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "id": "1f204f21",
   "metadata": {},
   "outputs": [],
   "source": [
    "dupl.to_excel('duplicated_verified_genes.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d155e624",
   "metadata": {},
   "source": [
    "Duplicated genes:\n",
    "\n",
    "ATP8/Q0080: Subunit 8 of the F0 sector of mitochondrial F1F0 ATP synthase; encoded on the mitochondrial genome; ATP8 and ATP6 mRNAs are not translated in the absence of the F1 sector of ATPase\n",
    "\n",
    "OLI1/Q0130: F0-ATP synthase subunit c (ATPase-associated proteolipid); encoded on the mitochondrial genome; mutation confers oligomycin resistance; expression is specifically dependent on the nuclear genes AEP1 and AEP2 1 2 3 4\n",
    "Name Description\n",
    "OLIgomycin resistance 1\n",
    "\n",
    "Both on mitochondria so not that relevant to my project.\n",
    "\n",
    "PAU9/YBL108C-A: Protein of unknown function; member of the seripauperin multigene family encoded mainly in subtelomeric regions; SWAT-GFP and mCherry fusion proteins localize to the endoplasmic reticulum and vacuole respectively\n",
    "\n",
    "YHR213W-A: Putative protein of unknown function; identified by gene-trapping, microarray-based expression analysis, and genome-wide homology searching "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1e0c10ee",
   "metadata": {},
   "source": [
    "# Analyse unvalid genes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82bdf79b",
   "metadata": {},
   "source": [
    "6 partial mapping genes:\n",
    "    YDL227C near telomere\n",
    "    YMR106C middle of chromosome arm\n",
    "    YNL322C near telomere and missing genes, suggesting GAP but could also be a fragment deletion.\n",
    "    YOR244W middle of chromosome arm\n",
    "    YOR245C middle of chromosome arm"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aeb9458b",
   "metadata": {},
   "source": [
    "### merge mRNA and gene data to get information on the type of issue in ORF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "3ad463f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'ID', 'Name', 'Parent', 'conditions', 'dbxref',\n",
       "       'extra_copy_number', 'inframe_stop_codon', 'matches_ref_protein',\n",
       "       'missing_start_codon', 'missing_stop_codon', 'transcript_id',\n",
       "       'valid_ORF'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 189,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mRNA_df = annotation.filter_feature_of_type(['mRNA'])\n",
    "mRNA_attr = mRNA_df.attributes_to_columns()\n",
    "mRNA_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "id": "2c97f343",
   "metadata": {},
   "outputs": [],
   "source": [
    "mRNA = mRNA_attr[['seq_id', 'start', 'end', 'ID', 'Name', 'Parent',\n",
    "                  'conditions', 'extra_copy_number', 'inframe_stop_codon', \n",
    "                  'matches_ref_protein', 'missing_start_codon', 'missing_stop_codon', \n",
    "                  'transcript_id', 'valid_ORF']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "id": "b9558d48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'source', 'type', 'start', 'end', 'score', 'strand', 'phase',\n",
       "       'attributes', 'Alias', 'ID', 'Name', 'Note', 'Ontology_term',\n",
       "       'copy_num_ID', 'coverage', 'curie', 'dbxref', 'display',\n",
       "       'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
       "       'partial_mapping', 'sequence_ID', 'valid_ORFs'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_attr.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "id": "290bb927",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene2 = gene_attr[['seq_id', 'start', 'end',\n",
    "       'Alias', 'ID', 'Name', 'copy_num_ID', 'coverage', 'display',\n",
    "       'extra_copy_number', 'gene', 'low_identity', 'orf_classification',\n",
    "       'partial_mapping', 'sequence_ID', 'valid_ORFs']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "id": "c6802b61",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Alias</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>gene</th>\n",
       "      <th>low_identity</th>\n",
       "      <th>orf_classification</th>\n",
       "      <th>partial_mapping</th>\n",
       "      <th>sequence_ID</th>\n",
       "      <th>valid_ORFs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>MAL31,MAL3T,MALT,maltose%20permease</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>YBR298C</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltose%20permease</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL31</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>None</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>YBR298C-A</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>0.990</td>\n",
       "      <td>Putative%20protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Uncharacterized</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>YBR299W</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>0.986</td>\n",
       "      <td>Maltase%20%28alpha-D-glucosidase%29</td>\n",
       "      <td>1</td>\n",
       "      <td>MAL32</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.981</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>None</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>YGR293C</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>0.993</td>\n",
       "      <td>Dubious%20open%20reading%20frame</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>Dubious</td>\n",
       "      <td>None</td>\n",
       "      <td>0.989</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>PAU12,seripauperin%20PAU12</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>YGR294W</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>1</td>\n",
       "      <td>PAU12</td>\n",
       "      <td>None</td>\n",
       "      <td>Verified</td>\n",
       "      <td>None</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seq_id  start    end  \\\n",
       "0   chrII-2_tig00004927   8297  10119   \n",
       "3   chrII-2_tig00004927  10675  10894   \n",
       "6   chrII-2_tig00004927  10985  12717   \n",
       "9   chrII-2_tig00004927  14242  14700   \n",
       "12  chrII-2_tig00004927  14660  15022   \n",
       "\n",
       "                                         Alias           ID       Name  \\\n",
       "0          MAL31,MAL3T,MALT,maltose%20permease    YBR298C_1    YBR298C   \n",
       "3                                         None  YBR298C-A_1  YBR298C-A   \n",
       "6   MAL32,MAL3S,MALS,alpha-glucosidase%20MAL32    YBR299W_1    YBR299W   \n",
       "9                                         None    YGR293C_2    YGR293C   \n",
       "12                  PAU12,seripauperin%20PAU12    YGR294W_1    YGR294W   \n",
       "\n",
       "    copy_num_ID coverage                                       display  \\\n",
       "0     YBR298C_1    0.986                            Maltose%20permease   \n",
       "3   YBR298C-A_1    0.990  Putative%20protein%20of%20unknown%20function   \n",
       "6     YBR299W_1    0.986           Maltase%20%28alpha-D-glucosidase%29   \n",
       "9     YGR293C_2    0.993              Dubious%20open%20reading%20frame   \n",
       "12    YGR294W_1      1.0             Protein%20of%20unknown%20function   \n",
       "\n",
       "   extra_copy_number   gene low_identity orf_classification partial_mapping  \\\n",
       "0                  1  MAL31         None           Verified            None   \n",
       "3                  1   None         None    Uncharacterized            None   \n",
       "6                  1  MAL32         None           Verified            None   \n",
       "9                  2   None         None            Dubious            None   \n",
       "12                 1  PAU12         None           Verified            None   \n",
       "\n",
       "   sequence_ID valid_ORFs  \n",
       "0        0.946          0  \n",
       "3        0.981          0  \n",
       "6        0.981          0  \n",
       "9        0.989          0  \n",
       "12       0.983          1  "
      ]
     },
     "execution_count": 193,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "id": "c27d20d8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>ID</th>\n",
       "      <th>Name</th>\n",
       "      <th>Parent</th>\n",
       "      <th>conditions</th>\n",
       "      <th>extra_copy_number</th>\n",
       "      <th>inframe_stop_codon</th>\n",
       "      <th>matches_ref_protein</th>\n",
       "      <th>missing_start_codon</th>\n",
       "      <th>missing_stop_codon</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>valid_ORF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>8297</td>\n",
       "      <td>10119</td>\n",
       "      <td>YBR298C_id001_1</td>\n",
       "      <td>YBR298C_id001</td>\n",
       "      <td>YBR298C_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288108_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10675</td>\n",
       "      <td>10894</td>\n",
       "      <td>YBR298C-A_mRNA_1</td>\n",
       "      <td>YBR298C-A_mRNA</td>\n",
       "      <td>YBR298C-A_1</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>RefSeq:NM_001184588.1_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>10985</td>\n",
       "      <td>12717</td>\n",
       "      <td>YBR299W_id001_1</td>\n",
       "      <td>YBR299W_id001</td>\n",
       "      <td>YBR299W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>SGD:S000288109_1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14242</td>\n",
       "      <td>14700</td>\n",
       "      <td>YGR293C_mRNA_2</td>\n",
       "      <td>YGR293C_mRNA</td>\n",
       "      <td>YGR293C_2</td>\n",
       "      <td>None</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chrII-2_tig00004927</td>\n",
       "      <td>14660</td>\n",
       "      <td>15022</td>\n",
       "      <td>YGR294W_id001_1</td>\n",
       "      <td>YGR294W_id001</td>\n",
       "      <td>YGR294W_1</td>\n",
       "      <td>GAL</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289696_1</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 seq_id  start    end                ID            Name  \\\n",
       "1   chrII-2_tig00004927   8297  10119   YBR298C_id001_1   YBR298C_id001   \n",
       "4   chrII-2_tig00004927  10675  10894  YBR298C-A_mRNA_1  YBR298C-A_mRNA   \n",
       "7   chrII-2_tig00004927  10985  12717   YBR299W_id001_1   YBR299W_id001   \n",
       "10  chrII-2_tig00004927  14242  14700    YGR293C_mRNA_2    YGR293C_mRNA   \n",
       "13  chrII-2_tig00004927  14660  15022   YGR294W_id001_1   YGR294W_id001   \n",
       "\n",
       "         Parent conditions extra_copy_number inframe_stop_codon  \\\n",
       "1     YBR298C_1        GAL                 1               None   \n",
       "4   YBR298C-A_1       None                 1               None   \n",
       "7     YBR299W_1        GAL                 1               None   \n",
       "10    YGR293C_2       None                 2               True   \n",
       "13    YGR294W_1        GAL                 1               None   \n",
       "\n",
       "   matches_ref_protein missing_start_codon missing_stop_codon  \\\n",
       "1                False                None               True   \n",
       "4                False                None               True   \n",
       "7                False                None               True   \n",
       "10               False                None               None   \n",
       "13                True                None               None   \n",
       "\n",
       "              transcript_id valid_ORF  \n",
       "1          SGD:S000288108_1     False  \n",
       "4   RefSeq:NM_001184588.1_1     False  \n",
       "7          SGD:S000288109_1     False  \n",
       "10                     None     False  \n",
       "13         SGD:S000289696_1      True  "
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mRNA.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "id": "d34b04ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA = pd.merge(gene2, mRNA,  how='right', left_on='ID', right_on = 'Parent')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "id": "66eb1b98",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     6167\n",
       "False     524\n",
       "Name: valid_ORF, dtype: int64"
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA['valid_ORF'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "22a15d53",
   "metadata": {},
   "source": [
    "### Focus on Verified unvalid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "id": "8362fb90",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified = gene_mRNA[gene_mRNA['orf_classification']== 'Verified']\n",
    "gene_mRNA_verified_unvalid = gene_mRNA_verified[gene_mRNA_verified['valid_ORF']== 'False']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "5e79ec84",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\4212071888.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  gene_mRNA_verified_unvalid.drop_duplicates(subset=['Name_x'], inplace=True)\n"
     ]
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid.drop_duplicates(subset=['Name_x'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "b9a7146b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(144, 30)"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 210,
   "id": "998049a7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "144"
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(gene_mRNA_verified_unvalid['Name_x']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "c26a4ceb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    101\n",
       "Name: missing_stop_codon, dtype: int64"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_stop_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "id": "c25bfaff",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    17\n",
       "Name: missing_start_codon, dtype: int64"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_start_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "id": "eaea2074",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    26\n",
       "Name: inframe_stop_codon, dtype: int64"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['inframe_stop_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "id": "f547502b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id_x</th>\n",
       "      <th>start_x</th>\n",
       "      <th>end_x</th>\n",
       "      <th>Alias</th>\n",
       "      <th>ID_x</th>\n",
       "      <th>Name_x</th>\n",
       "      <th>copy_num_ID</th>\n",
       "      <th>coverage</th>\n",
       "      <th>display</th>\n",
       "      <th>extra_copy_number_x</th>\n",
       "      <th>...</th>\n",
       "      <th>Name_y</th>\n",
       "      <th>Parent</th>\n",
       "      <th>conditions</th>\n",
       "      <th>extra_copy_number_y</th>\n",
       "      <th>inframe_stop_codon</th>\n",
       "      <th>matches_ref_protein</th>\n",
       "      <th>missing_start_codon</th>\n",
       "      <th>missing_stop_codon</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>valid_ORF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>chrIII_tig00004936</td>\n",
       "      <td>241438</td>\n",
       "      <td>245433</td>\n",
       "      <td>SSK22,mitogen-activated%20protein%20kinase%20k...</td>\n",
       "      <td>YCR073C</td>\n",
       "      <td>YCR073C</td>\n",
       "      <td>YCR073C_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MAP%20kinase%20kinase%20kinase%20of%20HOG1%20m...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YCR073C_mRNA</td>\n",
       "      <td>YCR073C</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001178783.1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>46438</td>\n",
       "      <td>46547</td>\n",
       "      <td>HO</td>\n",
       "      <td>YDL227C</td>\n",
       "      <td>YDL227C</td>\n",
       "      <td>YDL227C_0</td>\n",
       "      <td>0.062</td>\n",
       "      <td>Site-specific%20endonuclease</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YDL227C_id001</td>\n",
       "      <td>YDL227C</td>\n",
       "      <td>YPD</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000293396</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>874</th>\n",
       "      <td>chrIV_tig00000001</td>\n",
       "      <td>381046</td>\n",
       "      <td>383253</td>\n",
       "      <td>PRM7,YDL038C,pheromone-regulated%20protein%20PRM7</td>\n",
       "      <td>YDL039C</td>\n",
       "      <td>YDL039C</td>\n",
       "      <td>YDL039C_0</td>\n",
       "      <td>0.998</td>\n",
       "      <td>Pheromone-regulated%20protein</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YDL039C_mRNA</td>\n",
       "      <td>YDL039C</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001180098.1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1610</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>235689</td>\n",
       "      <td>236111</td>\n",
       "      <td>None</td>\n",
       "      <td>YIL060W</td>\n",
       "      <td>YIL060W</td>\n",
       "      <td>YIL060W_0</td>\n",
       "      <td>0.908</td>\n",
       "      <td>Mitochondrial%20protein%20of%20unknown%20function</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YIL060W_mRNA</td>\n",
       "      <td>YIL060W</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001179410.3</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1694</th>\n",
       "      <td>chrIX_tig00000023</td>\n",
       "      <td>373545</td>\n",
       "      <td>374108</td>\n",
       "      <td>MET28</td>\n",
       "      <td>YIR017C</td>\n",
       "      <td>YIR017C</td>\n",
       "      <td>YIR017C_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>bZIP%20transcriptional%20activator%20in%20the%...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YIR017C_id002</td>\n",
       "      <td>YIR017C</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000290068</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1920</th>\n",
       "      <td>chrVII-2_tig00000038</td>\n",
       "      <td>38804</td>\n",
       "      <td>40006</td>\n",
       "      <td>TAD1,tRNA-specific%20adenosine%20deaminase</td>\n",
       "      <td>YGL243W</td>\n",
       "      <td>YGL243W</td>\n",
       "      <td>YGL243W_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>tRNA-specific%20adenosine%20deaminase</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YGL243W_id002</td>\n",
       "      <td>YGL243W</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289433</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2292</th>\n",
       "      <td>chrVIII_tig00000017</td>\n",
       "      <td>445786</td>\n",
       "      <td>446360</td>\n",
       "      <td>CTR2,low-affinity%20Cu%20transporter</td>\n",
       "      <td>YHR175W</td>\n",
       "      <td>YHR175W</td>\n",
       "      <td>YHR175W_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Low-affinity%20copper%20transporter%20of%20the...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YHR175W_id001</td>\n",
       "      <td>YHR175W</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289883</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2795</th>\n",
       "      <td>chrVI_tig00004938</td>\n",
       "      <td>12513</td>\n",
       "      <td>13592</td>\n",
       "      <td>AAD3,putative%20aryl-alcohol%20dehydrogenase</td>\n",
       "      <td>YCR107W</td>\n",
       "      <td>YCR107W</td>\n",
       "      <td>YCR107W_0</td>\n",
       "      <td>0.982</td>\n",
       "      <td>Putative%20aryl-alcohol%20dehydrogenase</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YCR107W_mRNA</td>\n",
       "      <td>YCR107W</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001178814.1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2887</th>\n",
       "      <td>chrVI_tig00004938</td>\n",
       "      <td>229268</td>\n",
       "      <td>229777</td>\n",
       "      <td>ULI1</td>\n",
       "      <td>YFR026C</td>\n",
       "      <td>YFR026C</td>\n",
       "      <td>YFR026C_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YFR026C_id001</td>\n",
       "      <td>YFR026C</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000289199</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2937</th>\n",
       "      <td>chrV_tig00000015</td>\n",
       "      <td>680</td>\n",
       "      <td>5190</td>\n",
       "      <td>YRF1-4,Y'%20element%20ATP-dependent%20helicase...</td>\n",
       "      <td>YLR466W</td>\n",
       "      <td>YLR466W</td>\n",
       "      <td>YLR466W_0</td>\n",
       "      <td>0.998</td>\n",
       "      <td>Helicase%20encoded%20by%20the%20Y'%20element%2...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YLR466W_mRNA</td>\n",
       "      <td>YLR466W</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001182354.3</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3812</th>\n",
       "      <td>chrXIII-2_tig00000029</td>\n",
       "      <td>445067</td>\n",
       "      <td>446593</td>\n",
       "      <td>YPK2,YKR2,putative%20protein%20kinase%20YPK2</td>\n",
       "      <td>YMR104C</td>\n",
       "      <td>YMR104C</td>\n",
       "      <td>YMR104C_0</td>\n",
       "      <td>0.750</td>\n",
       "      <td>Protein%20kinase%20similar%20to%20S/T%20protei...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YMR104C_mRNA</td>\n",
       "      <td>YMR104C</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001182604.1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4065</th>\n",
       "      <td>chrXII_tig00000020</td>\n",
       "      <td>5009</td>\n",
       "      <td>5314</td>\n",
       "      <td>None</td>\n",
       "      <td>YLR162W</td>\n",
       "      <td>YLR162W</td>\n",
       "      <td>YLR162W_0</td>\n",
       "      <td>0.778</td>\n",
       "      <td>Protein%20of%20unknown%20function</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YLR162W_mRNA</td>\n",
       "      <td>YLR162W</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001182049.1</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4787</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>660792</td>\n",
       "      <td>661424</td>\n",
       "      <td>MSO1</td>\n",
       "      <td>YNR049C</td>\n",
       "      <td>YNR049C</td>\n",
       "      <td>YNR049C_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Lipid-interacting%20protein%20in%20SNARE%20com...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YNR049C_id003</td>\n",
       "      <td>YNR049C</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000291800</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4794</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>678731</td>\n",
       "      <td>680415</td>\n",
       "      <td>BIO5</td>\n",
       "      <td>YNR056C</td>\n",
       "      <td>YNR056C</td>\n",
       "      <td>YNR056C_0</td>\n",
       "      <td>0.999</td>\n",
       "      <td>Putative%20transmembrane%20protein%20involved%...</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YNR056C_id001</td>\n",
       "      <td>YNR056C</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000291807</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4800</th>\n",
       "      <td>chrXIV_tig00004929</td>\n",
       "      <td>691467</td>\n",
       "      <td>692450</td>\n",
       "      <td>PUL3</td>\n",
       "      <td>YNR062C</td>\n",
       "      <td>YNR062C</td>\n",
       "      <td>YNR062C_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Putative%20pulcherrimin%20transporter</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YNR062C_mRNA</td>\n",
       "      <td>YNR062C</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>RefSeq:NM_001183239.3</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5072</th>\n",
       "      <td>chrXI_tig00004932</td>\n",
       "      <td>469312</td>\n",
       "      <td>470934</td>\n",
       "      <td>MIC60,AIM28,FCJ1,FMP13</td>\n",
       "      <td>YKR016W</td>\n",
       "      <td>YKR016W</td>\n",
       "      <td>YKR016W_0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Component%20of%20the%20MICOS%20complex</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YKR016W_id001</td>\n",
       "      <td>YKR016W</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000290582</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5330</th>\n",
       "      <td>chrXV-4_tig00001046</td>\n",
       "      <td>849</td>\n",
       "      <td>1061</td>\n",
       "      <td>DGA1,diacylglycerol%20O-acyltransferase</td>\n",
       "      <td>YOR245C</td>\n",
       "      <td>YOR245C</td>\n",
       "      <td>YOR245C_0</td>\n",
       "      <td>0.169</td>\n",
       "      <td>Diacylglycerol%20acyltransferase</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>YOR245C_id001</td>\n",
       "      <td>YOR245C</td>\n",
       "      <td>GAL</td>\n",
       "      <td>0</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>None</td>\n",
       "      <td>SGD:S000292166</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   seq_id_x  start_x   end_x  \\\n",
       "157      chrIII_tig00004936   241438  245433   \n",
       "676       chrIV_tig00000001    46438   46547   \n",
       "874       chrIV_tig00000001   381046  383253   \n",
       "1610      chrIX_tig00000023   235689  236111   \n",
       "1694      chrIX_tig00000023   373545  374108   \n",
       "1920   chrVII-2_tig00000038    38804   40006   \n",
       "2292    chrVIII_tig00000017   445786  446360   \n",
       "2795      chrVI_tig00004938    12513   13592   \n",
       "2887      chrVI_tig00004938   229268  229777   \n",
       "2937       chrV_tig00000015      680    5190   \n",
       "3812  chrXIII-2_tig00000029   445067  446593   \n",
       "4065     chrXII_tig00000020     5009    5314   \n",
       "4787     chrXIV_tig00004929   660792  661424   \n",
       "4794     chrXIV_tig00004929   678731  680415   \n",
       "4800     chrXIV_tig00004929   691467  692450   \n",
       "5072      chrXI_tig00004932   469312  470934   \n",
       "5330    chrXV-4_tig00001046      849    1061   \n",
       "\n",
       "                                                  Alias     ID_x   Name_x  \\\n",
       "157   SSK22,mitogen-activated%20protein%20kinase%20k...  YCR073C  YCR073C   \n",
       "676                                                  HO  YDL227C  YDL227C   \n",
       "874   PRM7,YDL038C,pheromone-regulated%20protein%20PRM7  YDL039C  YDL039C   \n",
       "1610                                               None  YIL060W  YIL060W   \n",
       "1694                                              MET28  YIR017C  YIR017C   \n",
       "1920         TAD1,tRNA-specific%20adenosine%20deaminase  YGL243W  YGL243W   \n",
       "2292               CTR2,low-affinity%20Cu%20transporter  YHR175W  YHR175W   \n",
       "2795       AAD3,putative%20aryl-alcohol%20dehydrogenase  YCR107W  YCR107W   \n",
       "2887                                               ULI1  YFR026C  YFR026C   \n",
       "2937  YRF1-4,Y'%20element%20ATP-dependent%20helicase...  YLR466W  YLR466W   \n",
       "3812       YPK2,YKR2,putative%20protein%20kinase%20YPK2  YMR104C  YMR104C   \n",
       "4065                                               None  YLR162W  YLR162W   \n",
       "4787                                               MSO1  YNR049C  YNR049C   \n",
       "4794                                               BIO5  YNR056C  YNR056C   \n",
       "4800                                               PUL3  YNR062C  YNR062C   \n",
       "5072                             MIC60,AIM28,FCJ1,FMP13  YKR016W  YKR016W   \n",
       "5330            DGA1,diacylglycerol%20O-acyltransferase  YOR245C  YOR245C   \n",
       "\n",
       "     copy_num_ID coverage                                            display  \\\n",
       "157    YCR073C_0      1.0  MAP%20kinase%20kinase%20kinase%20of%20HOG1%20m...   \n",
       "676    YDL227C_0    0.062                       Site-specific%20endonuclease   \n",
       "874    YDL039C_0    0.998                      Pheromone-regulated%20protein   \n",
       "1610   YIL060W_0    0.908  Mitochondrial%20protein%20of%20unknown%20function   \n",
       "1694   YIR017C_0      1.0  bZIP%20transcriptional%20activator%20in%20the%...   \n",
       "1920   YGL243W_0      1.0              tRNA-specific%20adenosine%20deaminase   \n",
       "2292   YHR175W_0      1.0  Low-affinity%20copper%20transporter%20of%20the...   \n",
       "2795   YCR107W_0    0.982            Putative%20aryl-alcohol%20dehydrogenase   \n",
       "2887   YFR026C_0      1.0                  Protein%20of%20unknown%20function   \n",
       "2937   YLR466W_0    0.998  Helicase%20encoded%20by%20the%20Y'%20element%2...   \n",
       "3812   YMR104C_0    0.750  Protein%20kinase%20similar%20to%20S/T%20protei...   \n",
       "4065   YLR162W_0    0.778                  Protein%20of%20unknown%20function   \n",
       "4787   YNR049C_0      1.0  Lipid-interacting%20protein%20in%20SNARE%20com...   \n",
       "4794   YNR056C_0    0.999  Putative%20transmembrane%20protein%20involved%...   \n",
       "4800   YNR062C_0      1.0              Putative%20pulcherrimin%20transporter   \n",
       "5072   YKR016W_0      1.0             Component%20of%20the%20MICOS%20complex   \n",
       "5330   YOR245C_0    0.169                   Diacylglycerol%20acyltransferase   \n",
       "\n",
       "     extra_copy_number_x  ...         Name_y   Parent conditions  \\\n",
       "157                    0  ...   YCR073C_mRNA  YCR073C       None   \n",
       "676                    0  ...  YDL227C_id001  YDL227C        YPD   \n",
       "874                    0  ...   YDL039C_mRNA  YDL039C       None   \n",
       "1610                   0  ...   YIL060W_mRNA  YIL060W       None   \n",
       "1694                   0  ...  YIR017C_id002  YIR017C        GAL   \n",
       "1920                   0  ...  YGL243W_id002  YGL243W        GAL   \n",
       "2292                   0  ...  YHR175W_id001  YHR175W        GAL   \n",
       "2795                   0  ...   YCR107W_mRNA  YCR107W       None   \n",
       "2887                   0  ...  YFR026C_id001  YFR026C        GAL   \n",
       "2937                   0  ...   YLR466W_mRNA  YLR466W       None   \n",
       "3812                   0  ...   YMR104C_mRNA  YMR104C       None   \n",
       "4065                   0  ...   YLR162W_mRNA  YLR162W       None   \n",
       "4787                   0  ...  YNR049C_id003  YNR049C        GAL   \n",
       "4794                   0  ...  YNR056C_id001  YNR056C        GAL   \n",
       "4800                   0  ...   YNR062C_mRNA  YNR062C       None   \n",
       "5072                   0  ...  YKR016W_id001  YKR016W        GAL   \n",
       "5330                   0  ...  YOR245C_id001  YOR245C        GAL   \n",
       "\n",
       "     extra_copy_number_y inframe_stop_codon matches_ref_protein  \\\n",
       "157                    0               None               False   \n",
       "676                    0               None               False   \n",
       "874                    0               None               False   \n",
       "1610                   0               None               False   \n",
       "1694                   0               None               False   \n",
       "1920                   0               None               False   \n",
       "2292                   0               None               False   \n",
       "2795                   0               None               False   \n",
       "2887                   0               None               False   \n",
       "2937                   0               None               False   \n",
       "3812                   0               None               False   \n",
       "4065                   0               None               False   \n",
       "4787                   0               None               False   \n",
       "4794                   0               None               False   \n",
       "4800                   0               None               False   \n",
       "5072                   0               None               False   \n",
       "5330                   0               None               False   \n",
       "\n",
       "     missing_start_codon  missing_stop_codon          transcript_id valid_ORF  \n",
       "157                 True                None  RefSeq:NM_001178783.1     False  \n",
       "676                 True                None         SGD:S000293396     False  \n",
       "874                 True                None  RefSeq:NM_001180098.1     False  \n",
       "1610                True                None  RefSeq:NM_001179410.3     False  \n",
       "1694                True                None         SGD:S000290068     False  \n",
       "1920                True                None         SGD:S000289433     False  \n",
       "2292                True                None         SGD:S000289883     False  \n",
       "2795                True                None  RefSeq:NM_001178814.1     False  \n",
       "2887                True                None         SGD:S000289199     False  \n",
       "2937                True                None  RefSeq:NM_001182354.3     False  \n",
       "3812                True                None  RefSeq:NM_001182604.1     False  \n",
       "4065                True                None  RefSeq:NM_001182049.1     False  \n",
       "4787                True                None         SGD:S000291800     False  \n",
       "4794                True                None         SGD:S000291807     False  \n",
       "4800                True                None  RefSeq:NM_001183239.3     False  \n",
       "5072                True                None         SGD:S000290582     False  \n",
       "5330                True                None         SGD:S000292166     False  \n",
       "\n",
       "[17 rows x 30 columns]"
      ]
     },
     "execution_count": 297,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid[gene_mRNA_verified_unvalid['missing_start_codon'] == 'True']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "906aac04",
   "metadata": {},
   "source": [
    "# kick out duplicated genes who have a valid version"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "fc198634",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified_valid = gene_mRNA_verified[gene_mRNA_verified['valid_ORF'] == 'True' ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "8d77a96d",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid_list = set(gene_mRNA_verified_valid['Name_x'])\n",
    "verified_unvalid_list = set(gene_mRNA_verified_unvalid['Name_x'])\n",
    "dupl_val_unval = verified_valid_list & verified_unvalid_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "0a572a02",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "16"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 16 genes have 1 duplicated unvalid version I can forget about\n",
    "len(dupl_val_unval)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "id": "c215c360",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified_unvalid = gene_mRNA_verified_unvalid[~gene_mRNA_verified_unvalid['Name_x'].isin(list(dupl_val_unval))]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "id": "14c71a0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified_unvalid[gene_mRNA_verified_unvalid['missing_start_codon'] == 'True'].to_excel('missing_start.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "id": "29f7d6aa",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_mRNA_verified_unvalid[gene_mRNA_verified_unvalid['missing_stop_codon'] == 'True'].to_excel('missing_stop.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "95f8734a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "3a54ed6d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    88\n",
       "Name: missing_stop_codon, dtype: int64"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['missing_stop_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "id": "28ae03bb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True    23\n",
       "Name: inframe_stop_codon, dtype: int64"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_mRNA_verified_unvalid['inframe_stop_codon'].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e24de24f",
   "metadata": {},
   "source": [
    "# Inframe stop codon analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "id": "aac223ed",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframe_stop = gene_mRNA_verified_unvalid[gene_mRNA_verified_unvalid['inframe_stop_codon'] == 'True']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "id": "79b82a78",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframe_g = inframe_stop['Name_x']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "93ed4bc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframe_288c = gene_s288c[gene_attr_s288c['Name'].isin(inframe_g)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "24d688ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\2401597198.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  inframe_288c['lenght'] = inframe_288c['end'] - inframe_288c['start']\n"
     ]
    }
   ],
   "source": [
    "inframe_288c['lenght'] = inframe_288c['end'] - inframe_288c['start']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "id": "447ea22e",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\2496349264.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  inframe_stop['lenght'] = inframe_stop['end_x'] - inframe_stop['start_x']\n"
     ]
    }
   ],
   "source": [
    "inframe_stop['lenght'] = inframe_stop['end_x'] - inframe_stop['start_x']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "6f72bfe6",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframe_stop = inframe_stop[['Name_x','copy_num_ID', 'coverage', 'lenght']]\n",
    "inframe_288c = inframe_288c[['Name', 'lenght']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 237,
   "id": "c213b1b3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['seq_id', 'start', 'end', 'Name', 'gene', 'Alias', 'Note',\n",
       "       'Ontology_term', 'strand', 'phase', 'curie', 'orf_classification',\n",
       "       'display', 'Chromosome', 'lenght'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 237,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "inframe_288c.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "d3eddfbb",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframes_comp = pd.merge(inframe_288c, inframe_stop,  how='right', left_on='Name', right_on = 'Name_x')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "7651ea13",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframes_comp['diff'] = inframes_comp['lenght_x'] - inframes_comp['lenght_y']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "2d244275",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframes_comp = inframes_comp.sort_values(by='diff')\n",
    "inframes_comp = inframes_comp[['Name', 'lenght_x', 'lenght_y', 'diff']]\n",
    "inframes_comp.rename(columns = {'lenght_x': 'length s288c gene', 'lenght_y': 'length yps gene'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "6c3ce8c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "inframes_comp.to_excel('inframes.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "71e39f9a",
   "metadata": {},
   "source": [
    "## Unvalid ORFs to add to valid genes (because essential or valid ORF in y22 Liftoff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 356,
   "id": "124e6220",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_add = []\n",
    "with open('missing_stop_start_genes_to_add.txt', 'r')as file:\n",
    "    for l in file:\n",
    "        gene_to_add.append(l.strip('\\n'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "id": "bfbf8b9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_to_add += list(inframe_stop['Name_x'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "id": "31030907",
   "metadata": {},
   "outputs": [],
   "source": [
    "to_add = verified_unvalid[verified_unvalid['Name'].isin(gene_to_add)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "id": "209ad6bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\user\\AppData\\Local\\Temp\\ipykernel_2116\\907560873.py:1: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  verified_valid = verified_valid.append(to_add)\n"
     ]
    }
   ],
   "source": [
    "verified_valid = verified_valid.append(to_add)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 362,
   "id": "557e171e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5160, 23)"
      ]
     },
     "execution_count": 362,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_valid.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5b9c3f32",
   "metadata": {},
   "source": [
    "# Overlapped annotation, that's really problematic....\n",
    "Find overlap and deal with them. keep the genes with highest % identity "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "id": "05e981ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "# find genes with major overlap (I only round the last digit of start and end position to zero, \n",
    "# so not perfect but will cover the biggest overlaps)\n",
    "id_overlap = []\n",
    "dupl = {}\n",
    "for ind, row in verified_valid.iterrows():\n",
    "    start = str(row['start'])\n",
    "    start = start.replace(start[len(start) - 1:], '0')\n",
    "    end = str(row['end'])\n",
    "    end = end.replace(end[len(end) - 1:], '0')\n",
    "\n",
    "    tmp = str(row['seq_id']) + start + end\n",
    "    id_overlap.append(tmp)\n",
    "    if tmp in dupl:\n",
    "        dupl[tmp] += 1\n",
    "    else:\n",
    "        dupl[tmp] = 1\n",
    "verified_valid['overlap id'] = id_overlap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 366,
   "id": "88e74c52",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "30"
      ]
     },
     "execution_count": 366,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filtered_dict = {k:v for (k,v) in dupl.items() if v > 1}\n",
    "len(filtered_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "id": "12498a66",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid['sequence_ID'] = verified_valid['sequence_ID'].astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "id": "314876e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "ind = []\n",
    "for k in filtered_dict.keys():\n",
    "    tmp = verified_valid[verified_valid['overlap id'] == k]\n",
    "    ind.append(tmp.nsmallest(1,\"sequence_ID\").index[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 379,
   "id": "457a695a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5160, 24)"
      ]
     },
     "execution_count": 379,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_valid.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "id": "cbd07460",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(5130, 24)"
      ]
     },
     "execution_count": 380,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "verified_valid.drop(index=ind, inplace=True)\n",
    "verified_valid.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "id": "c03e225b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ok, I think I'm good now, let's export this and start the model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "id": "593b2aeb",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid_sh = verified_valid[['Name', 'gene', 'Chromosome']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "65c60d00",
   "metadata": {},
   "outputs": [],
   "source": [
    "verified_valid_sh.to_csv('yps1009_verified_genes_final.txt', sep = '\\t', usin)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
